{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import numpy as np\n",
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "# read vcf files\n",
    "vcf_reader = open('data/gavin_input-gavin.vcf','r')\n",
    "column_lines = []\n",
    "info = []\n",
    "sample_lines = []\n",
    "for line in vcf_reader.readlines():\n",
    "    if line.startswith('#CHROM'):\n",
    "        column_lines = line.strip('\\n').strip('#').split('\\t')\n",
    "    elif line.startswith('##INFO'):\n",
    "        info.append(line.strip('\\n').strip('#'))\n",
    "    elif line[0].isdigit():\n",
    "        line = line.strip('\\n')\n",
    "        sample_lines.append(line)\n",
    "        \n",
    "# parse the column names\n",
    "column_lines = column_lines[:-1]\n",
    "info_contents = [line[line.find('INFO=ID=')+len('INFO=ID=')+2:line.find(',')] for line in info[:-1]]\n",
    "line = info[-1]\n",
    "info_last = line[line.find(\"annotations: '\")+len(\"annotations: '\")+2:-3].split('|')\n",
    "columns = column_lines+info_contents+info_last\n",
    "columns = [col.strip(' ') for col in columns]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['CHROM',\n",
       " 'POS',\n",
       " 'ID',\n",
       " 'REF',\n",
       " 'ALT',\n",
       " 'QUAL',\n",
       " 'FILTER',\n",
       " 'CADD',\n",
       " 'CADD_SCALED',\n",
       " 'EXAC_AF',\n",
       " 'EXAC_AC_HOM',\n",
       " 'EXAC_AC_HET',\n",
       " 't_Allele',\n",
       " 'Gene_Name',\n",
       " 'Annotation',\n",
       " 'Putative_impact',\n",
       " 'Gene_ID',\n",
       " 'Feature_type',\n",
       " 'Feature_ID',\n",
       " 'Transcript_biotype',\n",
       " 'Rank_total',\n",
       " 'HGVS_c',\n",
       " 'HGVS_p',\n",
       " 'cDNA_position',\n",
       " 'CDS_position',\n",
       " 'Protein_position',\n",
       " 'Distance_to_feature',\n",
       " 'Errors',\n",
       " 'Classification',\n",
       " 'Confidence',\n",
       " 'Reason']"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "def cut_str(string,begin,end):\n",
    "    '''\n",
    "    cheery pick the begin till end of a string\n",
    "    '''\n",
    "    if begin not in string:\n",
    "        return ''\n",
    "    begin_ind = string.find(begin)+len(begin)\n",
    "    cut = string[begin_ind:]\n",
    "    end_ind = cut.find(end)+len(end)-1\n",
    "    return cut[:end_ind]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "# samples = []\n",
    "# for line in sample_lines:\n",
    "#     var_info = line.split('\\t')[:-1]\n",
    "#     if ',' in line:\n",
    "#         var = line.split('\\t')[-1].split(',')\n",
    "#     else:\n",
    "#         var=line.split('\\t')[-1]\n",
    "#     for gene in var:\n",
    "#         if 'MYO5B' in gene:\n",
    "#             var_gene = []\n",
    "#             var_gene_infos = gene.split(';')[-1].split('|')\n",
    "#             var_gene.append(cut_str(gene,'CADD=',';'))\n",
    "#             var_gene.append(cut_str(gene,'CADD_SCALED=',';'))\n",
    "#             var_gene.append(cut_str(gene,'EXAC_AF=',';'))\n",
    "#             var_gene.append(cut_str(gene,'EXAC_AC_HOM=',';'))\n",
    "#             var_gene.append(cut_str(gene,'EXAC_AC_HET=',';'))\n",
    "#     samples.append(var_info+var_gene+var_gene_infos)\n",
    "# for sample in samples:\n",
    "#     assert len(sample)==31"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "# parse the samples\n",
    "samples = []\n",
    "for line in sample_lines:\n",
    "    line = line.strip(' ')\n",
    "    var_info = line.split('\\t')[:-1]\n",
    "    if ',' in line:\n",
    "        genes = line.split('\\t')[-1].split(',')\n",
    "        for gene in genes:\n",
    "            if 'MYO5B' in gene:\n",
    "                myo_info = gene\n",
    "                var_gene_infos = myo_info.split(';')[-1].split('|')\n",
    "    else:\n",
    "        myo_info=line.split('\\t')[-1]\n",
    "        var_gene_infos = myo_info.split(';')[-2].split('|')\n",
    "    var_gene = []    \n",
    "    var_gene.append(cut_str(myo_info,'CADD=',';'))\n",
    "    var_gene.append(cut_str(myo_info,'CADD_SCALED=',';'))\n",
    "    var_gene.append(cut_str(myo_info,'EXAC_AF=',';'))\n",
    "    var_gene.append(cut_str(myo_info,'EXAC_AC_HOM=',';'))\n",
    "    var_gene.append(cut_str(myo_info,'EXAC_AC_HET=',';'))\n",
    "    samples.append(var_info+var_gene+var_gene_infos)\n",
    "for sample in samples:\n",
    "    assert len(sample)==31"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "datatable = pd.DataFrame(samples,columns=columns)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>CADD</th>\n",
       "      <th>CADD_SCALED</th>\n",
       "      <th>EXAC_AF</th>\n",
       "      <th>...</th>\n",
       "      <th>HGVS_c</th>\n",
       "      <th>HGVS_p</th>\n",
       "      <th>cDNA_position</th>\n",
       "      <th>CDS_position</th>\n",
       "      <th>Protein_position</th>\n",
       "      <th>Distance_to_feature</th>\n",
       "      <th>Errors</th>\n",
       "      <th>Classification</th>\n",
       "      <th>Confidence</th>\n",
       "      <th>Reason</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14</td>\n",
       "      <td>23882063</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>1.880579</td>\n",
       "      <td>15.47</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>c.5808G&gt;A</td>\n",
       "      <td>p.Ter1936Ter</td>\n",
       "      <td>5939/6055</td>\n",
       "      <td>5808/5808</td>\n",
       "      <td>1936/1935</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Benign</td>\n",
       "      <td>calibrated</td>\n",
       "      <td>Variant CADD score of 15.47 is less than 15.62...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14</td>\n",
       "      <td>23882064</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>1.823769</td>\n",
       "      <td>15.13</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>c.5807A&gt;G</td>\n",
       "      <td>p.Ter1936Trpext*?</td>\n",
       "      <td>5938/6055</td>\n",
       "      <td>5807/5808</td>\n",
       "      <td>1936/1935</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Benign</td>\n",
       "      <td>calibrated</td>\n",
       "      <td>Variant CADD score of 15.13 is less than 15.62...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14</td>\n",
       "      <td>23882082</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4.4282580000000005</td>\n",
       "      <td>24.2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>c.5791-2A&gt;G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>calibrated</td>\n",
       "      <td>Variant CADD score of 24.2 is greater than 22....</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>14</td>\n",
       "      <td>23882975</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>3.5186269999999995</td>\n",
       "      <td>23.1</td>\n",
       "      <td>8.237E-6</td>\n",
       "      <td>...</td>\n",
       "      <td>c.5783G&gt;A</td>\n",
       "      <td>p.Gly1928Asp</td>\n",
       "      <td>5914/6055</td>\n",
       "      <td>5783/5808</td>\n",
       "      <td>1928/1935</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>calibrated</td>\n",
       "      <td>Variant CADD score of 23.1 is greater than 22....</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>14</td>\n",
       "      <td>23882976</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>5.615639</td>\n",
       "      <td>26.6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>c.5782G&gt;T</td>\n",
       "      <td>p.Gly1928Cys</td>\n",
       "      <td>5913/6055</td>\n",
       "      <td>5782/5808</td>\n",
       "      <td>1928/1935</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>calibrated</td>\n",
       "      <td>Variant CADD score of 26.6 is greater than 22....</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>14</td>\n",
       "      <td>23882984</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>6.322065</td>\n",
       "      <td>29.2</td>\n",
       "      <td>8.237E-6</td>\n",
       "      <td>...</td>\n",
       "      <td>c.5774G&gt;A</td>\n",
       "      <td>p.Arg1925His</td>\n",
       "      <td>5905/6055</td>\n",
       "      <td>5774/5808</td>\n",
       "      <td>1925/1935</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>calibrated</td>\n",
       "      <td>Variant CADD score of 29.2 is greater than 22....</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>14</td>\n",
       "      <td>23882985</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>8.175613</td>\n",
       "      <td>35.0</td>\n",
       "      <td>8.237E-6</td>\n",
       "      <td>...</td>\n",
       "      <td>c.5773C&gt;T</td>\n",
       "      <td>p.Arg1925Cys</td>\n",
       "      <td>5904/6055</td>\n",
       "      <td>5773/5808</td>\n",
       "      <td>1925/1935</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>calibrated</td>\n",
       "      <td>Variant CADD score of 35.0 is greater than 22....</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>14</td>\n",
       "      <td>23882993</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>3.586021</td>\n",
       "      <td>23.2</td>\n",
       "      <td>8.236E-6</td>\n",
       "      <td>...</td>\n",
       "      <td>c.5765C&gt;T</td>\n",
       "      <td>p.Ala1922Val</td>\n",
       "      <td>5896/6055</td>\n",
       "      <td>5765/5808</td>\n",
       "      <td>1922/1935</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>calibrated</td>\n",
       "      <td>Variant CADD score of 23.2 is greater than 22....</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>14</td>\n",
       "      <td>23882997</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>6.560517999999999</td>\n",
       "      <td>31.0</td>\n",
       "      <td>8.237E-6</td>\n",
       "      <td>...</td>\n",
       "      <td>c.5761C&gt;T</td>\n",
       "      <td>p.Arg1921Trp</td>\n",
       "      <td>5892/6055</td>\n",
       "      <td>5761/5808</td>\n",
       "      <td>1921/1935</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>calibrated</td>\n",
       "      <td>Variant CADD score of 31.0 is greater than 22....</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>14</td>\n",
       "      <td>23883009</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>7.421627</td>\n",
       "      <td>34.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>c.5749G&gt;T</td>\n",
       "      <td>p.Val1917Phe</td>\n",
       "      <td>5880/6055</td>\n",
       "      <td>5749/5808</td>\n",
       "      <td>1917/1935</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>calibrated</td>\n",
       "      <td>Variant CADD score of 34.0 is greater than 22....</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>14</td>\n",
       "      <td>23883014</td>\n",
       "      <td>.</td>\n",
       "      <td>GACT</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>3.300168</td>\n",
       "      <td>22.9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>c.5741_5744delAGTCinsT</td>\n",
       "      <td>p.Glu1914_Ser1915delinsVal</td>\n",
       "      <td>5875/6055</td>\n",
       "      <td>5741/5808</td>\n",
       "      <td>1914/1935</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>calibrated</td>\n",
       "      <td>Variant CADD score of 22.9 is greater than 22....</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>14</td>\n",
       "      <td>23883018</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>7.559069</td>\n",
       "      <td>34.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>c.5740G&gt;A</td>\n",
       "      <td>p.Glu1914Lys</td>\n",
       "      <td>5871/6055</td>\n",
       "      <td>5740/5808</td>\n",
       "      <td>1914/1935</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>calibrated</td>\n",
       "      <td>Variant CADD score of 34.0 is greater than 22....</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>14</td>\n",
       "      <td>23883020</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>6.566591000000001</td>\n",
       "      <td>31.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>c.5738C&gt;T</td>\n",
       "      <td>p.Ala1913Val</td>\n",
       "      <td>5869/6055</td>\n",
       "      <td>5738/5808</td>\n",
       "      <td>1913/1935</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>calibrated</td>\n",
       "      <td>Variant CADD score of 31.0 is greater than 22....</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>14</td>\n",
       "      <td>23883021</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>7.710235000000001</td>\n",
       "      <td>35.0</td>\n",
       "      <td>8.236E-6</td>\n",
       "      <td>...</td>\n",
       "      <td>c.5737G&gt;A</td>\n",
       "      <td>p.Ala1913Thr</td>\n",
       "      <td>5868/6055</td>\n",
       "      <td>5737/5808</td>\n",
       "      <td>1913/1935</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>calibrated</td>\n",
       "      <td>Variant CADD score of 35.0 is greater than 22....</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>14</td>\n",
       "      <td>23883032</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>7.93867</td>\n",
       "      <td>35.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>c.5726G&gt;C</td>\n",
       "      <td>p.Arg1909Pro</td>\n",
       "      <td>5857/6055</td>\n",
       "      <td>5726/5808</td>\n",
       "      <td>1909/1935</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>calibrated</td>\n",
       "      <td>Variant CADD score of 35.0 is greater than 22....</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>14</td>\n",
       "      <td>23883032</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>8.266119</td>\n",
       "      <td>35.0</td>\n",
       "      <td>8.236E-6</td>\n",
       "      <td>...</td>\n",
       "      <td>c.5726G&gt;A</td>\n",
       "      <td>p.Arg1909Gln</td>\n",
       "      <td>5857/6055</td>\n",
       "      <td>5726/5808</td>\n",
       "      <td>1909/1935</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>calibrated</td>\n",
       "      <td>Variant CADD score of 35.0 is greater than 22....</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>14</td>\n",
       "      <td>23883033</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>7.408993</td>\n",
       "      <td>34.0</td>\n",
       "      <td>3.295E-5</td>\n",
       "      <td>...</td>\n",
       "      <td>c.5725C&gt;T</td>\n",
       "      <td>p.Arg1909Trp</td>\n",
       "      <td>5856/6055</td>\n",
       "      <td>5725/5808</td>\n",
       "      <td>1909/1935</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>calibrated</td>\n",
       "      <td>Variant CADD score of 34.0 is greater than 22....</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>14</td>\n",
       "      <td>23883041</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>6.666103</td>\n",
       "      <td>32.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>c.5717C&gt;G</td>\n",
       "      <td>p.Ala1906Gly</td>\n",
       "      <td>5848/6055</td>\n",
       "      <td>5717/5808</td>\n",
       "      <td>1906/1935</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>calibrated</td>\n",
       "      <td>Variant CADD score of 32.0 is greater than 22....</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>14</td>\n",
       "      <td>23883054</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>8.245451</td>\n",
       "      <td>35.0</td>\n",
       "      <td>1.647E-5</td>\n",
       "      <td>...</td>\n",
       "      <td>c.5704G&gt;A</td>\n",
       "      <td>p.Glu1902Lys</td>\n",
       "      <td>5835/6055</td>\n",
       "      <td>5704/5808</td>\n",
       "      <td>1902/1935</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>calibrated</td>\n",
       "      <td>Variant CADD score of 35.0 is greater than 22....</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>14</td>\n",
       "      <td>23883056</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>5.867249</td>\n",
       "      <td>27.3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>c.5702A&gt;T</td>\n",
       "      <td>p.His1901Leu</td>\n",
       "      <td>5833/6055</td>\n",
       "      <td>5702/5808</td>\n",
       "      <td>1901/1935</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>calibrated</td>\n",
       "      <td>Variant CADD score of 27.3 is greater than 22....</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>14</td>\n",
       "      <td>23883059</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>6.9252720000000005</td>\n",
       "      <td>33.0</td>\n",
       "      <td>8.236E-6</td>\n",
       "      <td>...</td>\n",
       "      <td>c.5699A&gt;T</td>\n",
       "      <td>p.Gln1900Leu</td>\n",
       "      <td>5830/6055</td>\n",
       "      <td>5699/5808</td>\n",
       "      <td>1900/1935</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>calibrated</td>\n",
       "      <td>Variant CADD score of 33.0 is greater than 22....</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>14</td>\n",
       "      <td>23883062</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>2.354094</td>\n",
       "      <td>18.52</td>\n",
       "      <td>8.236E-6</td>\n",
       "      <td>...</td>\n",
       "      <td>c.5696T&gt;C</td>\n",
       "      <td>p.Val1899Ala</td>\n",
       "      <td>5827/6055</td>\n",
       "      <td>5696/5808</td>\n",
       "      <td>1899/1935</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>genomewide</td>\n",
       "      <td>Variant MAF of 8.236E-6 is rare enough to be p...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>14</td>\n",
       "      <td>23883068</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>7.939234</td>\n",
       "      <td>35.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>c.5690G&gt;A</td>\n",
       "      <td>p.Arg1897His</td>\n",
       "      <td>5821/6055</td>\n",
       "      <td>5690/5808</td>\n",
       "      <td>1897/1935</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>calibrated</td>\n",
       "      <td>Variant CADD score of 35.0 is greater than 22....</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>14</td>\n",
       "      <td>23883069</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>8.357856</td>\n",
       "      <td>35.0</td>\n",
       "      <td>1.647E-5</td>\n",
       "      <td>...</td>\n",
       "      <td>c.5689C&gt;T</td>\n",
       "      <td>p.Arg1897Cys</td>\n",
       "      <td>5820/6055</td>\n",
       "      <td>5689/5808</td>\n",
       "      <td>1897/1935</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>calibrated</td>\n",
       "      <td>Variant CADD score of 35.0 is greater than 22....</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>14</td>\n",
       "      <td>23883071</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>5.127199</td>\n",
       "      <td>25.4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>c.5687T&gt;G</td>\n",
       "      <td>p.Phe1896Cys</td>\n",
       "      <td>5818/6055</td>\n",
       "      <td>5687/5808</td>\n",
       "      <td>1896/1935</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>calibrated</td>\n",
       "      <td>Variant CADD score of 25.4 is greater than 22....</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>14</td>\n",
       "      <td>23883071</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>6.718649</td>\n",
       "      <td>32.0</td>\n",
       "      <td>8.236E-6</td>\n",
       "      <td>...</td>\n",
       "      <td>c.5687T&gt;C</td>\n",
       "      <td>p.Phe1896Ser</td>\n",
       "      <td>5818/6055</td>\n",
       "      <td>5687/5808</td>\n",
       "      <td>1896/1935</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>calibrated</td>\n",
       "      <td>Variant CADD score of 32.0 is greater than 22....</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>14</td>\n",
       "      <td>23883092</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>6.177906</td>\n",
       "      <td>28.6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>c.5666C&gt;T</td>\n",
       "      <td>p.Ala1889Val</td>\n",
       "      <td>5797/6055</td>\n",
       "      <td>5666/5808</td>\n",
       "      <td>1889/1935</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>calibrated</td>\n",
       "      <td>Variant CADD score of 28.6 is greater than 22....</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>14</td>\n",
       "      <td>23883098</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>6.721912</td>\n",
       "      <td>32.0</td>\n",
       "      <td>8.236E-6</td>\n",
       "      <td>...</td>\n",
       "      <td>c.5660A&gt;G</td>\n",
       "      <td>p.Glu1887Gly</td>\n",
       "      <td>5791/6055</td>\n",
       "      <td>5660/5808</td>\n",
       "      <td>1887/1935</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>calibrated</td>\n",
       "      <td>Variant CADD score of 32.0 is greater than 22....</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>14</td>\n",
       "      <td>23883101</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>6.701846000000001</td>\n",
       "      <td>32.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>c.5657A&gt;G</td>\n",
       "      <td>p.Glu1886Gly</td>\n",
       "      <td>5788/6055</td>\n",
       "      <td>5657/5808</td>\n",
       "      <td>1886/1935</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>calibrated</td>\n",
       "      <td>Variant CADD score of 32.0 is greater than 22....</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>14</td>\n",
       "      <td>23883216</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>1.823764</td>\n",
       "      <td>15.13</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>c.5655G&gt;A</td>\n",
       "      <td>p.Ala1885Ala</td>\n",
       "      <td>5786/6055</td>\n",
       "      <td>5655/5808</td>\n",
       "      <td>1885/1935</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Benign</td>\n",
       "      <td>calibrated</td>\n",
       "      <td>Variant CADD score of 15.13 is less than 15.62...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>820</th>\n",
       "      <td>14</td>\n",
       "      <td>23902005</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>3.871197</td>\n",
       "      <td>23.5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>c.346-1G&gt;A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>calibrated</td>\n",
       "      <td>Variant CADD score of 23.5 is greater than 22....</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>821</th>\n",
       "      <td>14</td>\n",
       "      <td>23902297</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4.4865330000000005</td>\n",
       "      <td>24.3</td>\n",
       "      <td>8.236E-6</td>\n",
       "      <td>...</td>\n",
       "      <td>c.341T&gt;C</td>\n",
       "      <td>p.Ile114Thr</td>\n",
       "      <td>472/6055</td>\n",
       "      <td>341/5808</td>\n",
       "      <td>114/1935</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>calibrated</td>\n",
       "      <td>Variant CADD score of 24.3 is greater than 22....</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>822</th>\n",
       "      <td>14</td>\n",
       "      <td>23902298</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>5.20941</td>\n",
       "      <td>25.6</td>\n",
       "      <td>8.236E-6</td>\n",
       "      <td>...</td>\n",
       "      <td>c.340A&gt;T</td>\n",
       "      <td>p.Ile114Phe</td>\n",
       "      <td>471/6055</td>\n",
       "      <td>340/5808</td>\n",
       "      <td>114/1935</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>calibrated</td>\n",
       "      <td>Variant CADD score of 25.6 is greater than 22....</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>823</th>\n",
       "      <td>14</td>\n",
       "      <td>23902299</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>2.130051</td>\n",
       "      <td>17.05</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>c.339G&gt;C</td>\n",
       "      <td>p.Met113Ile</td>\n",
       "      <td>470/6055</td>\n",
       "      <td>339/5808</td>\n",
       "      <td>113/1935</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>genomewide</td>\n",
       "      <td>Variant MAF of 0.0 is rare enough to be potent...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>824</th>\n",
       "      <td>14</td>\n",
       "      <td>23902310</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4.304563</td>\n",
       "      <td>24.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>c.328G&gt;T</td>\n",
       "      <td>p.Gly110Cys</td>\n",
       "      <td>459/6055</td>\n",
       "      <td>328/5808</td>\n",
       "      <td>110/1935</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>calibrated</td>\n",
       "      <td>Variant CADD score of 24.0 is greater than 22....</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>825</th>\n",
       "      <td>14</td>\n",
       "      <td>23902340</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>3.3837330000000003</td>\n",
       "      <td>23.0</td>\n",
       "      <td>8.236E-6</td>\n",
       "      <td>...</td>\n",
       "      <td>c.298G&gt;T</td>\n",
       "      <td>p.Ala100Ser</td>\n",
       "      <td>429/6055</td>\n",
       "      <td>298/5808</td>\n",
       "      <td>100/1935</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>calibrated</td>\n",
       "      <td>Variant CADD score of 23.0 is greater than 22....</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>826</th>\n",
       "      <td>14</td>\n",
       "      <td>23902364</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>2.694014</td>\n",
       "      <td>20.8</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>c.274A&gt;G</td>\n",
       "      <td>p.Met92Val</td>\n",
       "      <td>405/6055</td>\n",
       "      <td>274/5808</td>\n",
       "      <td>92/1935</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>genomewide</td>\n",
       "      <td>Variant MAF of 0.0 is rare enough to be potent...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>827</th>\n",
       "      <td>14</td>\n",
       "      <td>23902366</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>6.136413</td>\n",
       "      <td>28.4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>c.272C&gt;T</td>\n",
       "      <td>p.Ala91Val</td>\n",
       "      <td>403/6055</td>\n",
       "      <td>272/5808</td>\n",
       "      <td>91/1935</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>calibrated</td>\n",
       "      <td>Variant CADD score of 28.4 is greater than 22....</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>828</th>\n",
       "      <td>14</td>\n",
       "      <td>23902426</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>2.922024</td>\n",
       "      <td>21.9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>c.212T&gt;C</td>\n",
       "      <td>p.Val71Ala</td>\n",
       "      <td>343/6055</td>\n",
       "      <td>212/5808</td>\n",
       "      <td>71/1935</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>genomewide</td>\n",
       "      <td>Variant MAF of 0.0 is rare enough to be potent...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>829</th>\n",
       "      <td>14</td>\n",
       "      <td>23902430</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>1.0409959999999998</td>\n",
       "      <td>10.89</td>\n",
       "      <td>8.236E-6</td>\n",
       "      <td>...</td>\n",
       "      <td>c.208A&gt;T</td>\n",
       "      <td>p.Thr70Ser</td>\n",
       "      <td>339/6055</td>\n",
       "      <td>208/5808</td>\n",
       "      <td>70/1935</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Benign</td>\n",
       "      <td>calibrated</td>\n",
       "      <td>Variant CADD score of 10.89 is less than 15.62...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>830</th>\n",
       "      <td>14</td>\n",
       "      <td>23902758</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>1.7601419999999999</td>\n",
       "      <td>14.75</td>\n",
       "      <td>8.236E-6</td>\n",
       "      <td>...</td>\n",
       "      <td>c.184G&gt;A</td>\n",
       "      <td>p.Glu62Lys</td>\n",
       "      <td>315/6055</td>\n",
       "      <td>184/5808</td>\n",
       "      <td>62/1935</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Benign</td>\n",
       "      <td>calibrated</td>\n",
       "      <td>Variant CADD score of 14.75 is less than 15.62...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>831</th>\n",
       "      <td>14</td>\n",
       "      <td>23902767</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>5.563407</td>\n",
       "      <td>26.4</td>\n",
       "      <td>8.236E-6</td>\n",
       "      <td>...</td>\n",
       "      <td>c.175G&gt;T</td>\n",
       "      <td>p.Val59Phe</td>\n",
       "      <td>306/6055</td>\n",
       "      <td>175/5808</td>\n",
       "      <td>59/1935</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>calibrated</td>\n",
       "      <td>Variant CADD score of 26.4 is greater than 22....</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>832</th>\n",
       "      <td>14</td>\n",
       "      <td>23902767</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>2.010589</td>\n",
       "      <td>16.28</td>\n",
       "      <td>8.236E-6</td>\n",
       "      <td>...</td>\n",
       "      <td>c.175G&gt;A</td>\n",
       "      <td>p.Val59Ile</td>\n",
       "      <td>306/6055</td>\n",
       "      <td>175/5808</td>\n",
       "      <td>59/1935</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>genomewide</td>\n",
       "      <td>Variant MAF of 8.236E-6 is rare enough to be p...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>833</th>\n",
       "      <td>14</td>\n",
       "      <td>23902776</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4.707939</td>\n",
       "      <td>24.6</td>\n",
       "      <td>8.236E-6</td>\n",
       "      <td>...</td>\n",
       "      <td>c.166G&gt;A</td>\n",
       "      <td>p.Gly56Ser</td>\n",
       "      <td>297/6055</td>\n",
       "      <td>166/5808</td>\n",
       "      <td>56/1935</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>calibrated</td>\n",
       "      <td>Variant CADD score of 24.6 is greater than 22....</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>834</th>\n",
       "      <td>14</td>\n",
       "      <td>23902781</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4.868483</td>\n",
       "      <td>24.9</td>\n",
       "      <td>8.236E-6</td>\n",
       "      <td>...</td>\n",
       "      <td>c.161G&gt;T</td>\n",
       "      <td>p.Arg54Leu</td>\n",
       "      <td>292/6055</td>\n",
       "      <td>161/5808</td>\n",
       "      <td>54/1935</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>calibrated</td>\n",
       "      <td>Variant CADD score of 24.9 is greater than 22....</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>835</th>\n",
       "      <td>14</td>\n",
       "      <td>23902782</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>3.990305</td>\n",
       "      <td>23.6</td>\n",
       "      <td>8.236E-6</td>\n",
       "      <td>...</td>\n",
       "      <td>c.160C&gt;G</td>\n",
       "      <td>p.Arg54Gly</td>\n",
       "      <td>291/6055</td>\n",
       "      <td>160/5808</td>\n",
       "      <td>54/1935</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>calibrated</td>\n",
       "      <td>Variant CADD score of 23.6 is greater than 22....</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>836</th>\n",
       "      <td>14</td>\n",
       "      <td>23902788</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>-1.5471270000000001</td>\n",
       "      <td>0.002</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>c.154G&gt;A</td>\n",
       "      <td>p.Val52Met</td>\n",
       "      <td>285/6055</td>\n",
       "      <td>154/5808</td>\n",
       "      <td>52/1935</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Benign</td>\n",
       "      <td>calibrated</td>\n",
       "      <td>Variant CADD score of 0.002 is less than 15.62...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>837</th>\n",
       "      <td>14</td>\n",
       "      <td>23902804</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>2.169108</td>\n",
       "      <td>17.31</td>\n",
       "      <td>8.236E-6</td>\n",
       "      <td>...</td>\n",
       "      <td>c.138T&gt;A</td>\n",
       "      <td>p.Phe46Leu</td>\n",
       "      <td>269/6055</td>\n",
       "      <td>138/5808</td>\n",
       "      <td>46/1935</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>genomewide</td>\n",
       "      <td>Variant MAF of 8.236E-6 is rare enough to be p...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>838</th>\n",
       "      <td>14</td>\n",
       "      <td>23902818</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>3.204917</td>\n",
       "      <td>22.7</td>\n",
       "      <td>8.236E-6</td>\n",
       "      <td>...</td>\n",
       "      <td>c.124G&gt;A</td>\n",
       "      <td>p.Asp42Asn</td>\n",
       "      <td>255/6055</td>\n",
       "      <td>124/5808</td>\n",
       "      <td>42/1935</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>genomewide</td>\n",
       "      <td>Variant MAF of 8.236E-6 is rare enough to be p...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>839</th>\n",
       "      <td>14</td>\n",
       "      <td>23902821</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>6.421265</td>\n",
       "      <td>29.7</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>c.121G&gt;A</td>\n",
       "      <td>p.Asp41Asn</td>\n",
       "      <td>252/6055</td>\n",
       "      <td>121/5808</td>\n",
       "      <td>41/1935</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>calibrated</td>\n",
       "      <td>Variant CADD score of 29.7 is greater than 22....</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>840</th>\n",
       "      <td>14</td>\n",
       "      <td>23902851</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>6.160776</td>\n",
       "      <td>28.5</td>\n",
       "      <td>8.236E-6</td>\n",
       "      <td>...</td>\n",
       "      <td>c.91T&gt;C</td>\n",
       "      <td>p.Phe31Leu</td>\n",
       "      <td>222/6055</td>\n",
       "      <td>91/5808</td>\n",
       "      <td>31/1935</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>calibrated</td>\n",
       "      <td>Variant CADD score of 28.5 is greater than 22....</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>841</th>\n",
       "      <td>14</td>\n",
       "      <td>23902866</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4.774503</td>\n",
       "      <td>24.7</td>\n",
       "      <td>8.236E-6</td>\n",
       "      <td>...</td>\n",
       "      <td>c.76G&gt;A</td>\n",
       "      <td>p.Ala26Thr</td>\n",
       "      <td>207/6055</td>\n",
       "      <td>76/5808</td>\n",
       "      <td>26/1935</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>calibrated</td>\n",
       "      <td>Variant CADD score of 24.7 is greater than 22....</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>842</th>\n",
       "      <td>14</td>\n",
       "      <td>23902888</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>5.255416</td>\n",
       "      <td>25.7</td>\n",
       "      <td>8.237E-6</td>\n",
       "      <td>...</td>\n",
       "      <td>c.54G&gt;C</td>\n",
       "      <td>p.Lys18Asn</td>\n",
       "      <td>185/6055</td>\n",
       "      <td>54/5808</td>\n",
       "      <td>18/1935</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>calibrated</td>\n",
       "      <td>Variant CADD score of 25.7 is greater than 22....</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>843</th>\n",
       "      <td>14</td>\n",
       "      <td>23902892</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>7.612844</td>\n",
       "      <td>34.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>c.50G&gt;A</td>\n",
       "      <td>p.Arg17His</td>\n",
       "      <td>181/6055</td>\n",
       "      <td>50/5808</td>\n",
       "      <td>17/1935</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>calibrated</td>\n",
       "      <td>Variant CADD score of 34.0 is greater than 22....</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>844</th>\n",
       "      <td>14</td>\n",
       "      <td>23902901</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4.636164</td>\n",
       "      <td>24.5</td>\n",
       "      <td>8.236E-6</td>\n",
       "      <td>...</td>\n",
       "      <td>c.41C&gt;T</td>\n",
       "      <td>p.Pro14Leu</td>\n",
       "      <td>172/6055</td>\n",
       "      <td>41/5808</td>\n",
       "      <td>14/1935</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>calibrated</td>\n",
       "      <td>Variant CADD score of 24.5 is greater than 22....</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>845</th>\n",
       "      <td>14</td>\n",
       "      <td>23902905</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>7.042627</td>\n",
       "      <td>33.0</td>\n",
       "      <td>8.236E-6</td>\n",
       "      <td>...</td>\n",
       "      <td>c.37G&gt;A</td>\n",
       "      <td>p.Ala13Thr</td>\n",
       "      <td>168/6055</td>\n",
       "      <td>37/5808</td>\n",
       "      <td>13/1935</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>calibrated</td>\n",
       "      <td>Variant CADD score of 33.0 is greater than 22....</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>846</th>\n",
       "      <td>14</td>\n",
       "      <td>23902913</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>5.493546</td>\n",
       "      <td>26.2</td>\n",
       "      <td>2.471E-5</td>\n",
       "      <td>...</td>\n",
       "      <td>c.29G&gt;C</td>\n",
       "      <td>p.Gly10Ala</td>\n",
       "      <td>160/6055</td>\n",
       "      <td>29/5808</td>\n",
       "      <td>10/1935</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>calibrated</td>\n",
       "      <td>Variant CADD score of 26.2 is greater than 22....</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>847</th>\n",
       "      <td>14</td>\n",
       "      <td>23902920</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.9583229999999999</td>\n",
       "      <td>10.42</td>\n",
       "      <td>8.236E-6</td>\n",
       "      <td>...</td>\n",
       "      <td>c.22G&gt;C</td>\n",
       "      <td>p.Val8Leu</td>\n",
       "      <td>153/6055</td>\n",
       "      <td>22/5808</td>\n",
       "      <td>8/1935</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Benign</td>\n",
       "      <td>calibrated</td>\n",
       "      <td>Variant CADD score of 10.42 is less than 15.62...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>848</th>\n",
       "      <td>14</td>\n",
       "      <td>23902920</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.8247270000000001</td>\n",
       "      <td>9.616</td>\n",
       "      <td>8.236E-6</td>\n",
       "      <td>...</td>\n",
       "      <td>c.22G&gt;A</td>\n",
       "      <td>p.Val8Ile</td>\n",
       "      <td>153/6055</td>\n",
       "      <td>22/5808</td>\n",
       "      <td>8/1935</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Benign</td>\n",
       "      <td>calibrated</td>\n",
       "      <td>Variant CADD score of 9.616 is less than 15.62...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>849</th>\n",
       "      <td>14</td>\n",
       "      <td>23902927</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.607012</td>\n",
       "      <td>8.205</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>c.15G&gt;C</td>\n",
       "      <td>p.Glu5Asp</td>\n",
       "      <td>146/6055</td>\n",
       "      <td>15/5808</td>\n",
       "      <td>5/1935</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Benign</td>\n",
       "      <td>calibrated</td>\n",
       "      <td>Variant CADD score of 8.205 is less than 15.62...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>850 rows × 31 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    CHROM       POS ID   REF ALT QUAL FILTER                 CADD CADD_SCALED  \\\n",
       "0      14  23882063  .     C   T    .      .             1.880579       15.47   \n",
       "1      14  23882064  .     T   C    .      .             1.823769       15.13   \n",
       "2      14  23882082  .     T   C    .      .   4.4282580000000005        24.2   \n",
       "3      14  23882975  .     C   T    .      .   3.5186269999999995        23.1   \n",
       "4      14  23882976  .     C   A    .      .             5.615639        26.6   \n",
       "5      14  23882984  .     C   T    .      .             6.322065        29.2   \n",
       "6      14  23882985  .     G   A    .      .             8.175613        35.0   \n",
       "7      14  23882993  .     G   A    .      .             3.586021        23.2   \n",
       "8      14  23882997  .     G   A    .      .    6.560517999999999        31.0   \n",
       "9      14  23883009  .     C   A    .      .             7.421627        34.0   \n",
       "10     14  23883014  .  GACT   A    .      .             3.300168        22.9   \n",
       "11     14  23883018  .     C   T    .      .             7.559069        34.0   \n",
       "12     14  23883020  .     G   A    .      .    6.566591000000001        31.0   \n",
       "13     14  23883021  .     C   T    .      .    7.710235000000001        35.0   \n",
       "14     14  23883032  .     C   G    .      .              7.93867        35.0   \n",
       "15     14  23883032  .     C   T    .      .             8.266119        35.0   \n",
       "16     14  23883033  .     G   A    .      .             7.408993        34.0   \n",
       "17     14  23883041  .     G   C    .      .             6.666103        32.0   \n",
       "18     14  23883054  .     C   T    .      .             8.245451        35.0   \n",
       "19     14  23883056  .     T   A    .      .             5.867249        27.3   \n",
       "20     14  23883059  .     T   A    .      .   6.9252720000000005        33.0   \n",
       "21     14  23883062  .     A   G    .      .             2.354094       18.52   \n",
       "22     14  23883068  .     C   T    .      .             7.939234        35.0   \n",
       "23     14  23883069  .     G   A    .      .             8.357856        35.0   \n",
       "24     14  23883071  .     A   C    .      .             5.127199        25.4   \n",
       "25     14  23883071  .     A   G    .      .             6.718649        32.0   \n",
       "26     14  23883092  .     G   A    .      .             6.177906        28.6   \n",
       "27     14  23883098  .     T   C    .      .             6.721912        32.0   \n",
       "28     14  23883101  .     T   C    .      .    6.701846000000001        32.0   \n",
       "29     14  23883216  .     C   T    .      .             1.823764       15.13   \n",
       "..    ...       ... ..   ...  ..  ...    ...                  ...         ...   \n",
       "820    14  23902005  .     C   T    .      .             3.871197        23.5   \n",
       "821    14  23902297  .     A   G    .      .   4.4865330000000005        24.3   \n",
       "822    14  23902298  .     T   A    .      .              5.20941        25.6   \n",
       "823    14  23902299  .     C   G    .      .             2.130051       17.05   \n",
       "824    14  23902310  .     C   A    .      .             4.304563        24.0   \n",
       "825    14  23902340  .     C   A    .      .   3.3837330000000003        23.0   \n",
       "826    14  23902364  .     T   C    .      .             2.694014        20.8   \n",
       "827    14  23902366  .     G   A    .      .             6.136413        28.4   \n",
       "828    14  23902426  .     A   G    .      .             2.922024        21.9   \n",
       "829    14  23902430  .     T   A    .      .   1.0409959999999998       10.89   \n",
       "830    14  23902758  .     C   T    .      .   1.7601419999999999       14.75   \n",
       "831    14  23902767  .     C   A    .      .             5.563407        26.4   \n",
       "832    14  23902767  .     C   T    .      .             2.010589       16.28   \n",
       "833    14  23902776  .     C   T    .      .             4.707939        24.6   \n",
       "834    14  23902781  .     C   A    .      .             4.868483        24.9   \n",
       "835    14  23902782  .     G   C    .      .             3.990305        23.6   \n",
       "836    14  23902788  .     C   T    .      .  -1.5471270000000001       0.002   \n",
       "837    14  23902804  .     A   T    .      .             2.169108       17.31   \n",
       "838    14  23902818  .     C   T    .      .             3.204917        22.7   \n",
       "839    14  23902821  .     C   T    .      .             6.421265        29.7   \n",
       "840    14  23902851  .     A   G    .      .             6.160776        28.5   \n",
       "841    14  23902866  .     C   T    .      .             4.774503        24.7   \n",
       "842    14  23902888  .     C   G    .      .             5.255416        25.7   \n",
       "843    14  23902892  .     C   T    .      .             7.612844        34.0   \n",
       "844    14  23902901  .     G   A    .      .             4.636164        24.5   \n",
       "845    14  23902905  .     C   T    .      .             7.042627        33.0   \n",
       "846    14  23902913  .     C   G    .      .             5.493546        26.2   \n",
       "847    14  23902920  .     C   G    .      .   0.9583229999999999       10.42   \n",
       "848    14  23902920  .     C   T    .      .   0.8247270000000001       9.616   \n",
       "849    14  23902927  .     C   G    .      .             0.607012       8.205   \n",
       "\n",
       "      EXAC_AF                        ...                          \\\n",
       "0         NaN                        ...                           \n",
       "1         NaN                        ...                           \n",
       "2         NaN                        ...                           \n",
       "3    8.237E-6                        ...                           \n",
       "4         NaN                        ...                           \n",
       "5    8.237E-6                        ...                           \n",
       "6    8.237E-6                        ...                           \n",
       "7    8.236E-6                        ...                           \n",
       "8    8.237E-6                        ...                           \n",
       "9         NaN                        ...                           \n",
       "10        NaN                        ...                           \n",
       "11        NaN                        ...                           \n",
       "12        NaN                        ...                           \n",
       "13   8.236E-6                        ...                           \n",
       "14        NaN                        ...                           \n",
       "15   8.236E-6                        ...                           \n",
       "16   3.295E-5                        ...                           \n",
       "17        NaN                        ...                           \n",
       "18   1.647E-5                        ...                           \n",
       "19        NaN                        ...                           \n",
       "20   8.236E-6                        ...                           \n",
       "21   8.236E-6                        ...                           \n",
       "22        NaN                        ...                           \n",
       "23   1.647E-5                        ...                           \n",
       "24        NaN                        ...                           \n",
       "25   8.236E-6                        ...                           \n",
       "26        NaN                        ...                           \n",
       "27   8.236E-6                        ...                           \n",
       "28        NaN                        ...                           \n",
       "29        NaN                        ...                           \n",
       "..        ...                        ...                           \n",
       "820       NaN                        ...                           \n",
       "821  8.236E-6                        ...                           \n",
       "822  8.236E-6                        ...                           \n",
       "823       NaN                        ...                           \n",
       "824       NaN                        ...                           \n",
       "825  8.236E-6                        ...                           \n",
       "826       NaN                        ...                           \n",
       "827       NaN                        ...                           \n",
       "828       NaN                        ...                           \n",
       "829  8.236E-6                        ...                           \n",
       "830  8.236E-6                        ...                           \n",
       "831  8.236E-6                        ...                           \n",
       "832  8.236E-6                        ...                           \n",
       "833  8.236E-6                        ...                           \n",
       "834  8.236E-6                        ...                           \n",
       "835  8.236E-6                        ...                           \n",
       "836       NaN                        ...                           \n",
       "837  8.236E-6                        ...                           \n",
       "838  8.236E-6                        ...                           \n",
       "839       NaN                        ...                           \n",
       "840  8.236E-6                        ...                           \n",
       "841  8.236E-6                        ...                           \n",
       "842  8.237E-6                        ...                           \n",
       "843       NaN                        ...                           \n",
       "844  8.236E-6                        ...                           \n",
       "845  8.236E-6                        ...                           \n",
       "846  2.471E-5                        ...                           \n",
       "847  8.236E-6                        ...                           \n",
       "848  8.236E-6                        ...                           \n",
       "849       NaN                        ...                           \n",
       "\n",
       "                     HGVS_c                      HGVS_p cDNA_position  \\\n",
       "0                 c.5808G>A                p.Ter1936Ter     5939/6055   \n",
       "1                 c.5807A>G           p.Ter1936Trpext*?     5938/6055   \n",
       "2               c.5791-2A>G                         NaN           NaN   \n",
       "3                 c.5783G>A                p.Gly1928Asp     5914/6055   \n",
       "4                 c.5782G>T                p.Gly1928Cys     5913/6055   \n",
       "5                 c.5774G>A                p.Arg1925His     5905/6055   \n",
       "6                 c.5773C>T                p.Arg1925Cys     5904/6055   \n",
       "7                 c.5765C>T                p.Ala1922Val     5896/6055   \n",
       "8                 c.5761C>T                p.Arg1921Trp     5892/6055   \n",
       "9                 c.5749G>T                p.Val1917Phe     5880/6055   \n",
       "10   c.5741_5744delAGTCinsT  p.Glu1914_Ser1915delinsVal     5875/6055   \n",
       "11                c.5740G>A                p.Glu1914Lys     5871/6055   \n",
       "12                c.5738C>T                p.Ala1913Val     5869/6055   \n",
       "13                c.5737G>A                p.Ala1913Thr     5868/6055   \n",
       "14                c.5726G>C                p.Arg1909Pro     5857/6055   \n",
       "15                c.5726G>A                p.Arg1909Gln     5857/6055   \n",
       "16                c.5725C>T                p.Arg1909Trp     5856/6055   \n",
       "17                c.5717C>G                p.Ala1906Gly     5848/6055   \n",
       "18                c.5704G>A                p.Glu1902Lys     5835/6055   \n",
       "19                c.5702A>T                p.His1901Leu     5833/6055   \n",
       "20                c.5699A>T                p.Gln1900Leu     5830/6055   \n",
       "21                c.5696T>C                p.Val1899Ala     5827/6055   \n",
       "22                c.5690G>A                p.Arg1897His     5821/6055   \n",
       "23                c.5689C>T                p.Arg1897Cys     5820/6055   \n",
       "24                c.5687T>G                p.Phe1896Cys     5818/6055   \n",
       "25                c.5687T>C                p.Phe1896Ser     5818/6055   \n",
       "26                c.5666C>T                p.Ala1889Val     5797/6055   \n",
       "27                c.5660A>G                p.Glu1887Gly     5791/6055   \n",
       "28                c.5657A>G                p.Glu1886Gly     5788/6055   \n",
       "29                c.5655G>A                p.Ala1885Ala     5786/6055   \n",
       "..                      ...                         ...           ...   \n",
       "820              c.346-1G>A                         NaN           NaN   \n",
       "821                c.341T>C                 p.Ile114Thr      472/6055   \n",
       "822                c.340A>T                 p.Ile114Phe      471/6055   \n",
       "823                c.339G>C                 p.Met113Ile      470/6055   \n",
       "824                c.328G>T                 p.Gly110Cys      459/6055   \n",
       "825                c.298G>T                 p.Ala100Ser      429/6055   \n",
       "826                c.274A>G                  p.Met92Val      405/6055   \n",
       "827                c.272C>T                  p.Ala91Val      403/6055   \n",
       "828                c.212T>C                  p.Val71Ala      343/6055   \n",
       "829                c.208A>T                  p.Thr70Ser      339/6055   \n",
       "830                c.184G>A                  p.Glu62Lys      315/6055   \n",
       "831                c.175G>T                  p.Val59Phe      306/6055   \n",
       "832                c.175G>A                  p.Val59Ile      306/6055   \n",
       "833                c.166G>A                  p.Gly56Ser      297/6055   \n",
       "834                c.161G>T                  p.Arg54Leu      292/6055   \n",
       "835                c.160C>G                  p.Arg54Gly      291/6055   \n",
       "836                c.154G>A                  p.Val52Met      285/6055   \n",
       "837                c.138T>A                  p.Phe46Leu      269/6055   \n",
       "838                c.124G>A                  p.Asp42Asn      255/6055   \n",
       "839                c.121G>A                  p.Asp41Asn      252/6055   \n",
       "840                 c.91T>C                  p.Phe31Leu      222/6055   \n",
       "841                 c.76G>A                  p.Ala26Thr      207/6055   \n",
       "842                 c.54G>C                  p.Lys18Asn      185/6055   \n",
       "843                 c.50G>A                  p.Arg17His      181/6055   \n",
       "844                 c.41C>T                  p.Pro14Leu      172/6055   \n",
       "845                 c.37G>A                  p.Ala13Thr      168/6055   \n",
       "846                 c.29G>C                  p.Gly10Ala      160/6055   \n",
       "847                 c.22G>C                   p.Val8Leu      153/6055   \n",
       "848                 c.22G>A                   p.Val8Ile      153/6055   \n",
       "849                 c.15G>C                   p.Glu5Asp      146/6055   \n",
       "\n",
       "    CDS_position Protein_position Distance_to_feature Errors Classification  \\\n",
       "0      5808/5808        1936/1935                 NaN    NaN         Benign   \n",
       "1      5807/5808        1936/1935                 NaN    NaN         Benign   \n",
       "2            NaN              NaN                 NaN    NaN     Pathogenic   \n",
       "3      5783/5808        1928/1935                 NaN    NaN     Pathogenic   \n",
       "4      5782/5808        1928/1935                 NaN    NaN     Pathogenic   \n",
       "5      5774/5808        1925/1935                 NaN    NaN     Pathogenic   \n",
       "6      5773/5808        1925/1935                 NaN    NaN     Pathogenic   \n",
       "7      5765/5808        1922/1935                 NaN    NaN     Pathogenic   \n",
       "8      5761/5808        1921/1935                 NaN    NaN     Pathogenic   \n",
       "9      5749/5808        1917/1935                 NaN    NaN     Pathogenic   \n",
       "10     5741/5808        1914/1935                 NaN    NaN     Pathogenic   \n",
       "11     5740/5808        1914/1935                 NaN    NaN     Pathogenic   \n",
       "12     5738/5808        1913/1935                 NaN    NaN     Pathogenic   \n",
       "13     5737/5808        1913/1935                 NaN    NaN     Pathogenic   \n",
       "14     5726/5808        1909/1935                 NaN    NaN     Pathogenic   \n",
       "15     5726/5808        1909/1935                 NaN    NaN     Pathogenic   \n",
       "16     5725/5808        1909/1935                 NaN    NaN     Pathogenic   \n",
       "17     5717/5808        1906/1935                 NaN    NaN     Pathogenic   \n",
       "18     5704/5808        1902/1935                 NaN    NaN     Pathogenic   \n",
       "19     5702/5808        1901/1935                 NaN    NaN     Pathogenic   \n",
       "20     5699/5808        1900/1935                 NaN    NaN     Pathogenic   \n",
       "21     5696/5808        1899/1935                 NaN    NaN     Pathogenic   \n",
       "22     5690/5808        1897/1935                 NaN    NaN     Pathogenic   \n",
       "23     5689/5808        1897/1935                 NaN    NaN     Pathogenic   \n",
       "24     5687/5808        1896/1935                 NaN    NaN     Pathogenic   \n",
       "25     5687/5808        1896/1935                 NaN    NaN     Pathogenic   \n",
       "26     5666/5808        1889/1935                 NaN    NaN     Pathogenic   \n",
       "27     5660/5808        1887/1935                 NaN    NaN     Pathogenic   \n",
       "28     5657/5808        1886/1935                 NaN    NaN     Pathogenic   \n",
       "29     5655/5808        1885/1935                 NaN    NaN         Benign   \n",
       "..           ...              ...                 ...    ...            ...   \n",
       "820          NaN              NaN                 NaN    NaN     Pathogenic   \n",
       "821     341/5808         114/1935                 NaN    NaN     Pathogenic   \n",
       "822     340/5808         114/1935                 NaN    NaN     Pathogenic   \n",
       "823     339/5808         113/1935                 NaN    NaN     Pathogenic   \n",
       "824     328/5808         110/1935                 NaN    NaN     Pathogenic   \n",
       "825     298/5808         100/1935                 NaN    NaN     Pathogenic   \n",
       "826     274/5808          92/1935                 NaN    NaN     Pathogenic   \n",
       "827     272/5808          91/1935                 NaN    NaN     Pathogenic   \n",
       "828     212/5808          71/1935                 NaN    NaN     Pathogenic   \n",
       "829     208/5808          70/1935                 NaN    NaN         Benign   \n",
       "830     184/5808          62/1935                 NaN    NaN         Benign   \n",
       "831     175/5808          59/1935                 NaN    NaN     Pathogenic   \n",
       "832     175/5808          59/1935                 NaN    NaN     Pathogenic   \n",
       "833     166/5808          56/1935                 NaN    NaN     Pathogenic   \n",
       "834     161/5808          54/1935                 NaN    NaN     Pathogenic   \n",
       "835     160/5808          54/1935                 NaN    NaN     Pathogenic   \n",
       "836     154/5808          52/1935                 NaN    NaN         Benign   \n",
       "837     138/5808          46/1935                 NaN    NaN     Pathogenic   \n",
       "838     124/5808          42/1935                 NaN    NaN     Pathogenic   \n",
       "839     121/5808          41/1935                 NaN    NaN     Pathogenic   \n",
       "840      91/5808          31/1935                 NaN    NaN     Pathogenic   \n",
       "841      76/5808          26/1935                 NaN    NaN     Pathogenic   \n",
       "842      54/5808          18/1935                 NaN    NaN     Pathogenic   \n",
       "843      50/5808          17/1935                 NaN    NaN     Pathogenic   \n",
       "844      41/5808          14/1935                 NaN    NaN     Pathogenic   \n",
       "845      37/5808          13/1935                 NaN    NaN     Pathogenic   \n",
       "846      29/5808          10/1935                 NaN    NaN     Pathogenic   \n",
       "847      22/5808           8/1935                 NaN    NaN         Benign   \n",
       "848      22/5808           8/1935                 NaN    NaN         Benign   \n",
       "849      15/5808           5/1935                 NaN    NaN         Benign   \n",
       "\n",
       "     Confidence                                             Reason  \n",
       "0    calibrated  Variant CADD score of 15.47 is less than 15.62...  \n",
       "1    calibrated  Variant CADD score of 15.13 is less than 15.62...  \n",
       "2    calibrated  Variant CADD score of 24.2 is greater than 22....  \n",
       "3    calibrated  Variant CADD score of 23.1 is greater than 22....  \n",
       "4    calibrated  Variant CADD score of 26.6 is greater than 22....  \n",
       "5    calibrated  Variant CADD score of 29.2 is greater than 22....  \n",
       "6    calibrated  Variant CADD score of 35.0 is greater than 22....  \n",
       "7    calibrated  Variant CADD score of 23.2 is greater than 22....  \n",
       "8    calibrated  Variant CADD score of 31.0 is greater than 22....  \n",
       "9    calibrated  Variant CADD score of 34.0 is greater than 22....  \n",
       "10   calibrated  Variant CADD score of 22.9 is greater than 22....  \n",
       "11   calibrated  Variant CADD score of 34.0 is greater than 22....  \n",
       "12   calibrated  Variant CADD score of 31.0 is greater than 22....  \n",
       "13   calibrated  Variant CADD score of 35.0 is greater than 22....  \n",
       "14   calibrated  Variant CADD score of 35.0 is greater than 22....  \n",
       "15   calibrated  Variant CADD score of 35.0 is greater than 22....  \n",
       "16   calibrated  Variant CADD score of 34.0 is greater than 22....  \n",
       "17   calibrated  Variant CADD score of 32.0 is greater than 22....  \n",
       "18   calibrated  Variant CADD score of 35.0 is greater than 22....  \n",
       "19   calibrated  Variant CADD score of 27.3 is greater than 22....  \n",
       "20   calibrated  Variant CADD score of 33.0 is greater than 22....  \n",
       "21   genomewide  Variant MAF of 8.236E-6 is rare enough to be p...  \n",
       "22   calibrated  Variant CADD score of 35.0 is greater than 22....  \n",
       "23   calibrated  Variant CADD score of 35.0 is greater than 22....  \n",
       "24   calibrated  Variant CADD score of 25.4 is greater than 22....  \n",
       "25   calibrated  Variant CADD score of 32.0 is greater than 22....  \n",
       "26   calibrated  Variant CADD score of 28.6 is greater than 22....  \n",
       "27   calibrated  Variant CADD score of 32.0 is greater than 22....  \n",
       "28   calibrated  Variant CADD score of 32.0 is greater than 22....  \n",
       "29   calibrated  Variant CADD score of 15.13 is less than 15.62...  \n",
       "..          ...                                                ...  \n",
       "820  calibrated  Variant CADD score of 23.5 is greater than 22....  \n",
       "821  calibrated  Variant CADD score of 24.3 is greater than 22....  \n",
       "822  calibrated  Variant CADD score of 25.6 is greater than 22....  \n",
       "823  genomewide  Variant MAF of 0.0 is rare enough to be potent...  \n",
       "824  calibrated  Variant CADD score of 24.0 is greater than 22....  \n",
       "825  calibrated  Variant CADD score of 23.0 is greater than 22....  \n",
       "826  genomewide  Variant MAF of 0.0 is rare enough to be potent...  \n",
       "827  calibrated  Variant CADD score of 28.4 is greater than 22....  \n",
       "828  genomewide  Variant MAF of 0.0 is rare enough to be potent...  \n",
       "829  calibrated  Variant CADD score of 10.89 is less than 15.62...  \n",
       "830  calibrated  Variant CADD score of 14.75 is less than 15.62...  \n",
       "831  calibrated  Variant CADD score of 26.4 is greater than 22....  \n",
       "832  genomewide  Variant MAF of 8.236E-6 is rare enough to be p...  \n",
       "833  calibrated  Variant CADD score of 24.6 is greater than 22....  \n",
       "834  calibrated  Variant CADD score of 24.9 is greater than 22....  \n",
       "835  calibrated  Variant CADD score of 23.6 is greater than 22....  \n",
       "836  calibrated  Variant CADD score of 0.002 is less than 15.62...  \n",
       "837  genomewide  Variant MAF of 8.236E-6 is rare enough to be p...  \n",
       "838  genomewide  Variant MAF of 8.236E-6 is rare enough to be p...  \n",
       "839  calibrated  Variant CADD score of 29.7 is greater than 22....  \n",
       "840  calibrated  Variant CADD score of 28.5 is greater than 22....  \n",
       "841  calibrated  Variant CADD score of 24.7 is greater than 22....  \n",
       "842  calibrated  Variant CADD score of 25.7 is greater than 22....  \n",
       "843  calibrated  Variant CADD score of 34.0 is greater than 22....  \n",
       "844  calibrated  Variant CADD score of 24.5 is greater than 22....  \n",
       "845  calibrated  Variant CADD score of 33.0 is greater than 22....  \n",
       "846  calibrated  Variant CADD score of 26.2 is greater than 22....  \n",
       "847  calibrated  Variant CADD score of 10.42 is less than 15.62...  \n",
       "848  calibrated  Variant CADD score of 9.616 is less than 15.62...  \n",
       "849  calibrated  Variant CADD score of 8.205 is less than 15.62...  \n",
       "\n",
       "[850 rows x 31 columns]"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "datatable.replace('',np.NaN)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "datatable['CADD'] = datatable['CADD'].astype('float')\n",
    "datatable['CADD_SCALED'] = datatable['CADD_SCALED'].astype('float')\n",
    "# datatable['EXAC_AF'] = datatable['EXAC_AF'].astype('float')\n",
    "# datatable['EXAC_AC_HOM'] = datatable['EXAC_AC_HOM'].astype('float')\n",
    "# datatable['Distance_to_feature'] = datatable['Distance_to_feature'].astype('int')\n",
    "# datatable['Errors'] = datatable['Errors'].astype('int')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "datatable.to_csv('data/gavin_res.csv',sep='\\t', index=False, header=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0      1.880579\n",
       "1      1.823769\n",
       "2      4.428258\n",
       "3      3.518627\n",
       "4      5.615639\n",
       "5      6.322065\n",
       "6      8.175613\n",
       "7      3.586021\n",
       "8      6.560518\n",
       "9      7.421627\n",
       "10     3.300168\n",
       "11     7.559069\n",
       "12     6.566591\n",
       "13     7.710235\n",
       "14     7.938670\n",
       "15     8.266119\n",
       "16     7.408993\n",
       "17     6.666103\n",
       "18     8.245451\n",
       "19     5.867249\n",
       "20     6.925272\n",
       "21     2.354094\n",
       "22     7.939234\n",
       "23     8.357856\n",
       "24     5.127199\n",
       "25     6.718649\n",
       "26     6.177906\n",
       "27     6.721912\n",
       "28     6.701846\n",
       "29     1.823764\n",
       "         ...   \n",
       "820    3.871197\n",
       "821    4.486533\n",
       "822    5.209410\n",
       "823    2.130051\n",
       "824    4.304563\n",
       "825    3.383733\n",
       "826    2.694014\n",
       "827    6.136413\n",
       "828    2.922024\n",
       "829    1.040996\n",
       "830    1.760142\n",
       "831    5.563407\n",
       "832    2.010589\n",
       "833    4.707939\n",
       "834    4.868483\n",
       "835    3.990305\n",
       "836   -1.547127\n",
       "837    2.169108\n",
       "838    3.204917\n",
       "839    6.421265\n",
       "840    6.160776\n",
       "841    4.774503\n",
       "842    5.255416\n",
       "843    7.612844\n",
       "844    4.636164\n",
       "845    7.042627\n",
       "846    5.493546\n",
       "847    0.958323\n",
       "848    0.824727\n",
       "849    0.607012\n",
       "Name: CADD, Length: 850, dtype: float64"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "datatable.CADD"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "py3",
   "language": "python",
   "name": "py3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.4"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
