{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "import numpy as np\n",
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "# read vcf files\n",
    "vcf_reader = open('/Users/gcc/Downloads/cadd_copy-gavin.vcf','r')\n",
    "column_lines = []\n",
    "info = []\n",
    "sample_lines = []\n",
    "for line in vcf_reader.readlines():\n",
    "    if line.startswith('#CHROM'):\n",
    "        column_lines = line.strip('\\n').strip('#').split('\\t')\n",
    "    elif line.startswith('##INFO'):\n",
    "        info.append(line.strip('\\n').strip('#'))\n",
    "    elif line[0].isdigit():\n",
    "        line = line.strip('\\n')\n",
    "        sample_lines.append(line)\n",
    "        \n",
    "# parse the column names\n",
    "column_lines = column_lines[:-1]\n",
    "info_contents = [line[line.find('INFO=ID=')+len('INFO=ID=')+2:line.find(',')] for line in info[:-1]]\n",
    "line = info[-1]\n",
    "info_last = line[line.find(\"annotations: '\")+len(\"annotations: '\")+2:-3].split('|')\n",
    "columns = column_lines+info_contents+info_last\n",
    "columns = [col.strip(' ') for col in columns]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['CHROM',\n",
       " 'POS',\n",
       " 'ID',\n",
       " 'REF',\n",
       " 'ALT',\n",
       " 'QUAL',\n",
       " 'FILTER',\n",
       " 'CADD',\n",
       " 'CADD_SCALED',\n",
       " 'EXAC_AF',\n",
       " 'EXAC_AC_HOM',\n",
       " 'EXAC_AC_HET',\n",
       " 't_Allele',\n",
       " 'Gene_Name',\n",
       " 'Annotation',\n",
       " 'Putative_impact',\n",
       " 'Gene_ID',\n",
       " 'Feature_type',\n",
       " 'Feature_ID',\n",
       " 'Transcript_biotype',\n",
       " 'Rank_total',\n",
       " 'HGVS_c',\n",
       " 'HGVS_p',\n",
       " 'cDNA_position',\n",
       " 'CDS_position',\n",
       " 'Protein_position',\n",
       " 'Distance_to_feature',\n",
       " 'Errors',\n",
       " 'Classification',\n",
       " 'Confidence',\n",
       " 'Reason']"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def cut_str(string,begin,end):\n",
    "    '''\n",
    "    cheery pick the begin till end of a string\n",
    "    '''\n",
    "    if begin not in string:\n",
    "        return ''\n",
    "    begin_ind = string.find(begin)+len(begin)\n",
    "    cut = string[begin_ind:]\n",
    "    end_ind = cut.find(end)+len(end)-1\n",
    "    return cut[:end_ind]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 92,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "# samples = []\n",
    "# for line in sample_lines:\n",
    "#     var_info = line.split('\\t')[:-1]\n",
    "#     if ',' in line:\n",
    "#         var = line.split('\\t')[-1].split(',')\n",
    "#     else:\n",
    "#         var=line.split('\\t')[-1]\n",
    "#     for gene in var:\n",
    "#         if 'MYO5B' in gene:\n",
    "#             var_gene = []\n",
    "#             var_gene_infos = gene.split(';')[-1].split('|')\n",
    "#             var_gene.append(cut_str(gene,'CADD=',';'))\n",
    "#             var_gene.append(cut_str(gene,'CADD_SCALED=',';'))\n",
    "#             var_gene.append(cut_str(gene,'EXAC_AF=',';'))\n",
    "#             var_gene.append(cut_str(gene,'EXAC_AC_HOM=',';'))\n",
    "#             var_gene.append(cut_str(gene,'EXAC_AC_HET=',';'))\n",
    "#     samples.append(var_info+var_gene+var_gene_infos)\n",
    "# for sample in samples:\n",
    "#     assert len(sample)==31"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "# parse the samples\n",
    "samples = []\n",
    "for line in sample_lines:\n",
    "    line = line.strip(' ')\n",
    "    var_info = line.split('\\t')[:-1]\n",
    "    if ',' in line:\n",
    "        genes = line.split('\\t')[-1].split(',')\n",
    "        for gene in genes:\n",
    "            if 'MYO5B' in gene:\n",
    "                myo_info = gene\n",
    "                var_gene_infos = myo_info.split(';')[-1].split('|')\n",
    "    else:\n",
    "        myo_info=line.split('\\t')[-1]\n",
    "        var_gene_infos = myo_info.split(';')[-2].split('|')\n",
    "    var_gene = []    \n",
    "    var_gene.append(cut_str(myo_info,'CADD=',';'))\n",
    "    var_gene.append(cut_str(myo_info,'CADD_SCALED=',';'))\n",
    "    var_gene.append(cut_str(myo_info,'EXAC_AF=',';'))\n",
    "    var_gene.append(cut_str(myo_info,'EXAC_AC_HOM=',';'))\n",
    "    var_gene.append(cut_str(myo_info,'EXAC_AC_HET=',';'))\n",
    "    samples.append(var_info+var_gene+var_gene_infos)\n",
    "for sample in samples:\n",
    "    assert len(sample)==31"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "datatable = pd.DataFrame(samples,columns=columns)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>CADD</th>\n",
       "      <th>CADD_SCALED</th>\n",
       "      <th>EXAC_AF</th>\n",
       "      <th>...</th>\n",
       "      <th>HGVS_c</th>\n",
       "      <th>HGVS_p</th>\n",
       "      <th>cDNA_position</th>\n",
       "      <th>CDS_position</th>\n",
       "      <th>Protein_position</th>\n",
       "      <th>Distance_to_feature</th>\n",
       "      <th>Errors</th>\n",
       "      <th>Classification</th>\n",
       "      <th>Confidence</th>\n",
       "      <th>Reason</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>18</td>\n",
       "      <td>47352774</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.334537</td>\n",
       "      <td>6.023</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>c.*67A&gt;G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>67</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Benign</td>\n",
       "      <td>calibrated</td>\n",
       "      <td>Variant CADD score of 6.023 is less than 8.87 ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>18</td>\n",
       "      <td>47361716</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>14.849415</td>\n",
       "      <td>48.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>c.5392C&gt;T</td>\n",
       "      <td>p.Gln1798*</td>\n",
       "      <td>5680/9520</td>\n",
       "      <td>5392/5547</td>\n",
       "      <td>1798/1848</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>calibrated</td>\n",
       "      <td>Variant CADD score of 48.0 is greater than 34....</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>18</td>\n",
       "      <td>47361725</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>15.701726</td>\n",
       "      <td>52.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>c.5383C&gt;T</td>\n",
       "      <td>p.Arg1795*</td>\n",
       "      <td>5671/9520</td>\n",
       "      <td>5383/5547</td>\n",
       "      <td>1795/1848</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>calibrated</td>\n",
       "      <td>Variant CADD score of 52.0 is greater than 34....</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>18</td>\n",
       "      <td>47365526</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>14.577744</td>\n",
       "      <td>46.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>c.4840C&gt;T</td>\n",
       "      <td>p.Gln1614*</td>\n",
       "      <td>5128/9520</td>\n",
       "      <td>4840/5547</td>\n",
       "      <td>1614/1848</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>calibrated</td>\n",
       "      <td>Variant CADD score of 46.0 is greater than 34....</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>18</td>\n",
       "      <td>47365610</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>CCA</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>9.553096</td>\n",
       "      <td>35.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>c.4755_4756insTG</td>\n",
       "      <td>p.Asp1586fs</td>\n",
       "      <td>5043/9520</td>\n",
       "      <td>4755/5547</td>\n",
       "      <td>1585/1848</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>calibrated</td>\n",
       "      <td>Variant CADD score of 35.0 is greater than 34....</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>18</td>\n",
       "      <td>47367768</td>\n",
       "      <td>.</td>\n",
       "      <td>AA</td>\n",
       "      <td>GC</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>6.332952</td>\n",
       "      <td>29.3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>c.4667_4668delTTinsGC</td>\n",
       "      <td>p.Leu1556Arg</td>\n",
       "      <td>4956/9520</td>\n",
       "      <td>4667/5547</td>\n",
       "      <td>1556/1848</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>genomewide</td>\n",
       "      <td>Variant MAF of 0.0 is rare enough to be potent...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>18</td>\n",
       "      <td>47369763</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4.451087</td>\n",
       "      <td>24.2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>c.4460-1G&gt;C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>genomewide</td>\n",
       "      <td>Variant MAF of 0.0 is rare enough to be potent...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>18</td>\n",
       "      <td>47373609</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>12.325906</td>\n",
       "      <td>39.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>c.4366C&gt;T</td>\n",
       "      <td>p.Gln1456*</td>\n",
       "      <td>4654/9520</td>\n",
       "      <td>4366/5547</td>\n",
       "      <td>1456/1848</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>calibrated</td>\n",
       "      <td>Variant CADD score of 39.0 is greater than 34....</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>18</td>\n",
       "      <td>47380012</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>6.116766</td>\n",
       "      <td>28.3</td>\n",
       "      <td>8.264E-6</td>\n",
       "      <td>...</td>\n",
       "      <td>c.4028T&gt;C</td>\n",
       "      <td>p.Leu1343Pro</td>\n",
       "      <td>4316/9520</td>\n",
       "      <td>4028/5547</td>\n",
       "      <td>1343/1848</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>genomewide</td>\n",
       "      <td>Variant MAF of 8.264E-6 is rare enough to be p...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>18</td>\n",
       "      <td>47402080</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>12.92118</td>\n",
       "      <td>40.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>c.3514C&gt;T</td>\n",
       "      <td>p.Gln1172*</td>\n",
       "      <td>3802/9520</td>\n",
       "      <td>3514/5547</td>\n",
       "      <td>1172/1848</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>calibrated</td>\n",
       "      <td>Variant CADD score of 40.0 is greater than 34....</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>18</td>\n",
       "      <td>47405401</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>12.776541</td>\n",
       "      <td>40.0</td>\n",
       "      <td>1.655E-5</td>\n",
       "      <td>...</td>\n",
       "      <td>c.3190C&gt;T</td>\n",
       "      <td>p.Arg1064*</td>\n",
       "      <td>3478/9520</td>\n",
       "      <td>3190/5547</td>\n",
       "      <td>1064/1848</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>calibrated</td>\n",
       "      <td>Variant CADD score of 40.0 is greater than 34....</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>18</td>\n",
       "      <td>47405427</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>AGAG</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.524487</td>\n",
       "      <td>7.615</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>c.3163_3164insCTC</td>\n",
       "      <td>p.Asn1054_Leu1055insPro</td>\n",
       "      <td>3451/9520</td>\n",
       "      <td>3163/5547</td>\n",
       "      <td>1055/1848</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Benign</td>\n",
       "      <td>calibrated</td>\n",
       "      <td>Variant CADD score of 7.615 is less than 8.87 ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>18</td>\n",
       "      <td>47406825</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>10.692736</td>\n",
       "      <td>36.0</td>\n",
       "      <td>2.483E-5</td>\n",
       "      <td>...</td>\n",
       "      <td>c.3046C&gt;T</td>\n",
       "      <td>p.Arg1016*</td>\n",
       "      <td>3334/9520</td>\n",
       "      <td>3046/5547</td>\n",
       "      <td>1016/1848</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>calibrated</td>\n",
       "      <td>Variant CADD score of 36.0 is greater than 34....</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>18</td>\n",
       "      <td>47408592</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>CC</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.132595</td>\n",
       "      <td>3.958</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>c.3046-1768dupG</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Benign</td>\n",
       "      <td>calibrated</td>\n",
       "      <td>Variant CADD score of 3.958 is less than 8.87 ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>18</td>\n",
       "      <td>47429043</td>\n",
       "      <td>.</td>\n",
       "      <td>CGGG</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>2.817738</td>\n",
       "      <td>21.5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>c.2729_2731delCCC</td>\n",
       "      <td>p.Ala910_Arg911delinsGly</td>\n",
       "      <td>3019/9520</td>\n",
       "      <td>2729/5547</td>\n",
       "      <td>910/1848</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>genomewide</td>\n",
       "      <td>Variant MAF of 0.0 is rare enough to be potent...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>18</td>\n",
       "      <td>47429104</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>11.752624</td>\n",
       "      <td>38.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>c.2671C&gt;T</td>\n",
       "      <td>p.Gln891*</td>\n",
       "      <td>2959/9520</td>\n",
       "      <td>2671/5547</td>\n",
       "      <td>891/1848</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>calibrated</td>\n",
       "      <td>Variant CADD score of 38.0 is greater than 34....</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>18</td>\n",
       "      <td>47432872</td>\n",
       "      <td>.</td>\n",
       "      <td>TC</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>8.837797</td>\n",
       "      <td>35.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>c.2330delG</td>\n",
       "      <td>p.Gly777fs</td>\n",
       "      <td>2618/9520</td>\n",
       "      <td>2330/5547</td>\n",
       "      <td>777/1848</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>calibrated</td>\n",
       "      <td>Variant CADD score of 35.0 is greater than 34....</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>18</td>\n",
       "      <td>47435205</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>TC</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>-0.336167</td>\n",
       "      <td>0.549</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>c.2203-2206dupG</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Benign</td>\n",
       "      <td>calibrated</td>\n",
       "      <td>Variant CADD score of 0.549 is less than 8.87 ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>18</td>\n",
       "      <td>47462646</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>6.923516</td>\n",
       "      <td>33.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>c.1979C&gt;T</td>\n",
       "      <td>p.Pro660Leu</td>\n",
       "      <td>2267/9520</td>\n",
       "      <td>1979/5547</td>\n",
       "      <td>660/1848</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>genomewide</td>\n",
       "      <td>Variant MAF of 0.0 is rare enough to be potent...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>18</td>\n",
       "      <td>47462659</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>7.604902</td>\n",
       "      <td>34.0</td>\n",
       "      <td>8.264E-6</td>\n",
       "      <td>...</td>\n",
       "      <td>c.1966C&gt;T</td>\n",
       "      <td>p.Arg656Cys</td>\n",
       "      <td>2254/9520</td>\n",
       "      <td>1966/5547</td>\n",
       "      <td>656/1848</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>genomewide</td>\n",
       "      <td>Variant MAF of 8.264E-6 is rare enough to be p...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>18</td>\n",
       "      <td>47462720</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>5.482661</td>\n",
       "      <td>26.2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>c.1906-1G&gt;A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>genomewide</td>\n",
       "      <td>Variant MAF of 0.0 is rare enough to be potent...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>18</td>\n",
       "      <td>47463664</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>2.905477</td>\n",
       "      <td>21.9</td>\n",
       "      <td>8.271E-6</td>\n",
       "      <td>...</td>\n",
       "      <td>c.1856C&gt;T</td>\n",
       "      <td>p.Pro619Leu</td>\n",
       "      <td>2144/9520</td>\n",
       "      <td>1856/5547</td>\n",
       "      <td>619/1848</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>genomewide</td>\n",
       "      <td>Variant MAF of 8.271E-6 is rare enough to be p...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>18</td>\n",
       "      <td>47480703</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>3.829816</td>\n",
       "      <td>23.4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>c.1648A&gt;T</td>\n",
       "      <td>p.Ile550Phe</td>\n",
       "      <td>1936/9520</td>\n",
       "      <td>1648/5547</td>\n",
       "      <td>550/1848</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>genomewide</td>\n",
       "      <td>Variant MAF of 0.0 is rare enough to be potent...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>18</td>\n",
       "      <td>47480738</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>6.318276</td>\n",
       "      <td>29.2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>c.1613T&gt;C</td>\n",
       "      <td>p.Phe538Ser</td>\n",
       "      <td>1901/9520</td>\n",
       "      <td>1613/5547</td>\n",
       "      <td>538/1848</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>genomewide</td>\n",
       "      <td>Variant MAF of 0.0 is rare enough to be potent...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>18</td>\n",
       "      <td>47480760</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>5.852421</td>\n",
       "      <td>27.3</td>\n",
       "      <td>8.265E-6</td>\n",
       "      <td>...</td>\n",
       "      <td>c.1591C&gt;T</td>\n",
       "      <td>p.Arg531Trp</td>\n",
       "      <td>1879/9520</td>\n",
       "      <td>1591/5547</td>\n",
       "      <td>531/1848</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>genomewide</td>\n",
       "      <td>Variant MAF of 8.265E-6 is rare enough to be p...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>18</td>\n",
       "      <td>47480769</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>5.782332</td>\n",
       "      <td>27.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>c.1582C&gt;T</td>\n",
       "      <td>p.Leu528Phe</td>\n",
       "      <td>1870/9520</td>\n",
       "      <td>1582/5547</td>\n",
       "      <td>528/1848</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>genomewide</td>\n",
       "      <td>Variant MAF of 0.0 is rare enough to be potent...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>18</td>\n",
       "      <td>47488641</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>6.073709</td>\n",
       "      <td>28.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>c.1540T&gt;C</td>\n",
       "      <td>p.Cys514Arg</td>\n",
       "      <td>1828/9520</td>\n",
       "      <td>1540/5547</td>\n",
       "      <td>514/1848</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>genomewide</td>\n",
       "      <td>Variant MAF of 0.0 is rare enough to be potent...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>18</td>\n",
       "      <td>47489356</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>5.265616</td>\n",
       "      <td>25.7</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>c.1367A&gt;G</td>\n",
       "      <td>p.Asn456Ser</td>\n",
       "      <td>1655/9520</td>\n",
       "      <td>1367/5547</td>\n",
       "      <td>456/1848</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>genomewide</td>\n",
       "      <td>Variant MAF of 0.0 is rare enough to be potent...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>18</td>\n",
       "      <td>47489375</td>\n",
       "      <td>.</td>\n",
       "      <td>AG</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>7.952294</td>\n",
       "      <td>35.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>c.1347delC</td>\n",
       "      <td>p.Phe450fs</td>\n",
       "      <td>1635/9520</td>\n",
       "      <td>1347/5547</td>\n",
       "      <td>449/1848</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>calibrated</td>\n",
       "      <td>Variant CADD score of 35.0 is greater than 34....</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>18</td>\n",
       "      <td>47489402</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>3.928802</td>\n",
       "      <td>23.5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>c.1323-2A&gt;G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>genomewide</td>\n",
       "      <td>Variant MAF of 0.0 is rare enough to be potent...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>218</th>\n",
       "      <td>18</td>\n",
       "      <td>47518698</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>5.635733</td>\n",
       "      <td>26.6</td>\n",
       "      <td>8.271E-6</td>\n",
       "      <td>...</td>\n",
       "      <td>c.716T&gt;C</td>\n",
       "      <td>p.Met239Thr</td>\n",
       "      <td>1004/9520</td>\n",
       "      <td>716/5547</td>\n",
       "      <td>239/1848</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>genomewide</td>\n",
       "      <td>Variant MAF of 8.271E-6 is rare enough to be p...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>219</th>\n",
       "      <td>18</td>\n",
       "      <td>47518722</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>5.061619</td>\n",
       "      <td>25.2</td>\n",
       "      <td>2.481E-5</td>\n",
       "      <td>...</td>\n",
       "      <td>c.692G&gt;C</td>\n",
       "      <td>p.Arg231Thr</td>\n",
       "      <td>980/9520</td>\n",
       "      <td>692/5547</td>\n",
       "      <td>231/1848</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>genomewide</td>\n",
       "      <td>Variant MAF of 2.481E-5 is rare enough to be p...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>220</th>\n",
       "      <td>18</td>\n",
       "      <td>47518777</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>6.620644</td>\n",
       "      <td>32.0</td>\n",
       "      <td>1.655E-5</td>\n",
       "      <td>...</td>\n",
       "      <td>c.637C&gt;A</td>\n",
       "      <td>p.Arg213Ser</td>\n",
       "      <td>925/9520</td>\n",
       "      <td>637/5547</td>\n",
       "      <td>213/1848</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>genomewide</td>\n",
       "      <td>Variant MAF of 1.655E-5 is rare enough to be p...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>221</th>\n",
       "      <td>18</td>\n",
       "      <td>47527653</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>5.910215</td>\n",
       "      <td>27.5</td>\n",
       "      <td>8.271E-6</td>\n",
       "      <td>...</td>\n",
       "      <td>c.584A&gt;T</td>\n",
       "      <td>p.Lys195Met</td>\n",
       "      <td>872/9520</td>\n",
       "      <td>584/5547</td>\n",
       "      <td>195/1848</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>genomewide</td>\n",
       "      <td>Variant MAF of 8.271E-6 is rare enough to be p...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>222</th>\n",
       "      <td>18</td>\n",
       "      <td>47527686</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4.734543</td>\n",
       "      <td>24.6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>c.551G&gt;A</td>\n",
       "      <td>p.Gly184Asp</td>\n",
       "      <td>839/9520</td>\n",
       "      <td>551/5547</td>\n",
       "      <td>184/1848</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>genomewide</td>\n",
       "      <td>Variant MAF of 0.0 is rare enough to be potent...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>223</th>\n",
       "      <td>18</td>\n",
       "      <td>47405336</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>1.690176</td>\n",
       "      <td>14.35</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>c.3255G&gt;T</td>\n",
       "      <td>p.Arg1085Arg</td>\n",
       "      <td>3543/9520</td>\n",
       "      <td>3255/5547</td>\n",
       "      <td>1085/1848</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Benign</td>\n",
       "      <td>genomewide</td>\n",
       "      <td>Variant CADD score of 14.35 is less than a glo...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>224</th>\n",
       "      <td>18</td>\n",
       "      <td>47405414</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.071664</td>\n",
       "      <td>3.308</td>\n",
       "      <td>1.656E-5</td>\n",
       "      <td>...</td>\n",
       "      <td>c.3177A&gt;G</td>\n",
       "      <td>p.Glu1059Glu</td>\n",
       "      <td>3465/9520</td>\n",
       "      <td>3177/5547</td>\n",
       "      <td>1059/1848</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Benign</td>\n",
       "      <td>calibrated</td>\n",
       "      <td>Variant CADD score of 3.308 is less than 8.87 ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>225</th>\n",
       "      <td>18</td>\n",
       "      <td>47406817</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.069248</td>\n",
       "      <td>3.282</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>c.3054A&gt;G</td>\n",
       "      <td>p.Ala1018Ala</td>\n",
       "      <td>3342/9520</td>\n",
       "      <td>3054/5547</td>\n",
       "      <td>1018/1848</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Benign</td>\n",
       "      <td>calibrated</td>\n",
       "      <td>Variant CADD score of 3.282 is less than 8.87 ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>226</th>\n",
       "      <td>18</td>\n",
       "      <td>47421416</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>-0.1534</td>\n",
       "      <td>1.366</td>\n",
       "      <td>8.263E-6</td>\n",
       "      <td>...</td>\n",
       "      <td>c.2940G&gt;A</td>\n",
       "      <td>p.Arg980Arg</td>\n",
       "      <td>3228/9520</td>\n",
       "      <td>2940/5547</td>\n",
       "      <td>980/1848</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Benign</td>\n",
       "      <td>calibrated</td>\n",
       "      <td>Variant CADD score of 1.366 is less than 8.87 ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>227</th>\n",
       "      <td>18</td>\n",
       "      <td>47428960</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.624515</td>\n",
       "      <td>8.325</td>\n",
       "      <td>8.264E-6</td>\n",
       "      <td>...</td>\n",
       "      <td>c.2811+4A&gt;G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Benign</td>\n",
       "      <td>calibrated</td>\n",
       "      <td>Variant CADD score of 8.325 is less than 8.87 ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>228</th>\n",
       "      <td>18</td>\n",
       "      <td>47429084</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.565037</td>\n",
       "      <td>7.911</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>c.2691C&gt;G</td>\n",
       "      <td>p.Leu897Leu</td>\n",
       "      <td>2979/9520</td>\n",
       "      <td>2691/5547</td>\n",
       "      <td>897/1848</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Benign</td>\n",
       "      <td>calibrated</td>\n",
       "      <td>Variant CADD score of 7.911 is less than 8.87 ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>229</th>\n",
       "      <td>18</td>\n",
       "      <td>47429209</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>-0.190754</td>\n",
       "      <td>1.147</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>c.2572-6C&gt;T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Benign</td>\n",
       "      <td>calibrated</td>\n",
       "      <td>Variant CADD score of 1.147 is less than 8.87 ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>230</th>\n",
       "      <td>18</td>\n",
       "      <td>47431078</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>1.248168</td>\n",
       "      <td>12.0</td>\n",
       "      <td>8.272E-6</td>\n",
       "      <td>...</td>\n",
       "      <td>c.2535C&gt;T</td>\n",
       "      <td>p.Phe845Phe</td>\n",
       "      <td>2823/9520</td>\n",
       "      <td>2535/5547</td>\n",
       "      <td>845/1848</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Benign</td>\n",
       "      <td>genomewide</td>\n",
       "      <td>Variant CADD score of 12.0 is less than a glob...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>231</th>\n",
       "      <td>18</td>\n",
       "      <td>47431179</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>-0.029409</td>\n",
       "      <td>2.313</td>\n",
       "      <td>8.27E-6</td>\n",
       "      <td>...</td>\n",
       "      <td>c.2434A&gt;C</td>\n",
       "      <td>p.Arg812Arg</td>\n",
       "      <td>2722/9520</td>\n",
       "      <td>2434/5547</td>\n",
       "      <td>812/1848</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Benign</td>\n",
       "      <td>calibrated</td>\n",
       "      <td>Variant CADD score of 2.313 is less than 8.87 ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>232</th>\n",
       "      <td>18</td>\n",
       "      <td>47432836</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.758393</td>\n",
       "      <td>9.204</td>\n",
       "      <td>8.269E-6</td>\n",
       "      <td>...</td>\n",
       "      <td>c.2367G&gt;T</td>\n",
       "      <td>p.Gly789Gly</td>\n",
       "      <td>2655/9520</td>\n",
       "      <td>2367/5547</td>\n",
       "      <td>789/1848</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Benign</td>\n",
       "      <td>genomewide</td>\n",
       "      <td>Variant CADD score of 9.204 is less than a glo...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>233</th>\n",
       "      <td>18</td>\n",
       "      <td>47432941</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.089257</td>\n",
       "      <td>3.493</td>\n",
       "      <td>8.275E-6</td>\n",
       "      <td>...</td>\n",
       "      <td>c.2262C&gt;G</td>\n",
       "      <td>p.Ala754Ala</td>\n",
       "      <td>2550/9520</td>\n",
       "      <td>2262/5547</td>\n",
       "      <td>754/1848</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Benign</td>\n",
       "      <td>calibrated</td>\n",
       "      <td>Variant CADD score of 3.493 is less than 8.87 ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>234</th>\n",
       "      <td>18</td>\n",
       "      <td>47438492</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>-0.199414</td>\n",
       "      <td>1.101</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>c.2142A&gt;G</td>\n",
       "      <td>p.Arg714Arg</td>\n",
       "      <td>2430/9520</td>\n",
       "      <td>2142/5547</td>\n",
       "      <td>714/1848</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Benign</td>\n",
       "      <td>calibrated</td>\n",
       "      <td>Variant CADD score of 1.101 is less than 8.87 ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>235</th>\n",
       "      <td>18</td>\n",
       "      <td>47455877</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.421282</td>\n",
       "      <td>6.797</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>c.2090+5C&gt;A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Benign</td>\n",
       "      <td>calibrated</td>\n",
       "      <td>Variant CADD score of 6.797 is less than 8.87 ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>236</th>\n",
       "      <td>18</td>\n",
       "      <td>47455973</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>1.025359</td>\n",
       "      <td>10.81</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>c.2004-5G&gt;C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Benign</td>\n",
       "      <td>genomewide</td>\n",
       "      <td>Variant CADD score of 10.81 is less than a glo...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>237</th>\n",
       "      <td>18</td>\n",
       "      <td>47462711</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.938279</td>\n",
       "      <td>10.3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>c.1914C&gt;G</td>\n",
       "      <td>p.Thr638Thr</td>\n",
       "      <td>2202/9520</td>\n",
       "      <td>1914/5547</td>\n",
       "      <td>638/1848</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Benign</td>\n",
       "      <td>genomewide</td>\n",
       "      <td>Variant CADD score of 10.3 is less than a glob...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>238</th>\n",
       "      <td>18</td>\n",
       "      <td>47463708</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>-0.723906</td>\n",
       "      <td>0.063</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>c.1812T&gt;C</td>\n",
       "      <td>p.Pro604Pro</td>\n",
       "      <td>2100/9520</td>\n",
       "      <td>1812/5547</td>\n",
       "      <td>604/1848</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Benign</td>\n",
       "      <td>calibrated</td>\n",
       "      <td>Variant CADD score of 0.063 is less than 8.87 ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>239</th>\n",
       "      <td>18</td>\n",
       "      <td>47479624</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.960552</td>\n",
       "      <td>10.43</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>c.1752+6G&gt;C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Benign</td>\n",
       "      <td>genomewide</td>\n",
       "      <td>Variant CADD score of 10.43 is less than a glo...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>240</th>\n",
       "      <td>18</td>\n",
       "      <td>47480678</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>1.128826</td>\n",
       "      <td>11.37</td>\n",
       "      <td>8.266E-6</td>\n",
       "      <td>...</td>\n",
       "      <td>c.1668+5C&gt;G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Benign</td>\n",
       "      <td>genomewide</td>\n",
       "      <td>Variant CADD score of 11.37 is less than a glo...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>241</th>\n",
       "      <td>18</td>\n",
       "      <td>47488781</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.247209</td>\n",
       "      <td>5.169</td>\n",
       "      <td>1.655E-5</td>\n",
       "      <td>...</td>\n",
       "      <td>c.1405-5C&gt;G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Benign</td>\n",
       "      <td>calibrated</td>\n",
       "      <td>Variant CADD score of 5.169 is less than 8.87 ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>242</th>\n",
       "      <td>18</td>\n",
       "      <td>47500740</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>1.763419</td>\n",
       "      <td>14.77</td>\n",
       "      <td>8.256E-6</td>\n",
       "      <td>...</td>\n",
       "      <td>c.1302C&gt;T</td>\n",
       "      <td>p.Ile434Ile</td>\n",
       "      <td>1590/9520</td>\n",
       "      <td>1302/5547</td>\n",
       "      <td>434/1848</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Benign</td>\n",
       "      <td>genomewide</td>\n",
       "      <td>Variant CADD score of 14.77 is less than a glo...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>243</th>\n",
       "      <td>18</td>\n",
       "      <td>47500862</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.797472</td>\n",
       "      <td>9.448</td>\n",
       "      <td>8.245E-6</td>\n",
       "      <td>...</td>\n",
       "      <td>c.1180C&gt;T</td>\n",
       "      <td>p.Leu394Leu</td>\n",
       "      <td>1468/9520</td>\n",
       "      <td>1180/5547</td>\n",
       "      <td>394/1848</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Benign</td>\n",
       "      <td>genomewide</td>\n",
       "      <td>Variant CADD score of 9.448 is less than a glo...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>244</th>\n",
       "      <td>18</td>\n",
       "      <td>47506856</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>1.677378</td>\n",
       "      <td>14.28</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>c.1014G&gt;A</td>\n",
       "      <td>p.Val338Val</td>\n",
       "      <td>1302/9520</td>\n",
       "      <td>1014/5547</td>\n",
       "      <td>338/1848</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Benign</td>\n",
       "      <td>genomewide</td>\n",
       "      <td>Variant CADD score of 14.28 is less than a glo...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>245</th>\n",
       "      <td>18</td>\n",
       "      <td>47511158</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.978845</td>\n",
       "      <td>10.54</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>c.876A&gt;T</td>\n",
       "      <td>p.Gly292Gly</td>\n",
       "      <td>1164/9520</td>\n",
       "      <td>876/5547</td>\n",
       "      <td>292/1848</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Benign</td>\n",
       "      <td>genomewide</td>\n",
       "      <td>Variant CADD score of 10.54 is less than a glo...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>246</th>\n",
       "      <td>18</td>\n",
       "      <td>47516882</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>-0.609241</td>\n",
       "      <td>0.119</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>c.765T&gt;C</td>\n",
       "      <td>p.Asp255Asp</td>\n",
       "      <td>1053/9520</td>\n",
       "      <td>765/5547</td>\n",
       "      <td>255/1848</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Benign</td>\n",
       "      <td>calibrated</td>\n",
       "      <td>Variant CADD score of 0.119 is less than 8.87 ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>247</th>\n",
       "      <td>18</td>\n",
       "      <td>47527640</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.989007</td>\n",
       "      <td>10.6</td>\n",
       "      <td>8.268E-6</td>\n",
       "      <td>...</td>\n",
       "      <td>c.597C&gt;T</td>\n",
       "      <td>p.Ser199Ser</td>\n",
       "      <td>885/9520</td>\n",
       "      <td>597/5547</td>\n",
       "      <td>199/1848</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Benign</td>\n",
       "      <td>genomewide</td>\n",
       "      <td>Variant CADD score of 10.6 is less than a glob...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>248 rows  31 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    CHROM       POS ID   REF   ALT QUAL FILTER       CADD CADD_SCALED  \\\n",
       "0      18  47352774  .     T     C    .      .   0.334537       6.023   \n",
       "1      18  47361716  .     G     A    .      .  14.849415        48.0   \n",
       "2      18  47361725  .     G     A    .      .  15.701726        52.0   \n",
       "3      18  47365526  .     G     A    .      .  14.577744        46.0   \n",
       "4      18  47365610  .     C   CCA    .      .   9.553096        35.0   \n",
       "5      18  47367768  .    AA    GC    .      .   6.332952        29.3   \n",
       "6      18  47369763  .     C     G    .      .   4.451087        24.2   \n",
       "7      18  47373609  .     G     A    .      .  12.325906        39.0   \n",
       "8      18  47380012  .     A     G    .      .   6.116766        28.3   \n",
       "9      18  47402080  .     G     A    .      .   12.92118        40.0   \n",
       "10     18  47405401  .     G     A    .      .  12.776541        40.0   \n",
       "11     18  47405427  .     A  AGAG    .      .   0.524487       7.615   \n",
       "12     18  47406825  .     G     A    .      .  10.692736        36.0   \n",
       "13     18  47408592  .     C    CC    .      .   0.132595       3.958   \n",
       "14     18  47429043  .  CGGG     C    .      .   2.817738        21.5   \n",
       "15     18  47429104  .     G     A    .      .  11.752624        38.0   \n",
       "16     18  47432872  .    TC     T    .      .   8.837797        35.0   \n",
       "17     18  47435205  .     T    TC    .      .  -0.336167       0.549   \n",
       "18     18  47462646  .     G     A    .      .   6.923516        33.0   \n",
       "19     18  47462659  .     G     A    .      .   7.604902        34.0   \n",
       "20     18  47462720  .     C     T    .      .   5.482661        26.2   \n",
       "21     18  47463664  .     G     A    .      .   2.905477        21.9   \n",
       "22     18  47480703  .     T     A    .      .   3.829816        23.4   \n",
       "23     18  47480738  .     A     G    .      .   6.318276        29.2   \n",
       "24     18  47480760  .     G     A    .      .   5.852421        27.3   \n",
       "25     18  47480769  .     G     A    .      .   5.782332        27.1   \n",
       "26     18  47488641  .     A     G    .      .   6.073709        28.1   \n",
       "27     18  47489356  .     T     C    .      .   5.265616        25.7   \n",
       "28     18  47489375  .    AG     A    .      .   7.952294        35.0   \n",
       "29     18  47489402  .     T     C    .      .   3.928802        23.5   \n",
       "..    ...       ... ..   ...   ...  ...    ...        ...         ...   \n",
       "218    18  47518698  .     A     G    .      .   5.635733        26.6   \n",
       "219    18  47518722  .     C     G    .      .   5.061619        25.2   \n",
       "220    18  47518777  .     G     T    .      .   6.620644        32.0   \n",
       "221    18  47527653  .     T     A    .      .   5.910215        27.5   \n",
       "222    18  47527686  .     C     T    .      .   4.734543        24.6   \n",
       "223    18  47405336  .     C     A    .      .   1.690176       14.35   \n",
       "224    18  47405414  .     T     C    .      .   0.071664       3.308   \n",
       "225    18  47406817  .     T     C    .      .   0.069248       3.282   \n",
       "226    18  47421416  .     C     T    .      .    -0.1534       1.366   \n",
       "227    18  47428960  .     T     C    .      .   0.624515       8.325   \n",
       "228    18  47429084  .     G     C    .      .   0.565037       7.911   \n",
       "229    18  47429209  .     G     A    .      .  -0.190754       1.147   \n",
       "230    18  47431078  .     G     A    .      .   1.248168        12.0   \n",
       "231    18  47431179  .     T     G    .      .  -0.029409       2.313   \n",
       "232    18  47432836  .     C     A    .      .   0.758393       9.204   \n",
       "233    18  47432941  .     G     C    .      .   0.089257       3.493   \n",
       "234    18  47438492  .     T     C    .      .  -0.199414       1.101   \n",
       "235    18  47455877  .     G     T    .      .   0.421282       6.797   \n",
       "236    18  47455973  .     C     G    .      .   1.025359       10.81   \n",
       "237    18  47462711  .     G     C    .      .   0.938279        10.3   \n",
       "238    18  47463708  .     A     G    .      .  -0.723906       0.063   \n",
       "239    18  47479624  .     C     G    .      .   0.960552       10.43   \n",
       "240    18  47480678  .     G     C    .      .   1.128826       11.37   \n",
       "241    18  47488781  .     G     C    .      .   0.247209       5.169   \n",
       "242    18  47500740  .     G     A    .      .   1.763419       14.77   \n",
       "243    18  47500862  .     G     A    .      .   0.797472       9.448   \n",
       "244    18  47506856  .     C     T    .      .   1.677378       14.28   \n",
       "245    18  47511158  .     T     A    .      .   0.978845       10.54   \n",
       "246    18  47516882  .     A     G    .      .  -0.609241       0.119   \n",
       "247    18  47527640  .     G     A    .      .   0.989007        10.6   \n",
       "\n",
       "      EXAC_AF                        ...                          \\\n",
       "0         NaN                        ...                           \n",
       "1         NaN                        ...                           \n",
       "2         NaN                        ...                           \n",
       "3         NaN                        ...                           \n",
       "4         NaN                        ...                           \n",
       "5         NaN                        ...                           \n",
       "6         NaN                        ...                           \n",
       "7         NaN                        ...                           \n",
       "8    8.264E-6                        ...                           \n",
       "9         NaN                        ...                           \n",
       "10   1.655E-5                        ...                           \n",
       "11        NaN                        ...                           \n",
       "12   2.483E-5                        ...                           \n",
       "13        NaN                        ...                           \n",
       "14        NaN                        ...                           \n",
       "15        NaN                        ...                           \n",
       "16        NaN                        ...                           \n",
       "17        NaN                        ...                           \n",
       "18        NaN                        ...                           \n",
       "19   8.264E-6                        ...                           \n",
       "20        NaN                        ...                           \n",
       "21   8.271E-6                        ...                           \n",
       "22        NaN                        ...                           \n",
       "23        NaN                        ...                           \n",
       "24   8.265E-6                        ...                           \n",
       "25        NaN                        ...                           \n",
       "26        NaN                        ...                           \n",
       "27        NaN                        ...                           \n",
       "28        NaN                        ...                           \n",
       "29        NaN                        ...                           \n",
       "..        ...                        ...                           \n",
       "218  8.271E-6                        ...                           \n",
       "219  2.481E-5                        ...                           \n",
       "220  1.655E-5                        ...                           \n",
       "221  8.271E-6                        ...                           \n",
       "222       NaN                        ...                           \n",
       "223       NaN                        ...                           \n",
       "224  1.656E-5                        ...                           \n",
       "225       NaN                        ...                           \n",
       "226  8.263E-6                        ...                           \n",
       "227  8.264E-6                        ...                           \n",
       "228       NaN                        ...                           \n",
       "229       NaN                        ...                           \n",
       "230  8.272E-6                        ...                           \n",
       "231   8.27E-6                        ...                           \n",
       "232  8.269E-6                        ...                           \n",
       "233  8.275E-6                        ...                           \n",
       "234       NaN                        ...                           \n",
       "235       NaN                        ...                           \n",
       "236       NaN                        ...                           \n",
       "237       NaN                        ...                           \n",
       "238       NaN                        ...                           \n",
       "239       NaN                        ...                           \n",
       "240  8.266E-6                        ...                           \n",
       "241  1.655E-5                        ...                           \n",
       "242  8.256E-6                        ...                           \n",
       "243  8.245E-6                        ...                           \n",
       "244       NaN                        ...                           \n",
       "245       NaN                        ...                           \n",
       "246       NaN                        ...                           \n",
       "247  8.268E-6                        ...                           \n",
       "\n",
       "                    HGVS_c                    HGVS_p cDNA_position  \\\n",
       "0                 c.*67A>G                       NaN           NaN   \n",
       "1                c.5392C>T                p.Gln1798*     5680/9520   \n",
       "2                c.5383C>T                p.Arg1795*     5671/9520   \n",
       "3                c.4840C>T                p.Gln1614*     5128/9520   \n",
       "4         c.4755_4756insTG               p.Asp1586fs     5043/9520   \n",
       "5    c.4667_4668delTTinsGC              p.Leu1556Arg     4956/9520   \n",
       "6              c.4460-1G>C                       NaN           NaN   \n",
       "7                c.4366C>T                p.Gln1456*     4654/9520   \n",
       "8                c.4028T>C              p.Leu1343Pro     4316/9520   \n",
       "9                c.3514C>T                p.Gln1172*     3802/9520   \n",
       "10               c.3190C>T                p.Arg1064*     3478/9520   \n",
       "11       c.3163_3164insCTC   p.Asn1054_Leu1055insPro     3451/9520   \n",
       "12               c.3046C>T                p.Arg1016*     3334/9520   \n",
       "13         c.3046-1768dupG                       NaN           NaN   \n",
       "14       c.2729_2731delCCC  p.Ala910_Arg911delinsGly     3019/9520   \n",
       "15               c.2671C>T                 p.Gln891*     2959/9520   \n",
       "16              c.2330delG                p.Gly777fs     2618/9520   \n",
       "17         c.2203-2206dupG                       NaN           NaN   \n",
       "18               c.1979C>T               p.Pro660Leu     2267/9520   \n",
       "19               c.1966C>T               p.Arg656Cys     2254/9520   \n",
       "20             c.1906-1G>A                       NaN           NaN   \n",
       "21               c.1856C>T               p.Pro619Leu     2144/9520   \n",
       "22               c.1648A>T               p.Ile550Phe     1936/9520   \n",
       "23               c.1613T>C               p.Phe538Ser     1901/9520   \n",
       "24               c.1591C>T               p.Arg531Trp     1879/9520   \n",
       "25               c.1582C>T               p.Leu528Phe     1870/9520   \n",
       "26               c.1540T>C               p.Cys514Arg     1828/9520   \n",
       "27               c.1367A>G               p.Asn456Ser     1655/9520   \n",
       "28              c.1347delC                p.Phe450fs     1635/9520   \n",
       "29             c.1323-2A>G                       NaN           NaN   \n",
       "..                     ...                       ...           ...   \n",
       "218               c.716T>C               p.Met239Thr     1004/9520   \n",
       "219               c.692G>C               p.Arg231Thr      980/9520   \n",
       "220               c.637C>A               p.Arg213Ser      925/9520   \n",
       "221               c.584A>T               p.Lys195Met      872/9520   \n",
       "222               c.551G>A               p.Gly184Asp      839/9520   \n",
       "223              c.3255G>T              p.Arg1085Arg     3543/9520   \n",
       "224              c.3177A>G              p.Glu1059Glu     3465/9520   \n",
       "225              c.3054A>G              p.Ala1018Ala     3342/9520   \n",
       "226              c.2940G>A               p.Arg980Arg     3228/9520   \n",
       "227            c.2811+4A>G                       NaN           NaN   \n",
       "228              c.2691C>G               p.Leu897Leu     2979/9520   \n",
       "229            c.2572-6C>T                       NaN           NaN   \n",
       "230              c.2535C>T               p.Phe845Phe     2823/9520   \n",
       "231              c.2434A>C               p.Arg812Arg     2722/9520   \n",
       "232              c.2367G>T               p.Gly789Gly     2655/9520   \n",
       "233              c.2262C>G               p.Ala754Ala     2550/9520   \n",
       "234              c.2142A>G               p.Arg714Arg     2430/9520   \n",
       "235            c.2090+5C>A                       NaN           NaN   \n",
       "236            c.2004-5G>C                       NaN           NaN   \n",
       "237              c.1914C>G               p.Thr638Thr     2202/9520   \n",
       "238              c.1812T>C               p.Pro604Pro     2100/9520   \n",
       "239            c.1752+6G>C                       NaN           NaN   \n",
       "240            c.1668+5C>G                       NaN           NaN   \n",
       "241            c.1405-5C>G                       NaN           NaN   \n",
       "242              c.1302C>T               p.Ile434Ile     1590/9520   \n",
       "243              c.1180C>T               p.Leu394Leu     1468/9520   \n",
       "244              c.1014G>A               p.Val338Val     1302/9520   \n",
       "245               c.876A>T               p.Gly292Gly     1164/9520   \n",
       "246               c.765T>C               p.Asp255Asp     1053/9520   \n",
       "247               c.597C>T               p.Ser199Ser      885/9520   \n",
       "\n",
       "    CDS_position Protein_position Distance_to_feature Errors Classification  \\\n",
       "0            NaN              NaN                  67    NaN         Benign   \n",
       "1      5392/5547        1798/1848                 NaN    NaN     Pathogenic   \n",
       "2      5383/5547        1795/1848                 NaN    NaN     Pathogenic   \n",
       "3      4840/5547        1614/1848                 NaN    NaN     Pathogenic   \n",
       "4      4755/5547        1585/1848                 NaN    NaN     Pathogenic   \n",
       "5      4667/5547        1556/1848                 NaN    NaN     Pathogenic   \n",
       "6            NaN              NaN                 NaN    NaN     Pathogenic   \n",
       "7      4366/5547        1456/1848                 NaN    NaN     Pathogenic   \n",
       "8      4028/5547        1343/1848                 NaN    NaN     Pathogenic   \n",
       "9      3514/5547        1172/1848                 NaN    NaN     Pathogenic   \n",
       "10     3190/5547        1064/1848                 NaN    NaN     Pathogenic   \n",
       "11     3163/5547        1055/1848                 NaN    NaN         Benign   \n",
       "12     3046/5547        1016/1848                 NaN    NaN     Pathogenic   \n",
       "13           NaN              NaN                 NaN    NaN         Benign   \n",
       "14     2729/5547         910/1848                 NaN    NaN     Pathogenic   \n",
       "15     2671/5547         891/1848                 NaN    NaN     Pathogenic   \n",
       "16     2330/5547         777/1848                 NaN    NaN     Pathogenic   \n",
       "17           NaN              NaN                 NaN    NaN         Benign   \n",
       "18     1979/5547         660/1848                 NaN    NaN     Pathogenic   \n",
       "19     1966/5547         656/1848                 NaN    NaN     Pathogenic   \n",
       "20           NaN              NaN                 NaN    NaN     Pathogenic   \n",
       "21     1856/5547         619/1848                 NaN    NaN     Pathogenic   \n",
       "22     1648/5547         550/1848                 NaN    NaN     Pathogenic   \n",
       "23     1613/5547         538/1848                 NaN    NaN     Pathogenic   \n",
       "24     1591/5547         531/1848                 NaN    NaN     Pathogenic   \n",
       "25     1582/5547         528/1848                 NaN    NaN     Pathogenic   \n",
       "26     1540/5547         514/1848                 NaN    NaN     Pathogenic   \n",
       "27     1367/5547         456/1848                 NaN    NaN     Pathogenic   \n",
       "28     1347/5547         449/1848                 NaN    NaN     Pathogenic   \n",
       "29           NaN              NaN                 NaN    NaN     Pathogenic   \n",
       "..           ...              ...                 ...    ...            ...   \n",
       "218     716/5547         239/1848                 NaN    NaN     Pathogenic   \n",
       "219     692/5547         231/1848                 NaN    NaN     Pathogenic   \n",
       "220     637/5547         213/1848                 NaN    NaN     Pathogenic   \n",
       "221     584/5547         195/1848                 NaN    NaN     Pathogenic   \n",
       "222     551/5547         184/1848                 NaN    NaN     Pathogenic   \n",
       "223    3255/5547        1085/1848                 NaN    NaN         Benign   \n",
       "224    3177/5547        1059/1848                 NaN    NaN         Benign   \n",
       "225    3054/5547        1018/1848                 NaN    NaN         Benign   \n",
       "226    2940/5547         980/1848                 NaN    NaN         Benign   \n",
       "227          NaN              NaN                 NaN    NaN         Benign   \n",
       "228    2691/5547         897/1848                 NaN    NaN         Benign   \n",
       "229          NaN              NaN                 NaN    NaN         Benign   \n",
       "230    2535/5547         845/1848                 NaN    NaN         Benign   \n",
       "231    2434/5547         812/1848                 NaN    NaN         Benign   \n",
       "232    2367/5547         789/1848                 NaN    NaN         Benign   \n",
       "233    2262/5547         754/1848                 NaN    NaN         Benign   \n",
       "234    2142/5547         714/1848                 NaN    NaN         Benign   \n",
       "235          NaN              NaN                 NaN    NaN         Benign   \n",
       "236          NaN              NaN                 NaN    NaN         Benign   \n",
       "237    1914/5547         638/1848                 NaN    NaN         Benign   \n",
       "238    1812/5547         604/1848                 NaN    NaN         Benign   \n",
       "239          NaN              NaN                 NaN    NaN         Benign   \n",
       "240          NaN              NaN                 NaN    NaN         Benign   \n",
       "241          NaN              NaN                 NaN    NaN         Benign   \n",
       "242    1302/5547         434/1848                 NaN    NaN         Benign   \n",
       "243    1180/5547         394/1848                 NaN    NaN         Benign   \n",
       "244    1014/5547         338/1848                 NaN    NaN         Benign   \n",
       "245     876/5547         292/1848                 NaN    NaN         Benign   \n",
       "246     765/5547         255/1848                 NaN    NaN         Benign   \n",
       "247     597/5547         199/1848                 NaN    NaN         Benign   \n",
       "\n",
       "     Confidence                                             Reason  \n",
       "0    calibrated  Variant CADD score of 6.023 is less than 8.87 ...  \n",
       "1    calibrated  Variant CADD score of 48.0 is greater than 34....  \n",
       "2    calibrated  Variant CADD score of 52.0 is greater than 34....  \n",
       "3    calibrated  Variant CADD score of 46.0 is greater than 34....  \n",
       "4    calibrated  Variant CADD score of 35.0 is greater than 34....  \n",
       "5    genomewide  Variant MAF of 0.0 is rare enough to be potent...  \n",
       "6    genomewide  Variant MAF of 0.0 is rare enough to be potent...  \n",
       "7    calibrated  Variant CADD score of 39.0 is greater than 34....  \n",
       "8    genomewide  Variant MAF of 8.264E-6 is rare enough to be p...  \n",
       "9    calibrated  Variant CADD score of 40.0 is greater than 34....  \n",
       "10   calibrated  Variant CADD score of 40.0 is greater than 34....  \n",
       "11   calibrated  Variant CADD score of 7.615 is less than 8.87 ...  \n",
       "12   calibrated  Variant CADD score of 36.0 is greater than 34....  \n",
       "13   calibrated  Variant CADD score of 3.958 is less than 8.87 ...  \n",
       "14   genomewide  Variant MAF of 0.0 is rare enough to be potent...  \n",
       "15   calibrated  Variant CADD score of 38.0 is greater than 34....  \n",
       "16   calibrated  Variant CADD score of 35.0 is greater than 34....  \n",
       "17   calibrated  Variant CADD score of 0.549 is less than 8.87 ...  \n",
       "18   genomewide  Variant MAF of 0.0 is rare enough to be potent...  \n",
       "19   genomewide  Variant MAF of 8.264E-6 is rare enough to be p...  \n",
       "20   genomewide  Variant MAF of 0.0 is rare enough to be potent...  \n",
       "21   genomewide  Variant MAF of 8.271E-6 is rare enough to be p...  \n",
       "22   genomewide  Variant MAF of 0.0 is rare enough to be potent...  \n",
       "23   genomewide  Variant MAF of 0.0 is rare enough to be potent...  \n",
       "24   genomewide  Variant MAF of 8.265E-6 is rare enough to be p...  \n",
       "25   genomewide  Variant MAF of 0.0 is rare enough to be potent...  \n",
       "26   genomewide  Variant MAF of 0.0 is rare enough to be potent...  \n",
       "27   genomewide  Variant MAF of 0.0 is rare enough to be potent...  \n",
       "28   calibrated  Variant CADD score of 35.0 is greater than 34....  \n",
       "29   genomewide  Variant MAF of 0.0 is rare enough to be potent...  \n",
       "..          ...                                                ...  \n",
       "218  genomewide  Variant MAF of 8.271E-6 is rare enough to be p...  \n",
       "219  genomewide  Variant MAF of 2.481E-5 is rare enough to be p...  \n",
       "220  genomewide  Variant MAF of 1.655E-5 is rare enough to be p...  \n",
       "221  genomewide  Variant MAF of 8.271E-6 is rare enough to be p...  \n",
       "222  genomewide  Variant MAF of 0.0 is rare enough to be potent...  \n",
       "223  genomewide  Variant CADD score of 14.35 is less than a glo...  \n",
       "224  calibrated  Variant CADD score of 3.308 is less than 8.87 ...  \n",
       "225  calibrated  Variant CADD score of 3.282 is less than 8.87 ...  \n",
       "226  calibrated  Variant CADD score of 1.366 is less than 8.87 ...  \n",
       "227  calibrated  Variant CADD score of 8.325 is less than 8.87 ...  \n",
       "228  calibrated  Variant CADD score of 7.911 is less than 8.87 ...  \n",
       "229  calibrated  Variant CADD score of 1.147 is less than 8.87 ...  \n",
       "230  genomewide  Variant CADD score of 12.0 is less than a glob...  \n",
       "231  calibrated  Variant CADD score of 2.313 is less than 8.87 ...  \n",
       "232  genomewide  Variant CADD score of 9.204 is less than a glo...  \n",
       "233  calibrated  Variant CADD score of 3.493 is less than 8.87 ...  \n",
       "234  calibrated  Variant CADD score of 1.101 is less than 8.87 ...  \n",
       "235  calibrated  Variant CADD score of 6.797 is less than 8.87 ...  \n",
       "236  genomewide  Variant CADD score of 10.81 is less than a glo...  \n",
       "237  genomewide  Variant CADD score of 10.3 is less than a glob...  \n",
       "238  calibrated  Variant CADD score of 0.063 is less than 8.87 ...  \n",
       "239  genomewide  Variant CADD score of 10.43 is less than a glo...  \n",
       "240  genomewide  Variant CADD score of 11.37 is less than a glo...  \n",
       "241  calibrated  Variant CADD score of 5.169 is less than 8.87 ...  \n",
       "242  genomewide  Variant CADD score of 14.77 is less than a glo...  \n",
       "243  genomewide  Variant CADD score of 9.448 is less than a glo...  \n",
       "244  genomewide  Variant CADD score of 14.28 is less than a glo...  \n",
       "245  genomewide  Variant CADD score of 10.54 is less than a glo...  \n",
       "246  calibrated  Variant CADD score of 0.119 is less than 8.87 ...  \n",
       "247  genomewide  Variant CADD score of 10.6 is less than a glob...  \n",
       "\n",
       "[248 rows x 31 columns]"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "datatable.replace('',np.NaN)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "datatable['CADD'] = datatable['CADD'].astype('float')\n",
    "datatable['CADD_SCALED'] = datatable['CADD_SCALED'].astype('float')\n",
    "# datatable['EXAC_AF'] = datatable['EXAC_AF'].astype('float')\n",
    "# datatable['EXAC_AC_HOM'] = datatable['EXAC_AC_HOM'].astype('float')\n",
    "# datatable['Distance_to_feature'] = datatable['Distance_to_feature'].astype('int')\n",
    "# datatable['Errors'] = datatable['Errors'].astype('int')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "datatable.to_csv('data/gavin_res.csv',sep='\\t', index=False, header=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0       0.334537\n",
       "1      14.849415\n",
       "2      15.701726\n",
       "3      14.577744\n",
       "4       9.553096\n",
       "5       6.332952\n",
       "6       4.451087\n",
       "7      12.325906\n",
       "8       6.116766\n",
       "9      12.921180\n",
       "10     12.776541\n",
       "11      0.524487\n",
       "12     10.692736\n",
       "13      0.132595\n",
       "14      2.817738\n",
       "15     11.752624\n",
       "16      8.837797\n",
       "17     -0.336167\n",
       "18      6.923516\n",
       "19      7.604902\n",
       "20      5.482661\n",
       "21      2.905477\n",
       "22      3.829816\n",
       "23      6.318276\n",
       "24      5.852421\n",
       "25      5.782332\n",
       "26      6.073709\n",
       "27      5.265616\n",
       "28      7.952294\n",
       "29      3.928802\n",
       "         ...    \n",
       "218     5.635733\n",
       "219     5.061619\n",
       "220     6.620644\n",
       "221     5.910215\n",
       "222     4.734543\n",
       "223     1.690176\n",
       "224     0.071664\n",
       "225     0.069248\n",
       "226    -0.153400\n",
       "227     0.624515\n",
       "228     0.565037\n",
       "229    -0.190754\n",
       "230     1.248168\n",
       "231    -0.029409\n",
       "232     0.758393\n",
       "233     0.089257\n",
       "234    -0.199414\n",
       "235     0.421282\n",
       "236     1.025359\n",
       "237     0.938279\n",
       "238    -0.723906\n",
       "239     0.960552\n",
       "240     1.128826\n",
       "241     0.247209\n",
       "242     1.763419\n",
       "243     0.797472\n",
       "244     1.677378\n",
       "245     0.978845\n",
       "246    -0.609241\n",
       "247     0.989007\n",
       "Name: CADD, Length: 248, dtype: float64"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "datatable.CADD"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "py3",
   "language": "python",
   "name": "py3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
