{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import os\n",
    "from os.path import isfile, join\n",
    "\n",
    "from sklearn.model_selection import train_test_split\n",
    "from sklearn.linear_model import LogisticRegression\n",
    "from sklearn.svm import SVC\n",
    "from sklearn.metrics import confusion_matrix\n",
    "\n",
    "import numpy as np\n",
    "import matplotlib.pyplot as plt\n",
    "\n",
    "# pandas display setting\n",
    "pd.options.display.max_rows = 999"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(248, 117)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>INFO</th>\n",
       "      <th>Type</th>\n",
       "      <th>Length</th>\n",
       "      <th>isTv</th>\n",
       "      <th>isDerived</th>\n",
       "      <th>...</th>\n",
       "      <th>Intron</th>\n",
       "      <th>oAA</th>\n",
       "      <th>nAA</th>\n",
       "      <th>Grantham</th>\n",
       "      <th>PolyPhenCat</th>\n",
       "      <th>PolyPhenVal</th>\n",
       "      <th>SIFTcat</th>\n",
       "      <th>SIFTval</th>\n",
       "      <th>RawScore</th>\n",
       "      <th>PHRED</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>18</td>\n",
       "      <td>47352774</td>\n",
       "      <td>MYO5B:c.5616-2A&gt;G</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.334537</td>\n",
       "      <td>6.023</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>18</td>\n",
       "      <td>47361716</td>\n",
       "      <td>MYO5B:c.5392C&gt;T</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Q</td>\n",
       "      <td>*</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>14.849415</td>\n",
       "      <td>48.000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>18</td>\n",
       "      <td>47361725</td>\n",
       "      <td>MYO5B:c.5383C&gt;T</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R</td>\n",
       "      <td>*</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>15.701726</td>\n",
       "      <td>52.000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>18</td>\n",
       "      <td>47365526</td>\n",
       "      <td>MYO5B:c.4840C&gt;T</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Q</td>\n",
       "      <td>*</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>14.577744</td>\n",
       "      <td>46.000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>18</td>\n",
       "      <td>47365610</td>\n",
       "      <td>MYO5B:c.4755_4756dupT</td>\n",
       "      <td>C</td>\n",
       "      <td>CCA</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>INS</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>9.553096</td>\n",
       "      <td>35.000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows Ã— 117 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   CHROM       POS                     ID REF  ALT        INFO Type  Length  \\\n",
       "0     18  47352774      MYO5B:c.5616-2A>G   T    C  Pathogenic  SNV       0   \n",
       "1     18  47361716        MYO5B:c.5392C>T   G    A  Pathogenic  SNV       0   \n",
       "2     18  47361725        MYO5B:c.5383C>T   G    A  Pathogenic  SNV       0   \n",
       "3     18  47365526        MYO5B:c.4840C>T   G    A  Pathogenic  SNV       0   \n",
       "4     18  47365610  MYO5B:c.4755_4756dupT   C  CCA  Pathogenic  INS       2   \n",
       "\n",
       "    isTv  isDerived   ...   Intron  oAA  nAA Grantham  PolyPhenCat  \\\n",
       "0  False       True   ...      NaN  NaN  NaN      NaN          NaN   \n",
       "1  False       True   ...      NaN    Q    *      NaN          NaN   \n",
       "2  False       True   ...      NaN    R    *      NaN          NaN   \n",
       "3  False       True   ...      NaN    Q    *      NaN          NaN   \n",
       "4    NaN       True   ...      NaN  NaN  NaN      NaN          NaN   \n",
       "\n",
       "   PolyPhenVal  SIFTcat  SIFTval   RawScore   PHRED  \n",
       "0          NaN      NaN      NaN   0.334537   6.023  \n",
       "1          NaN      NaN      NaN  14.849415  48.000  \n",
       "2          NaN      NaN      NaN  15.701726  52.000  \n",
       "3          NaN      NaN      NaN  14.577744  46.000  \n",
       "4          NaN      NaN      NaN   9.553096  35.000  \n",
       "\n",
       "[5 rows x 117 columns]"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# read file from data folder\n",
    "datafolder = 'data/'\n",
    "datafile = 'myo5b_variants_patho_benign_cadd1.3fullannot_v1.xlsx'\n",
    "datatable = pd.ExcelFile(os.path.join(datafolder,datafile))\n",
    "datatable_pd = datatable.parse(datatable.sheet_names[0])\n",
    "\n",
    "# number of features and samples\n",
    "print(datatable_pd.shape)\n",
    "\n",
    "# display data\n",
    "datatable_pd.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "def hist_compare(datatable_pd, col_name):\n",
    "    '''\n",
    "    compare the column(col_name) value for different classes\n",
    "    in the database(datatable_pd) and plot histograms/bar plot \n",
    "    for the pathogenic and benign class\n",
    "    \n",
    "    currently only support 'Pathogenic' and 'Benign' classes\n",
    "    no legend in the plot\n",
    "    '''\n",
    "    col = datatable_pd[col_name]\n",
    "    col_patho = col[datatable_pd['INFO']=='Pathogenic']\n",
    "    col_benig = col[datatable_pd['INFO']=='Benign']\n",
    "    if datatable_pd[col_name].dtype == np.object:\n",
    "        x = datatable_pd[col_name].unique()\n",
    "        y_path = []\n",
    "        y_beni = []\n",
    "        for name in x:\n",
    "            y_name = datatable_pd['INFO'][datatable_pd[col_name]==name]\n",
    "            y_path.append(y_name[y_name=='Pathogenic'].count())\n",
    "            y_beni.append(y_name[y_name=='Benign'].count())\n",
    "        plt.bar(x,y_path,alpha=0.5)\n",
    "        plt.bar(x,y_beni,alpha=0.5)\n",
    "    else:\n",
    "        n,bins,patches = plt.hist(col_patho,50,alpha=0.5)\n",
    "        n,bins,patches = plt.hist(col_benig,50,alpha=0.5)\n",
    "    plt.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "x = datatable_pd.REF.unique()\n",
    "y_path = []\n",
    "y_beni = []\n",
    "for name in x:\n",
    "    y_name = datatable_pd['INFO'][datatable_pd['REF']==name]\n",
    "    y_path.append(y_name[y_name=='Pathogenic'].count())\n",
    "    y_beni.append(y_name[y_name=='Benign'].count())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAXQAAAD8CAYAAABn919SAAAABHNCSVQICAgIfAhkiAAAAAlwSFlzAAALEgAACxIB0t1+/AAAADl0RVh0U29mdHdhcmUAbWF0cGxvdGxpYiB2ZXJzaW9uIDIuMS4wLCBodHRwOi8vbWF0cGxvdGxpYi5vcmcvpW3flQAAE55JREFUeJzt3XuwJGV5x/HvIwsionI72XARFiOgKIJ6CsULuiARo8KabNCtxKyK2SSFt3gJLP4hVCzBxFJJxViFYlyNChSKSyFBCCzegsgii1wWWFggglwOtwAGL+CTP/o9MBxnztw5nJfvp+rUTHe/3f3MO92/6enpmROZiSRp/nvKXBcgSRoNA12SKmGgS1IlDHRJqoSBLkmVMNAlqRIGuiRVwkCXpEoY6JJUiQWP58q22267XLRo0eO5Skma9y655JI7M3OiW7vHNdAXLVrE2rVrH89VStK8FxE39dLOUy6SVAkDXZIqYaBLUiUMdEmqhIEuSZUw0CWpEga6JFXCQJekSvQU6BGxVUScFhFXR8T6iNgvIraJiHMjYkO53XrcxUqSOuv1m6InAGdn5tKI2AzYAjgaOC8zj4+Io4CjgCPHVKceb2uO63+exStHX4eknnU9Qo+IZwH7AycBZOZvMvNe4FBgVWm2ClgyriIlSd31csplV2AK+PeIuDQivhgRTwcWZuatpc1twMJxFSlJ6q6XQF8AvAT4fGa+GPglzemVR2RmAtlu5ohYERFrI2Lt1NTUsPVKkjroJdBvBm7OzIvK8Gk0AX97RGwPUG7vaDdzZp6YmZOZOTkx0fXXHyVJA+oa6Jl5G/DziNijjDoQuAo4A1hexi0HVo+lQklST3q9yuW9wNfKFS4bgXfSvBicGhGHAzcBh42nRElSL3oK9MxcB0y2mXTgaMuRJA3Kb4pKUiUMdEmqhIEuSZUw0CWpEga6JFXCQJekShjoklQJA12SKmGgS1IlDHRJqoSBLkmVMNAlqRIGuiRVwkCXpEoY6JJUCQNdkiphoEtSJQx0SaqEgS5JlTDQJakSBrokVcJAl6RKGOiSVAkDXZIqYaBLUiUW9NIoIm4E7gceBh7KzMmI2AY4BVgE3Agclpn3jKdMSVI3/RyhL87MfTJzsgwfBZyXmbsB55VhSdIcGeaUy6HAqnJ/FbBk+HIkSYPqNdATOCciLomIFWXcwsy8tdy/DVg48uokST3r6Rw68KrMvCUi/gA4NyKubp2YmRkR2W7G8gKwAmDnnXceqljpCW3Ncf21X7xyPHXoSaunI/TMvKXc3gGcDuwL3B4R2wOU2zs6zHtiZk5m5uTExMRoqpYk/Z6ugR4RT4+IZ0zfB/4YuAI4A1hemi0HVo+rSElSd72cclkInB4R0+2/nplnR8TFwKkRcThwE3DY+MqUJHXTNdAzcyOwd5vxdwEHjqMoSVL//KaoJFXCQJekShjoklQJA12SKmGgS1IlDHRJqoSBLkmVMNAlqRIGuiRVwkCXpEoY6JJUCQNdkiphoEtSJQx0SaqEgS5JlTDQJakSBrokVcJAl6RKGOiSVIle/km0pJqtOa7/eRavHH0dGppH6JJUCQNdkiphoEtSJQx0SaqEgS5Jleg50CNik4i4NCLOLMO7RsRFEXFdRJwSEZuNr0xJUjf9HKG/H1jfMvxJ4DOZ+VzgHuDwURYmSepPT4EeETsBbwS+WIYDOAA4rTRZBSwZR4GSpN70eoT+WeAfgN+V4W2BezPzoTJ8M7DjiGuTJPWha6BHxJuAOzLzkkFWEBErImJtRKydmpoaZBGSpB70coT+SuCQiLgROJnmVMsJwFYRMf3TATsBt7SbOTNPzMzJzJycmJgYQcmSpHa6BnpmrszMnTJzEfA24PzM/AtgDbC0NFsOrB5blZKkroa5Dv1I4IMRcR3NOfWTRlOSJGkQff3aYmZeAFxQ7m8E9h19SZKkQfhNUUmqhIEuSZUw0CWpEga6JFXCQJekShjoklQJA12SKmGgS1IlDHRJqoSBLkmVMNAlqRIGuiRVwkCXpEoY6JJUCQNdkiphoEtSJQx0SaqEgS5JlTDQJakSBrokVcJAl6RKGOiSVAkDXZIqYaBLUiUMdEmqRNdAj4jNI+InEXFZRFwZEceW8btGxEURcV1EnBIRm42/XElSJ70cof8aOCAz9wb2AQ6OiJcDnwQ+k5nPBe4BDh9fmZKkbroGejYeKIOblr8EDgBOK+NXAUvGUqEkqSc9nUOPiE0iYh1wB3AucD1wb2Y+VJrcDOw4nhIlSb3oKdAz8+HM3AfYCdgXeF6vK4iIFRGxNiLWTk1NDVimJKmbvq5yycx7gTXAfsBWEbGgTNoJuKXDPCdm5mRmTk5MTAxVrCSps16ucpmIiK3K/acBBwHraYJ9aWm2HFg9riIlSd0t6N6E7YFVEbEJzQvAqZl5ZkRcBZwcER8HLgVOGmOdkqQuugZ6Zv4MeHGb8RtpzqdLkp4A/KaoJFXCQJekSvRyDv2JYc1x/bVfvHI8dUjSE5RH6JJUCQNdkiphoEtSJQx0SaqEgS5JlTDQJakSBrokVcJAl6RKGOiSVAkDXZIqYaBLUiUMdEmqhIEuSZUw0CWpEga6JFXCQJekShjoklQJA12SKmGgS1IlDHRJqoSBLkmVMNAlqRJdAz0inh0RayLiqoi4MiLeX8ZvExHnRsSGcrv1+MuVJHXSyxH6Q8CHMnNP4OXAERGxJ3AUcF5m7gacV4YlSXOka6Bn5q2Z+dNy/35gPbAjcCiwqjRbBSwZV5GSpO76OoceEYuAFwMXAQsz89Yy6TZg4UgrkyT1pedAj4gtgW8CH8jM+1qnZWYC2WG+FRGxNiLWTk1NDVWsJKmzngI9IjalCfOvZea3yujbI2L7Mn174I5282bmiZk5mZmTExMTo6hZktRGL1e5BHASsD4zP90y6Qxgebm/HFg9+vIkSb1a0EObVwJvBy6PiHVl3NHA8cCpEXE4cBNw2HhKlOaHCzfe1Vf7/RaPqRA9aXUN9Mz8IRAdJh842nIkSYPym6KSVAkDXZIqYaBLUiUMdEmqhIEuSZUw0CWpEga6JFXCQJekShjoklQJA12SKmGgS1IlDHRJqoSBLkmVMNAlqRIGuiRVopd/cKEnoX7/WQP4DxukueYRuiRVwkCXpEoY6JJUCQNdkiphoEtSJQx0SaqEgS5JlTDQJakSBrokVaJroEfElyLijoi4omXcNhFxbkRsKLdbj7dMSVI3vRyhfxk4eMa4o4DzMnM34LwyLEmaQ10DPTO/D9w9Y/ShwKpyfxWwZMR1SZL6NOg59IWZeWu5fxuwcET1SJIGNPSHopmZQHaaHhErImJtRKydmpoadnWSpA4GDfTbI2J7gHJ7R6eGmXliZk5m5uTExMSAq5MkdTNooJ8BLC/3lwOrR1OOJGlQvVy2+A3gQmCPiLg5Ig4HjgcOiogNwOvKsCRpDnX9j0WZuazDpANHXIskaQh+U1SSKmGgS1Il/CfR0pOc/xC8Hh6hS1IlDHRJqoSBLkmVMNAlqRIGuiRVYt5c5dLvJ/F+Ci/pycYjdEmqhIEuSZUw0CWpEga6JFXCQJekShjoklQJA12SKmGgS1IlDHRJqoSBLkmVMNAlqRIGuiRVwkCXpErMm19blB4Xa47rr/3ilXWsW1XwCF2SKmGgS1IlhjrlEhEHAycAmwBfzMzjR1LViH3m3Gv7nufvD9r90YFh3goP+Tb6wpM+3Nfs+x3+qZGtexj99vmo+nvo53q+mq+niubwuZ6rbXScBj5Cj4hNgM8BbwD2BJZFxJ6jKkyS1J9hTrnsC1yXmRsz8zfAycChoylLktSvYQJ9R+DnLcM3l3GSpDkQmTnYjBFLgYMz891l+O3AyzLzPTParQBWlME9gGsGL7et7YA752j+uZp3Ltc9X+uey3Vb95Nn3cPW3ckumTnRtVVmDvQH7Ad8t2V4JbBy0OUNUcfauZp/rua17vm1but+8qx72LqH/RvmlMvFwG4RsWtEbAa8DThjiOVJkoYw8GWLmflQRLwH+C7NZYtfyswrR1aZJKkvQ12HnplnAWeNqJZBnTiH88/VvHO57vla91yu27qfPOsetu6hDPyhqCTpicWv/ktSLebyE9lh/4AlQALPG6Y98AHgV8Czepj3iOllALvTnHK6Efhf4H7gZ8CpwFuBdeXvAZrLNdcBt3YY/5Wynn2BC4ANwE+Bta01A38FXAvcB/ya5vr/NcD+vTxuYK+W9d8N3FDu/1eXvvtDmi+PXQ9cUh737oO079bfneYFdgPOLP19N/Bg6b+Z06fne0y/AAcDPwGuLo/5FGDnlukfLNPWA/cA97Ys5/8G6LOFwNeBjWU5FwJvGbQ98FngFuApg7Sfsa5vAz9uM37blu3jtjL/9PBmHdpfXp6L3wIPA78pw5cBS8u2126/m95/prf1U4GFHfaD7wB7tcz7l8CVZT0PlufnFy21XF3W+XCZdh/wmjbb1d7Msj902KbeNKOPHgR+WYb7es5HnomP14rGUnyzQ/4AOHaY9sBFZfw7e5j3inL78bKx/Wm5fTPwWuCF07ct814ATLZZ5mPGl43hRuAVLePOA64CjqX5mYVLy8Z1CPBU4K/LOt8xwOP+MrC0h36LsmH+bcu4vYFXD9J+tv6ebV6aF7JDpqdPP+6Z01vme6Rfyv0NwPNbph9CCfyyvLOBrcryjwCOAp45o/Zh+mwX4L2DtKd5N30T8GNgcb/tZ6xrK5ovBa4HnjPLYzgG+HAf++MxwDmt85Rtb6rUcmzL+M3L8/HmlnGvLc9Tu/3gVcCScv9gmrDcsQxvAqwGPtnSlxuBu4DrW7ahK9ptV532h1Jjx22qDB9P88K/HnhOP8/5OP4el5WMpXDYkuboYXfgmkHbA39E80r/GuCcLvPuQ3MUsjvNkfZXgHdRjq5nWfcF9Bbo/zhjo39MzcD3gX8GVo2in2ZuwLMs4wDg+32ss2P7bv3daV7gcGBVt+mz1PRVZn/B/jmwa7fH2kefHQh8r48+m7V9qessYDnNB299tZ8x7V3AvwEfA46eZRnHMESgl23vF8DtNAF7zYwa2u43M/eDNtN/wO+/SB3Tst4DaEL2CzRf8tmpl2145nPbbZsqbVYD/1368qR+nvNx/M3nc+iHAmdn5rXAXRHx0gHbv43mbdgPgD0iYmGneYEX0LwtewbN27nbaV6xLxn2wRQvoHl72bZmmiOKp81o002//dROv49xtvbd+rvTvNN90216Jx2nR8QzgS0z84YutfejWz39tl8GfAM4HXgjzamznttHxKZtpn2j3B+XQ2mOcM+kOTXxq5btb7Z+Hvi5LF5Fs3/eQnPa561d1jfoeijLvZSmL9/QQ/uxms+BvowmGCi33TbMTu2XASdn5u+AbwJ/Psu8y4DTyu3VwCDh2I9/BQ6KiBPK+jeb2SAiTo+IKyLiWx2W0W8/jVsv/d1Vt8c92/SI2DYi1kXEtRHR7veJdynTb4yIVwxSX5t1fi4iLouIi/ttX7649yfAtzPzPppTVs/vs/3rS7uFNOeFf1he5H8bES8cxWNsYxnNaYuTy9+dDLD9RcRFEbG+7Aczp+0VEeuA99Ec8AC8hOaFBJrz2iPZ5mduU6UvtwVuKH35cBmebt/Xcz4K8/Jf0EXENjRvofaKiKQ5h5YR8ZEs74N6bL+KZuM+NyKgCcwbaIJ05rx70/z42O/KpPtK+6/RnD4YhStpNsbVZb1PBR6ieet3D825wQdLGwAy8y0RMQl8aubC+u2nLnUtHbZ9ROxFl/6eZV1XAvvTnOZampn7z3jc09OBtv0y3beXZeZdwD4lzLfMzPsi4oGI2HV6/Zm5T0ScSZsX0R5dCfxZSz1HRMR2NB9y99v+9TTnvS8v/bYFzYfvm/bR/kGaI+XDgK2BG8q0Z9IE3kcHfJxttWx7mwP/2TJpt4j4CI+edmvnkf2gPLaXld+OetOM6Wsy83Ka5/InPNofe9Kcu96DZn/dkuYUTL8HYN22qcPK4zu6fMnyWa2PqYfnfOTm6xH6UuCrmblLZi7KzGfTBMOr+2x/AnBMGbcoM3cAdoiIXWbOS3Ne7wuZuQD4EfAWmoDcDnhFRLwxIvaPiBdO3w7wuD4HvKMcFU6v90PAaaXmDZQP8iLikIjYLCLeTbPDjqKfOjkfeGr5oTUAIuJFEdFpOW3bA/9C9/7uNO+1wCuBp7dM32Lm9Ig4pGVZrf3yT8BHI+L5HaYfB3ye5m359PI37+GxdnI+sHlE/F2H9fXTfhnw7ul+oznX/yJgiz7aHxQRW5RpB7dMeynNabBRW0oTZF/IzAUt+83dNNvf1yn7zfQMLftN634w87FB81x9KiJ2ahm3oCxjd5rTLetoTut9orR/GbBdH9swpcbZtqllwH8Anyh9uTewsI/nfPTm8gT+oH80lw4dPGPc+4DP99n+Bn7/UqpPA0fOnLd1GdPrornS5X9oPpF/gOayxWto3l4ubFnGBfTwoWgZ93LgezSXWl1F8/s4ky3rPZ8mvO6nuTzsRpoPol7Xbz/R4wd8pe0ONJeVXU9z5PIdYLc+22e3/p5tXTSXXZ5Fc8XEnaWPbmwzfSPN1QaP6Reac88Xl+foRzTnPacvpQzgI2XaVWX501cvPPJY++yz7cu2cAPN5ZJrgLf22X45TQg+c0bbbwF/02f7I2nOK8eMaT+l+aXUmfUcw4AfipZaLmvd/sq298OW7e95NJ9NbSh9/sh+w6P7wXU0Hzo+sh+U6ctpLlG8qky/hOZKsI/RXHmyQ9l27izLuJ/mgKDjNtzuue20TQGLSl8+po9o3jmd0+tzPuo/vykqSZWYr6dcJEkzGOiSVAkDXZIqYaBLUiUMdEmqhIEuSZUw0CWpEga6JFXi/wH1whE2Egfx2QAAAABJRU5ErkJggg==\n",
      "text/plain": [
       "<matplotlib.figure.Figure at 0x116647898>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "plt.bar(x,y_path,alpha=0.5)\n",
    "plt.bar(x,y_beni,alpha=0.5)\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAbIAAAD8CAYAAAASViG0AAAABHNCSVQICAgIfAhkiAAAAAlwSFlzAAALEgAACxIB0t1+/AAAADl0RVh0U29mdHdhcmUAbWF0cGxvdGxpYiB2ZXJzaW9uIDIuMS4wLCBodHRwOi8vbWF0cGxvdGxpYi5vcmcvpW3flQAAEydJREFUeJzt3XmQZWV5x/HvIwMiAoVAZxwZhsYIKiGCZgrFrTIscSKGRQk4FXVUrCktTMxiFDCVoJVyILFcKpJUJqCMVlAoFSEEjQTH4ILIICgDKMsACmEZUVDiBvrkj/M23Lne7rud2zMvfD9VXd1ne+5zzj33/Pqce/p2ZCaSJNXqCVu6AUmSxmGQSZKqZpBJkqpmkEmSqmaQSZKqZpBJkqpmkEmSqmaQSZKqZpBJkqq2YD4fbPfdd8/p6en5fEhJqt5VV131g8yc2tJ9bK3mNcimp6dZv379fD6kJFUvIm7f0j1szby0KEmqmkEmSaqaQSZJqppBJkmqmkEmSaqaQSZJqppBJkmqmkEmSaqaQSZJqtq8frLHWNatbqfOspPbqSNJ2ip4RiZJqppBJkmqmkEmSaqaQSZJqppBJkmqmkEmSaqaQSZJqppBJkmqmkEmSaqaQSZJqppBJkmqmkEmSaqaQSZJqppBJkmqmkEmSaqaQSZJqppBJkmqmkEmSaqaQSZJqppBJkmq2oJBZoqI24CfAL8CHs7MpRGxK3AuMA3cBhyXmT+aTJuSJPU2UJAVyzLzBx3DJwGXZuZpEXFSGX5nq91Js1m3up06y05up46kLWacS4tHAWvLz2uBo8dvR5Kk4QwaZAl8ISKuiohVZdzCzLyr/Hw3sLD17iRJ6mPQS4svzsw7I+K3gEsi4judEzMzIyJ7LViCbxXAkiVLxmpWkqRuA52RZead5fu9wPnAQcA9EbEIoHy/d5Zl12Tm0sxcOjU11U7XkiQVfYMsIp4cETvN/Az8AbABuBBYWWZbCVwwqSYlSZrNIJcWFwLnR8TM/Odk5ucj4krgvIg4AbgdOG5ybUqS1FvfIMvMjcABPcbfBxw6iaYkSRqUn+whSaqaQSZJqppBJkmqmkEmSaqaQSZJqppBJkmqmkEmSaqaQSZJqppBJkmqmkEmSaqaQSZJqppBJkmqmkEmSaqaQSZJqppBJkmqmkEmSaqaQSZJqppBJkmqmkEmSaqaQSZJqppBJkmq2oIt3YCGtG51O3WWndxOHUnawjwjkyRVzSCTJFXNIJMkVc0gkyRVzSCTJFXNIJMkVW3gIIuIbSLi6oi4qAzvHRFXRMTNEXFuRGw3uTYlSeptmDOytwE3dAyfDnwgM58B/Ag4oc3GJEkaxEBBFhGLgSOAM8twAIcAnyqzrAWOnkSDkiTNZdAzsg8C7wB+XYZ3A+7PzIfL8B3AHr0WjIhVEbE+ItZv2rRprGYlSerWN8gi4hXAvZl51SgPkJlrMnNpZi6dmpoapYQkSbMa5LMWXwQcGREvB7YHdgY+BOwSEQvKWdli4M7JtSlJUm99z8gy8+TMXJyZ08CrgS9m5p8A64Bjy2wrgQsm1qUkSbMY5+/I3gn8ZUTcTPOe2VnttCRJ0uCG+jcumfkl4Evl543AQe23JEnS4PxkD0lS1QwySVLVDDJJUtUMMklS1QwySVLVhrprUdLjyLrV49dYdvL4NaQ+PCOTJFXNIJMkVc0gkyRVzSCTJFXNIJMkVc0gkyRVzSCTJFXNIJMkVc0gkyRVzSCTJFXNIJMkVc0gkyRVzSCTJFXNIJMkVc0gkyRVzSCTJFXNIJMkVc0gkyRVzSCTJFXNIJMkVc0gkyRVrW+QRcT2EfGNiPhWRFwXEe8u4/eOiCsi4uaIODcitpt8u5IkbW6QM7JfAIdk5gHAgcDyiHgBcDrwgcx8BvAj4ITJtSlJUm99gywbD5bBbctXAocAnyrj1wJHT6RDSZLmMNB7ZBGxTURcA9wLXALcAtyfmQ+XWe4A9phMi5IkzW6gIMvMX2XmgcBi4CDgWYM+QESsioj1EbF+06ZNI7YpSVJvQ921mJn3A+uAg4FdImJBmbQYuHOWZdZk5tLMXDo1NTVWs5IkdRvkrsWpiNil/Pwk4HDgBppAO7bMthK4YFJNSpI0mwX9Z2ERsDYitqEJvvMy86KIuB74ZET8PXA1cNYE+5Qkqae+QZaZ3wae22P8Rpr3yyRJ2mL8ZA9JUtUMMklS1QwySVLVDDJJUtUMMklS1QwySVLVDDJJUtUMMklS1QwySVLVDDJJUtUMMklS1QwySVLVDDJJUtUMMklS1QwySVLVDDJJUtUMMklS1QwySVLVDDJJUtUMMklS1QwySVLVDDJJUtUMMklS1QwySVLVDDJJUtUMMklS1QwySVLV+gZZROwZEesi4vqIuC4i3lbG7xoRl0TETeX7UybfriRJmxvkjOxh4K8ycz/gBcCJEbEfcBJwaWbuA1xahiVJmld9gywz78rMb5affwLcAOwBHAWsLbOtBY6eVJOSJM1mqPfIImIaeC5wBbAwM+8qk+4GFrbamSRJA1gw6IwRsSPwaeDPM/PHEfHItMzMiMhZllsFrAJYsmTJyI1evvG+kZftdPCyVspIkrYSA52RRcS2NCH275n5mTL6nohYVKYvAu7ttWxmrsnMpZm5dGpqqo2eJUl6xCB3LQZwFnBDZr6/Y9KFwMry80rggvbbkyRpboNcWnwR8Frg2oi4pow7BTgNOC8iTgBuB46bTIuSJM2ub5Bl5leAmGXyoe22I0nScPxkD0lS1QwySVLVDDJJUtUMMklS1QwySVLVDDJJUtUMMklS1QwySVLVDDJJUtUMMklS1Qb+Ny6SHl/a+NdJ/tskzQfPyCRJVTPIJElVM8gkSVUzyCRJVTPIJElVM8gkSVUzyCRJVTPIJElVM8gkSVUzyCRJVTPIJElVM8gkSVUzyCRJVTPIJElVM8gkSVUzyCRJVTPIJElV6xtkEfGRiLg3IjZ0jNs1Ii6JiJvK96dMtk1Jknob5IzsbGB517iTgEszcx/g0jIsSdK86xtkmXkZ8MOu0UcBa8vPa4GjW+5LkqSBjPoe2cLMvKv8fDewcLYZI2JVRKyPiPWbNm0a8eEkSept7Js9MjOBnGP6msxcmplLp6amxn04SZI2M2qQ3RMRiwDK93vba0mSpMGNGmQXAivLzyuBC9ppR5Kk4Qxy+/0ngMuBZ0bEHRFxAnAacHhE3AQcVoYlSZp3C/rNkJkrZpl0aMu9SJI0ND/ZQ5JUNYNMklS1vpcWtXW5fON9rdQ5eFkrZSRpi/OMTJJUNYNMklQ1g0ySVDWDTJJUNYNMklQ1g0ySVDWDTJJUNYNMklQ1g0ySVDWDTJJUNYNMklQ1g0ySVDWDTJJUNT/9XlXyvwBImuEZmSSpagaZJKlqBpkkqWoGmSSpagaZJKlqBpkkqWoGmSSpagaZJKlqBpkkqWoGmSSpamN9RFVELAc+BGwDnJmZp7XS1Xxat7qdOstO3mzwA5fc2ErZvzh831bq9DOpfi8/6+2t1D34hPe1UqefifU7of2stdq96k5IG/taz9fFpLZDZdv38WjkM7KI2AY4A/hDYD9gRUTs11ZjkiQNYpxLiwcBN2fmxsz8JfBJ4Kh22pIkaTDjBNkewPc7hu8o4yRJmjeRmaMtGHEssDwz31SGXws8PzPf2jXfKmBVGXwm8N3R2+1rd+AH1rWuda07z7Un2TPAXpk5NcH6VRvnZo87gT07hheXcZvJzDXAmjEeZ2ARsT4zl1rXuta17nzWnmTP6m+cS4tXAvtExN4RsR3wauDCdtqSJGkwI5+RZebDEfFW4L9obr//SGZe11pnkiQNYKy/I8vMi4GLW+qlDZO6hGld61q3/rqTrD0vb5+ot5Fv9pAkaWvgR1RJkuqWmY+JL+BoIIFntV0H2Ae4CLgFuApYB7y0peU+C3y9Rx+vAb4NXAd8CzgT2KVP70+l+cP0mce7GNh3xO3Qs1b5uhi4CfgmcB6wsO06wAdp7oJ9wgi9LwTOATaWx7wcOKbNOjQfCHAZzZ+TXF2enx1arHUicE35+iVwbfn5tCF7vhK4DfgZ8BDwa+AXZfh+4O6ynWcea7tZnrMDyvRrgYdLjYeAn5daPwN+CNxa5ru0o2bn+P8ufe4LXFLq/LRj+Zla/wf8uEy/r/TwqrKdbirL3NfV/wOl35n1v7PUuRX4VUc/XyrTVgIbyjpdDfwbcE9Zp5+X9byuLDPz/frS3zV9xh9b1vPtwHfKuCuB13U8V7uXbfjmHs/jgTTHk+XD7tt0vXaANzDHvgQsB77R0ee5wJKOeguATXTte2X8e8vzMVP/XX1e+wcw937R95jZc78f56C/NX2Vjf9l4N1t1gG2B24EjuyYZ3/g9eMuB+xC80flNwBP7xi/vDyJe5ThbYA3As+co+8oO/WbO8YdALxkhG0wa62y0/5Rx/jfB/Zvsw7NlYLbga8Dy1rofS/gT9uqQ3MwuR04uGPascwS6OPWogmi3cddd+DUcoBY2jH9VODtw+xHwNmlx+5lz6YcwLv62mw8zWuj+/k/G3hfRw8PAO/tmL6MJrS6t9M/0oTFzGvpp8Df9NjOD3btWzfQvD6fVsY/lybEnl2Gp8t83b94TgMbeqzjb4wH3kxzM9zOZXhnYGXH9LfQHDP+p0e908u0tcPs2/R57XTvSzTHpJtm1ruMO7JzvWk+hvCrZd+JjvGnledt+zK8E3DqsPtR137R91jbc98f5sW9tX4BO9L8BrIv8N026wAndO5MLS/3RuCfgb8DTukY/+VeO2GfWocAl7W0PXvWKv1+bNJ1ynIX0/zGvGbI3g/tdWAYYRvMWgd4D/Ce+arVffAZdd0ZLMj67keMH2S/8fx31io93N/VZ8/tNLNcqXlB2Van9JjvwY7aF9Oc/V7UMf3jwBs6hqcZMLBmGw98j45fUHsscxnlo/6AxR3jg+aM67eB/+XRoOi7b/d77XTvS93rPUvNjwHH0ZwdvbCM24HmjHinOfoYaD/qGO57zJzt67HyHtlRwOcz80bgvoj4vRbr/A7Npa9JLLcC+ET5WtExfpBlu+1PcxbXhtlqDfsYo9aZ2S7nA0dExLZDPOYo227YOsNuhzZrjfo4g2qzn1EfY3/gJ0Mus4LmMtU6Nn8t9ZrvE8AU8LyOfaut/QaAiNiZ5iC/cZbpewKLMvMbNJfVj++Y/ELg1sy8heYy6BFD9Djsa2fOmhGxPXAY8B9sfpx6BvC9zOx+nmaMsh+N/Bw8VoJsBc21WMr3uXbksepExPkRsSEiPjPOchGxkOZ68FdKAD4UEfv3WO53I+KaiLglIo7vnv5YU/64/uXAZzPzx8AVwMvGqHdGRHwrIq4cs69W6rRda2t4nK3AjjSvpY00Z3IPRcT+3evfuW+V5dbTY9+KiN1ozmr2iYh2/q/PbzqeJsDgN48ZAx3P5lq/UV47EbFbOdbc2LHerwDWZebPgE8DR5f/fNK97BvKst8vIT22WY61PVUfZBGxK81p7JkRcRvw18BxERFt1KF5M/d5M/Nl5jHA64Fdx1muzPMU4Nay3DSP7rCPLJuZ12bmgcDngCfNsQrXAaOeiQ5aa9jHGKXOy2je77i2bJcXM9wvJt3b/USaSzLDfk7dXHVG2Q5t1Rr1cYap0VY/oz7GdTTvtwy6zAE0r6VTaN53mgZW9Fj/R/YtmktjL6H3a+4+mkD4IU1IDq0EyYMR8fRZZlkBvL7s4xcCz4mIfUpIvAr42zLtn4DlEbET/Z/fUV47m613Odas4dH1XgEcVupdBexGc7y7GVhS+iIzP1qWfYDmPf1R9qN+x8xZVR9kNNfqP56Ze2XmdGbuSXOJ4SUt1bkZeFFEHNkx7w4tLLeC5o6k6cycpnnSX12mrQbeFxGLO+afK8QAvgg8sXxIMwAR8ZyIGHY7zFqL5o3YF0bEER3jX9rrTHKMOiuAN3Vsl72BwyNiBwbzRWD7iHhLx7hBlx20zoeBlRHx/I7+X1nOsidda9Seh6nR1n40m3Poev5pblp4akcPTwCO6Zh+KbCqezsBT6a5UWM5sIjmDt/38+hrqfs196ayX72S5v2il5V96/3A6RHx7I75xz0+rgbOKJcZiYgdI+J1EbEvsGNm7tGxn68u/R0KfDsz9yzT9qI5EzqG/s/vKK+dfwDe1bXeO5R+d6Y5ji7pqHkizS8JPwXOAj5cLj/O/I/K7UqNUfajc5j7mDm7Ud5Y25q+aK6Jd9+i+mfAv7RVB3gWzaWGjTR34nwBOGzU5Wh+Y7yTjjuAynLfpPkPAtC8WXstzW29X6P5LWlRn3V4Gs3liltofrv5T2CfEbdrz1plnT5Pc6fT9TSXPua6/X6YOnvR/Ba8c1eNzwDHD9H7olLvVprbitcNs/wgdYCDaW7K+S7NHXD/yty3349ciwFv9hjgcU6lz80eg+xHjHmzRxnX/fxvoNzxVqZ/jeb2/M4e/rjHdjqD5iwgutb/F2W5dTSX8R7s3rdobkt/gOZuxw00t99fWeqvp7lMuW9X39MMfrNHAO8o9TbQ3OL/Gpqbu7pvZX9OWaeP0nU7Ps1dhJ+b6/mlOeD3fe302pdo3oObWe+v0rwXti/NMeiTXfPuSnMr/hOBbWnuXLy5rNvXgHcB2w2zH/XYL3oea+f68pM9JElVeyxcWpQkPY4ZZJKkqhlkkqSqGWSSpKoZZJKkqhlkkqSqGWSSpKoZZJKkqv0/FMB6DerNxqQAAAAASUVORK5CYII=\n",
      "text/plain": [
       "<matplotlib.figure.Figure at 0x116a52eb8>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "hist_compare(datatable_pd,'ALT')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>count</th>\n",
       "      <th>mean</th>\n",
       "      <th>std</th>\n",
       "      <th>min</th>\n",
       "      <th>25%</th>\n",
       "      <th>50%</th>\n",
       "      <th>75%</th>\n",
       "      <th>max</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>tOverlapMotifs</th>\n",
       "      <td>7.0</td>\n",
       "      <td>4.00</td>\n",
       "      <td>2.24</td>\n",
       "      <td>1.00</td>\n",
       "      <td>2.50</td>\n",
       "      <td>4.00</td>\n",
       "      <td>6.00</td>\n",
       "      <td>6.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>motifDist</th>\n",
       "      <td>7.0</td>\n",
       "      <td>-0.13</td>\n",
       "      <td>0.65</td>\n",
       "      <td>-1.32</td>\n",
       "      <td>-0.39</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.32</td>\n",
       "      <td>0.53</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>mirSVR-Aln</th>\n",
       "      <td>18.0</td>\n",
       "      <td>141.44</td>\n",
       "      <td>15.37</td>\n",
       "      <td>120.00</td>\n",
       "      <td>123.25</td>\n",
       "      <td>147.00</td>\n",
       "      <td>154.25</td>\n",
       "      <td>161.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>mirSVR-E</th>\n",
       "      <td>18.0</td>\n",
       "      <td>-12.33</td>\n",
       "      <td>5.04</td>\n",
       "      <td>-24.38</td>\n",
       "      <td>-15.14</td>\n",
       "      <td>-11.92</td>\n",
       "      <td>-9.63</td>\n",
       "      <td>-4.41</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>mirSVR-Score</th>\n",
       "      <td>18.0</td>\n",
       "      <td>-0.27</td>\n",
       "      <td>0.31</td>\n",
       "      <td>-1.09</td>\n",
       "      <td>-0.31</td>\n",
       "      <td>-0.26</td>\n",
       "      <td>-0.04</td>\n",
       "      <td>-0.01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>scoreSegDup</th>\n",
       "      <td>19.0</td>\n",
       "      <td>0.93</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.93</td>\n",
       "      <td>0.93</td>\n",
       "      <td>0.93</td>\n",
       "      <td>0.93</td>\n",
       "      <td>0.93</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TFBS</th>\n",
       "      <td>21.0</td>\n",
       "      <td>3.43</td>\n",
       "      <td>2.93</td>\n",
       "      <td>1.00</td>\n",
       "      <td>1.00</td>\n",
       "      <td>3.00</td>\n",
       "      <td>3.00</td>\n",
       "      <td>11.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TFBSPeaks</th>\n",
       "      <td>21.0</td>\n",
       "      <td>4.05</td>\n",
       "      <td>3.47</td>\n",
       "      <td>1.00</td>\n",
       "      <td>1.00</td>\n",
       "      <td>3.00</td>\n",
       "      <td>5.00</td>\n",
       "      <td>11.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TFBSPeaksMax</th>\n",
       "      <td>21.0</td>\n",
       "      <td>57.83</td>\n",
       "      <td>27.67</td>\n",
       "      <td>25.16</td>\n",
       "      <td>36.81</td>\n",
       "      <td>45.93</td>\n",
       "      <td>63.90</td>\n",
       "      <td>105.54</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>EncOCpolIISig</th>\n",
       "      <td>28.0</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.02</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>EncOCpolIIPVal</th>\n",
       "      <td>28.0</td>\n",
       "      <td>0.40</td>\n",
       "      <td>1.04</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>3.26</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>EncOCmycSig</th>\n",
       "      <td>28.0</td>\n",
       "      <td>0.06</td>\n",
       "      <td>0.06</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.06</td>\n",
       "      <td>0.14</td>\n",
       "      <td>0.14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>EncOCmycPVal</th>\n",
       "      <td>28.0</td>\n",
       "      <td>1.11</td>\n",
       "      <td>1.19</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.74</td>\n",
       "      <td>2.28</td>\n",
       "      <td>3.46</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>EncOCctcfSig</th>\n",
       "      <td>28.0</td>\n",
       "      <td>0.26</td>\n",
       "      <td>0.52</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.16</td>\n",
       "      <td>0.16</td>\n",
       "      <td>1.71</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>EncOCctcfPVal</th>\n",
       "      <td>28.0</td>\n",
       "      <td>2.22</td>\n",
       "      <td>4.05</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>1.62</td>\n",
       "      <td>1.65</td>\n",
       "      <td>13.40</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>EncOCFairePVal</th>\n",
       "      <td>28.0</td>\n",
       "      <td>0.73</td>\n",
       "      <td>0.80</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.86</td>\n",
       "      <td>1.18</td>\n",
       "      <td>3.17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>EncOCFaireSig</th>\n",
       "      <td>28.0</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.02</td>\n",
       "      <td>0.03</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>EncOCDNasePVal</th>\n",
       "      <td>28.0</td>\n",
       "      <td>2.63</td>\n",
       "      <td>5.27</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>1.01</td>\n",
       "      <td>1.02</td>\n",
       "      <td>16.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>EncOCCombPVal</th>\n",
       "      <td>28.0</td>\n",
       "      <td>2.66</td>\n",
       "      <td>5.15</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.49</td>\n",
       "      <td>0.57</td>\n",
       "      <td>0.96</td>\n",
       "      <td>15.26</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>EncOCC</th>\n",
       "      <td>28.0</td>\n",
       "      <td>3.21</td>\n",
       "      <td>0.88</td>\n",
       "      <td>1.00</td>\n",
       "      <td>3.00</td>\n",
       "      <td>3.00</td>\n",
       "      <td>4.00</td>\n",
       "      <td>4.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>EncOCDNaseSig</th>\n",
       "      <td>28.0</td>\n",
       "      <td>0.11</td>\n",
       "      <td>0.23</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.03</td>\n",
       "      <td>0.03</td>\n",
       "      <td>0.75</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TG_ASN</th>\n",
       "      <td>29.0</td>\n",
       "      <td>0.25</td>\n",
       "      <td>0.21</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.07</td>\n",
       "      <td>0.26</td>\n",
       "      <td>0.32</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ESP_AF</th>\n",
       "      <td>34.0</td>\n",
       "      <td>0.11</td>\n",
       "      <td>0.20</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.02</td>\n",
       "      <td>0.08</td>\n",
       "      <td>0.98</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ESP_AFR</th>\n",
       "      <td>34.0</td>\n",
       "      <td>0.11</td>\n",
       "      <td>0.19</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.03</td>\n",
       "      <td>0.07</td>\n",
       "      <td>0.93</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TG_EUR</th>\n",
       "      <td>34.0</td>\n",
       "      <td>0.16</td>\n",
       "      <td>0.22</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.03</td>\n",
       "      <td>0.06</td>\n",
       "      <td>0.22</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ESP_EUR</th>\n",
       "      <td>34.0</td>\n",
       "      <td>0.11</td>\n",
       "      <td>0.21</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.02</td>\n",
       "      <td>0.06</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TG_AFR</th>\n",
       "      <td>38.0</td>\n",
       "      <td>0.15</td>\n",
       "      <td>0.19</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.03</td>\n",
       "      <td>0.07</td>\n",
       "      <td>0.21</td>\n",
       "      <td>0.88</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TG_AMR</th>\n",
       "      <td>41.0</td>\n",
       "      <td>0.16</td>\n",
       "      <td>0.20</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.07</td>\n",
       "      <td>0.23</td>\n",
       "      <td>0.99</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TG_AF</th>\n",
       "      <td>43.0</td>\n",
       "      <td>0.14</td>\n",
       "      <td>0.19</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.02</td>\n",
       "      <td>0.08</td>\n",
       "      <td>0.18</td>\n",
       "      <td>0.97</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Dst2Splice</th>\n",
       "      <td>88.0</td>\n",
       "      <td>3.48</td>\n",
       "      <td>8.55</td>\n",
       "      <td>-15.00</td>\n",
       "      <td>-2.00</td>\n",
       "      <td>2.50</td>\n",
       "      <td>10.00</td>\n",
       "      <td>20.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Grantham</th>\n",
       "      <td>107.0</td>\n",
       "      <td>75.22</td>\n",
       "      <td>47.71</td>\n",
       "      <td>0.00</td>\n",
       "      <td>29.00</td>\n",
       "      <td>77.00</td>\n",
       "      <td>101.00</td>\n",
       "      <td>215.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PolyPhenVal</th>\n",
       "      <td>107.0</td>\n",
       "      <td>0.47</td>\n",
       "      <td>0.41</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.02</td>\n",
       "      <td>0.46</td>\n",
       "      <td>0.94</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SIFTval</th>\n",
       "      <td>107.0</td>\n",
       "      <td>0.20</td>\n",
       "      <td>0.27</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.07</td>\n",
       "      <td>0.32</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>relProtPos</th>\n",
       "      <td>190.0</td>\n",
       "      <td>0.35</td>\n",
       "      <td>0.20</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.20</td>\n",
       "      <td>0.34</td>\n",
       "      <td>0.49</td>\n",
       "      <td>0.98</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>relCDSpos</th>\n",
       "      <td>190.0</td>\n",
       "      <td>0.35</td>\n",
       "      <td>0.20</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.20</td>\n",
       "      <td>0.34</td>\n",
       "      <td>0.48</td>\n",
       "      <td>0.98</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>protPos</th>\n",
       "      <td>190.0</td>\n",
       "      <td>653.15</td>\n",
       "      <td>374.26</td>\n",
       "      <td>1.00</td>\n",
       "      <td>373.50</td>\n",
       "      <td>620.00</td>\n",
       "      <td>895.50</td>\n",
       "      <td>1811.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CDSpos</th>\n",
       "      <td>190.0</td>\n",
       "      <td>1958.25</td>\n",
       "      <td>1122.79</td>\n",
       "      <td>1.00</td>\n",
       "      <td>1118.75</td>\n",
       "      <td>1859.00</td>\n",
       "      <td>2684.50</td>\n",
       "      <td>5433.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>relcDNApos</th>\n",
       "      <td>208.0</td>\n",
       "      <td>0.28</td>\n",
       "      <td>0.18</td>\n",
       "      <td>0.03</td>\n",
       "      <td>0.16</td>\n",
       "      <td>0.24</td>\n",
       "      <td>0.34</td>\n",
       "      <td>0.96</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cDNApos</th>\n",
       "      <td>208.0</td>\n",
       "      <td>2675.77</td>\n",
       "      <td>1753.61</td>\n",
       "      <td>301.00</td>\n",
       "      <td>1489.75</td>\n",
       "      <td>2273.50</td>\n",
       "      <td>3219.75</td>\n",
       "      <td>9120.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GerpRSpval</th>\n",
       "      <td>216.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GerpRS</th>\n",
       "      <td>216.0</td>\n",
       "      <td>549.74</td>\n",
       "      <td>223.38</td>\n",
       "      <td>21.10</td>\n",
       "      <td>386.20</td>\n",
       "      <td>470.30</td>\n",
       "      <td>681.10</td>\n",
       "      <td>1012.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>EncExp</th>\n",
       "      <td>240.0</td>\n",
       "      <td>32.25</td>\n",
       "      <td>18.71</td>\n",
       "      <td>0.64</td>\n",
       "      <td>20.83</td>\n",
       "      <td>29.90</td>\n",
       "      <td>41.35</td>\n",
       "      <td>142.50</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PHRED</th>\n",
       "      <td>248.0</td>\n",
       "      <td>21.37</td>\n",
       "      <td>12.65</td>\n",
       "      <td>0.00</td>\n",
       "      <td>9.50</td>\n",
       "      <td>23.60</td>\n",
       "      <td>32.00</td>\n",
       "      <td>52.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>mamPhCons</th>\n",
       "      <td>248.0</td>\n",
       "      <td>0.75</td>\n",
       "      <td>0.39</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.56</td>\n",
       "      <td>0.99</td>\n",
       "      <td>1.00</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>mamPhyloP</th>\n",
       "      <td>248.0</td>\n",
       "      <td>1.39</td>\n",
       "      <td>1.31</td>\n",
       "      <td>-3.10</td>\n",
       "      <td>0.38</td>\n",
       "      <td>1.82</td>\n",
       "      <td>2.53</td>\n",
       "      <td>2.94</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>mapAbility20bp</th>\n",
       "      <td>248.0</td>\n",
       "      <td>0.95</td>\n",
       "      <td>0.19</td>\n",
       "      <td>0.00</td>\n",
       "      <td>1.00</td>\n",
       "      <td>1.00</td>\n",
       "      <td>1.00</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>mapAbility35bp</th>\n",
       "      <td>248.0</td>\n",
       "      <td>0.99</td>\n",
       "      <td>0.09</td>\n",
       "      <td>0.25</td>\n",
       "      <td>1.00</td>\n",
       "      <td>1.00</td>\n",
       "      <td>1.00</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>minDistTSE</th>\n",
       "      <td>248.0</td>\n",
       "      <td>97240.96</td>\n",
       "      <td>67571.23</td>\n",
       "      <td>10.00</td>\n",
       "      <td>68717.75</td>\n",
       "      <td>106788.00</td>\n",
       "      <td>151562.75</td>\n",
       "      <td>372078.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>minDistTSS</th>\n",
       "      <td>248.0</td>\n",
       "      <td>143594.91</td>\n",
       "      <td>115304.28</td>\n",
       "      <td>4.00</td>\n",
       "      <td>7757.25</td>\n",
       "      <td>210288.00</td>\n",
       "      <td>240729.50</td>\n",
       "      <td>288672.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GerpN</th>\n",
       "      <td>248.0</td>\n",
       "      <td>5.17</td>\n",
       "      <td>0.88</td>\n",
       "      <td>0.80</td>\n",
       "      <td>5.13</td>\n",
       "      <td>5.40</td>\n",
       "      <td>5.58</td>\n",
       "      <td>6.17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GC</th>\n",
       "      <td>248.0</td>\n",
       "      <td>0.51</td>\n",
       "      <td>0.09</td>\n",
       "      <td>0.28</td>\n",
       "      <td>0.45</td>\n",
       "      <td>0.52</td>\n",
       "      <td>0.58</td>\n",
       "      <td>0.83</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>mutIndex</th>\n",
       "      <td>248.0</td>\n",
       "      <td>18.57</td>\n",
       "      <td>44.02</td>\n",
       "      <td>-85.00</td>\n",
       "      <td>-8.00</td>\n",
       "      <td>16.00</td>\n",
       "      <td>36.25</td>\n",
       "      <td>121.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>EncH3K4Me3</th>\n",
       "      <td>248.0</td>\n",
       "      <td>3.16</td>\n",
       "      <td>3.27</td>\n",
       "      <td>0.80</td>\n",
       "      <td>2.00</td>\n",
       "      <td>2.86</td>\n",
       "      <td>3.12</td>\n",
       "      <td>29.16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>fitCons</th>\n",
       "      <td>248.0</td>\n",
       "      <td>0.48</td>\n",
       "      <td>0.22</td>\n",
       "      <td>0.05</td>\n",
       "      <td>0.50</td>\n",
       "      <td>0.58</td>\n",
       "      <td>0.63</td>\n",
       "      <td>0.68</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>priPhCons</th>\n",
       "      <td>248.0</td>\n",
       "      <td>0.65</td>\n",
       "      <td>0.35</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.31</td>\n",
       "      <td>0.78</td>\n",
       "      <td>0.97</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>priPhyloP</th>\n",
       "      <td>248.0</td>\n",
       "      <td>0.33</td>\n",
       "      <td>0.53</td>\n",
       "      <td>-2.95</td>\n",
       "      <td>0.39</td>\n",
       "      <td>0.53</td>\n",
       "      <td>0.64</td>\n",
       "      <td>0.65</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>EncH3K4Me1</th>\n",
       "      <td>248.0</td>\n",
       "      <td>4.53</td>\n",
       "      <td>4.66</td>\n",
       "      <td>0.48</td>\n",
       "      <td>2.00</td>\n",
       "      <td>3.00</td>\n",
       "      <td>4.76</td>\n",
       "      <td>36.72</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>EncH3K27Ac</th>\n",
       "      <td>248.0</td>\n",
       "      <td>3.88</td>\n",
       "      <td>5.75</td>\n",
       "      <td>1.00</td>\n",
       "      <td>2.00</td>\n",
       "      <td>3.00</td>\n",
       "      <td>4.73</td>\n",
       "      <td>86.80</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CpG</th>\n",
       "      <td>248.0</td>\n",
       "      <td>0.05</td>\n",
       "      <td>0.04</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.03</td>\n",
       "      <td>0.04</td>\n",
       "      <td>0.07</td>\n",
       "      <td>0.29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ConsScore</th>\n",
       "      <td>248.0</td>\n",
       "      <td>6.18</td>\n",
       "      <td>1.74</td>\n",
       "      <td>2.00</td>\n",
       "      <td>5.00</td>\n",
       "      <td>7.00</td>\n",
       "      <td>7.00</td>\n",
       "      <td>8.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>EncNucleo</th>\n",
       "      <td>248.0</td>\n",
       "      <td>1.60</td>\n",
       "      <td>0.83</td>\n",
       "      <td>0.00</td>\n",
       "      <td>1.00</td>\n",
       "      <td>1.60</td>\n",
       "      <td>2.10</td>\n",
       "      <td>4.10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>dnaRoll</th>\n",
       "      <td>248.0</td>\n",
       "      <td>0.82</td>\n",
       "      <td>3.01</td>\n",
       "      <td>-7.91</td>\n",
       "      <td>-0.86</td>\n",
       "      <td>0.32</td>\n",
       "      <td>1.48</td>\n",
       "      <td>11.24</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>dnaMGW</th>\n",
       "      <td>248.0</td>\n",
       "      <td>0.18</td>\n",
       "      <td>0.40</td>\n",
       "      <td>-1.19</td>\n",
       "      <td>-0.03</td>\n",
       "      <td>0.16</td>\n",
       "      <td>0.35</td>\n",
       "      <td>1.83</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>RawScore</th>\n",
       "      <td>248.0</td>\n",
       "      <td>4.50</td>\n",
       "      <td>4.12</td>\n",
       "      <td>-2.26</td>\n",
       "      <td>0.81</td>\n",
       "      <td>3.99</td>\n",
       "      <td>6.72</td>\n",
       "      <td>15.70</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Length</th>\n",
       "      <td>248.0</td>\n",
       "      <td>0.31</td>\n",
       "      <td>1.78</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>26.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GerpS</th>\n",
       "      <td>248.0</td>\n",
       "      <td>3.14</td>\n",
       "      <td>3.41</td>\n",
       "      <td>-10.30</td>\n",
       "      <td>2.13</td>\n",
       "      <td>4.58</td>\n",
       "      <td>5.50</td>\n",
       "      <td>6.17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>verPhCons</th>\n",
       "      <td>248.0</td>\n",
       "      <td>0.79</td>\n",
       "      <td>0.38</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.86</td>\n",
       "      <td>1.00</td>\n",
       "      <td>1.00</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>bStatistic</th>\n",
       "      <td>248.0</td>\n",
       "      <td>766.54</td>\n",
       "      <td>86.24</td>\n",
       "      <td>614.00</td>\n",
       "      <td>697.00</td>\n",
       "      <td>832.00</td>\n",
       "      <td>844.00</td>\n",
       "      <td>866.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cHmmBivFlnk</th>\n",
       "      <td>248.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.02</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cHmmEnh</th>\n",
       "      <td>248.0</td>\n",
       "      <td>0.02</td>\n",
       "      <td>0.03</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.02</td>\n",
       "      <td>0.04</td>\n",
       "      <td>0.21</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cHmmEnhBiv</th>\n",
       "      <td>248.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cHmmEnhG</th>\n",
       "      <td>248.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.03</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cHmmHet</th>\n",
       "      <td>248.0</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.02</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>dnaProT</th>\n",
       "      <td>248.0</td>\n",
       "      <td>0.42</td>\n",
       "      <td>2.80</td>\n",
       "      <td>-5.32</td>\n",
       "      <td>-2.19</td>\n",
       "      <td>1.40</td>\n",
       "      <td>2.53</td>\n",
       "      <td>8.96</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cHmmQuies</th>\n",
       "      <td>248.0</td>\n",
       "      <td>0.45</td>\n",
       "      <td>0.08</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.42</td>\n",
       "      <td>0.47</td>\n",
       "      <td>0.50</td>\n",
       "      <td>0.63</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cHmmReprPCWk</th>\n",
       "      <td>248.0</td>\n",
       "      <td>0.27</td>\n",
       "      <td>0.10</td>\n",
       "      <td>0.02</td>\n",
       "      <td>0.25</td>\n",
       "      <td>0.30</td>\n",
       "      <td>0.34</td>\n",
       "      <td>0.39</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cHmmTssA</th>\n",
       "      <td>248.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.02</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cHmmTssAFlnk</th>\n",
       "      <td>248.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cHmmTssBiv</th>\n",
       "      <td>248.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.04</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.45</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cHmmTx</th>\n",
       "      <td>248.0</td>\n",
       "      <td>0.06</td>\n",
       "      <td>0.04</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.03</td>\n",
       "      <td>0.06</td>\n",
       "      <td>0.08</td>\n",
       "      <td>0.15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cHmmTxFlnk</th>\n",
       "      <td>248.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cHmmTxWk</th>\n",
       "      <td>248.0</td>\n",
       "      <td>0.14</td>\n",
       "      <td>0.06</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.10</td>\n",
       "      <td>0.13</td>\n",
       "      <td>0.16</td>\n",
       "      <td>0.36</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cHmmZnfRpts</th>\n",
       "      <td>248.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.02</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>dnaHelT</th>\n",
       "      <td>248.0</td>\n",
       "      <td>-0.56</td>\n",
       "      <td>2.19</td>\n",
       "      <td>-5.01</td>\n",
       "      <td>-2.32</td>\n",
       "      <td>-0.59</td>\n",
       "      <td>1.19</td>\n",
       "      <td>4.24</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cHmmReprPC</th>\n",
       "      <td>248.0</td>\n",
       "      <td>0.03</td>\n",
       "      <td>0.03</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.02</td>\n",
       "      <td>0.03</td>\n",
       "      <td>0.14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>verPhyloP</th>\n",
       "      <td>248.0</td>\n",
       "      <td>2.64</td>\n",
       "      <td>2.38</td>\n",
       "      <td>-4.08</td>\n",
       "      <td>0.56</td>\n",
       "      <td>2.43</td>\n",
       "      <td>4.98</td>\n",
       "      <td>6.33</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                count       mean        std     min       25%        50%  \\\n",
       "tOverlapMotifs    7.0       4.00       2.24    1.00      2.50       4.00   \n",
       "motifDist         7.0      -0.13       0.65   -1.32     -0.39       0.00   \n",
       "mirSVR-Aln       18.0     141.44      15.37  120.00    123.25     147.00   \n",
       "mirSVR-E         18.0     -12.33       5.04  -24.38    -15.14     -11.92   \n",
       "mirSVR-Score     18.0      -0.27       0.31   -1.09     -0.31      -0.26   \n",
       "scoreSegDup      19.0       0.93       0.00    0.93      0.93       0.93   \n",
       "TFBS             21.0       3.43       2.93    1.00      1.00       3.00   \n",
       "TFBSPeaks        21.0       4.05       3.47    1.00      1.00       3.00   \n",
       "TFBSPeaksMax     21.0      57.83      27.67   25.16     36.81      45.93   \n",
       "EncOCpolIISig    28.0       0.01       0.02    0.00      0.00       0.00   \n",
       "EncOCpolIIPVal   28.0       0.40       1.04    0.00      0.00       0.00   \n",
       "EncOCmycSig      28.0       0.06       0.06    0.00      0.00       0.06   \n",
       "EncOCmycPVal     28.0       1.11       1.19    0.00      0.00       0.74   \n",
       "EncOCctcfSig     28.0       0.26       0.52    0.00      0.00       0.16   \n",
       "EncOCctcfPVal    28.0       2.22       4.05    0.00      0.00       1.62   \n",
       "EncOCFairePVal   28.0       0.73       0.80    0.00      0.00       0.86   \n",
       "EncOCFaireSig    28.0       0.01       0.01    0.00      0.01       0.01   \n",
       "EncOCDNasePVal   28.0       2.63       5.27    0.00      0.00       1.01   \n",
       "EncOCCombPVal    28.0       2.66       5.15    0.00      0.49       0.57   \n",
       "EncOCC           28.0       3.21       0.88    1.00      3.00       3.00   \n",
       "EncOCDNaseSig    28.0       0.11       0.23    0.00      0.01       0.03   \n",
       "TG_ASN           29.0       0.25       0.21    0.00      0.07       0.26   \n",
       "ESP_AF           34.0       0.11       0.20    0.00      0.00       0.02   \n",
       "ESP_AFR          34.0       0.11       0.19    0.00      0.00       0.03   \n",
       "TG_EUR           34.0       0.16       0.22    0.00      0.03       0.06   \n",
       "ESP_EUR          34.0       0.11       0.21    0.00      0.00       0.02   \n",
       "TG_AFR           38.0       0.15       0.19    0.00      0.03       0.07   \n",
       "TG_AMR           41.0       0.16       0.20    0.00      0.01       0.07   \n",
       "TG_AF            43.0       0.14       0.19    0.00      0.02       0.08   \n",
       "Dst2Splice       88.0       3.48       8.55  -15.00     -2.00       2.50   \n",
       "Grantham        107.0      75.22      47.71    0.00     29.00      77.00   \n",
       "PolyPhenVal     107.0       0.47       0.41    0.00      0.02       0.46   \n",
       "SIFTval         107.0       0.20       0.27    0.00      0.00       0.07   \n",
       "relProtPos      190.0       0.35       0.20    0.00      0.20       0.34   \n",
       "relCDSpos       190.0       0.35       0.20    0.00      0.20       0.34   \n",
       "protPos         190.0     653.15     374.26    1.00    373.50     620.00   \n",
       "CDSpos          190.0    1958.25    1122.79    1.00   1118.75    1859.00   \n",
       "relcDNApos      208.0       0.28       0.18    0.03      0.16       0.24   \n",
       "cDNApos         208.0    2675.77    1753.61  301.00   1489.75    2273.50   \n",
       "GerpRSpval      216.0       0.00       0.00    0.00      0.00       0.00   \n",
       "GerpRS          216.0     549.74     223.38   21.10    386.20     470.30   \n",
       "EncExp          240.0      32.25      18.71    0.64     20.83      29.90   \n",
       "PHRED           248.0      21.37      12.65    0.00      9.50      23.60   \n",
       "mamPhCons       248.0       0.75       0.39    0.00      0.56       0.99   \n",
       "mamPhyloP       248.0       1.39       1.31   -3.10      0.38       1.82   \n",
       "mapAbility20bp  248.0       0.95       0.19    0.00      1.00       1.00   \n",
       "mapAbility35bp  248.0       0.99       0.09    0.25      1.00       1.00   \n",
       "minDistTSE      248.0   97240.96   67571.23   10.00  68717.75  106788.00   \n",
       "minDistTSS      248.0  143594.91  115304.28    4.00   7757.25  210288.00   \n",
       "GerpN           248.0       5.17       0.88    0.80      5.13       5.40   \n",
       "GC              248.0       0.51       0.09    0.28      0.45       0.52   \n",
       "mutIndex        248.0      18.57      44.02  -85.00     -8.00      16.00   \n",
       "EncH3K4Me3      248.0       3.16       3.27    0.80      2.00       2.86   \n",
       "fitCons         248.0       0.48       0.22    0.05      0.50       0.58   \n",
       "priPhCons       248.0       0.65       0.35    0.00      0.31       0.78   \n",
       "priPhyloP       248.0       0.33       0.53   -2.95      0.39       0.53   \n",
       "EncH3K4Me1      248.0       4.53       4.66    0.48      2.00       3.00   \n",
       "EncH3K27Ac      248.0       3.88       5.75    1.00      2.00       3.00   \n",
       "CpG             248.0       0.05       0.04    0.00      0.03       0.04   \n",
       "ConsScore       248.0       6.18       1.74    2.00      5.00       7.00   \n",
       "EncNucleo       248.0       1.60       0.83    0.00      1.00       1.60   \n",
       "dnaRoll         248.0       0.82       3.01   -7.91     -0.86       0.32   \n",
       "dnaMGW          248.0       0.18       0.40   -1.19     -0.03       0.16   \n",
       "RawScore        248.0       4.50       4.12   -2.26      0.81       3.99   \n",
       "Length          248.0       0.31       1.78    0.00      0.00       0.00   \n",
       "GerpS           248.0       3.14       3.41  -10.30      2.13       4.58   \n",
       "verPhCons       248.0       0.79       0.38    0.00      0.86       1.00   \n",
       "bStatistic      248.0     766.54      86.24  614.00    697.00     832.00   \n",
       "cHmmBivFlnk     248.0       0.00       0.02    0.00      0.00       0.00   \n",
       "cHmmEnh         248.0       0.02       0.03    0.00      0.00       0.02   \n",
       "cHmmEnhBiv      248.0       0.00       0.01    0.00      0.00       0.00   \n",
       "cHmmEnhG        248.0       0.00       0.00    0.00      0.00       0.00   \n",
       "cHmmHet         248.0       0.01       0.02    0.00      0.00       0.00   \n",
       "dnaProT         248.0       0.42       2.80   -5.32     -2.19       1.40   \n",
       "cHmmQuies       248.0       0.45       0.08    0.01      0.42       0.47   \n",
       "cHmmReprPCWk    248.0       0.27       0.10    0.02      0.25       0.30   \n",
       "cHmmTssA        248.0       0.00       0.02    0.00      0.00       0.00   \n",
       "cHmmTssAFlnk    248.0       0.00       0.01    0.00      0.00       0.00   \n",
       "cHmmTssBiv      248.0       0.00       0.04    0.00      0.00       0.00   \n",
       "cHmmTx          248.0       0.06       0.04    0.00      0.03       0.06   \n",
       "cHmmTxFlnk      248.0       0.00       0.00    0.00      0.00       0.00   \n",
       "cHmmTxWk        248.0       0.14       0.06    0.00      0.10       0.13   \n",
       "cHmmZnfRpts     248.0       0.00       0.00    0.00      0.00       0.00   \n",
       "dnaHelT         248.0      -0.56       2.19   -5.01     -2.32      -0.59   \n",
       "cHmmReprPC      248.0       0.03       0.03    0.00      0.01       0.02   \n",
       "verPhyloP       248.0       2.64       2.38   -4.08      0.56       2.43   \n",
       "\n",
       "                      75%        max  \n",
       "tOverlapMotifs       6.00       6.00  \n",
       "motifDist            0.32       0.53  \n",
       "mirSVR-Aln         154.25     161.00  \n",
       "mirSVR-E            -9.63      -4.41  \n",
       "mirSVR-Score        -0.04      -0.01  \n",
       "scoreSegDup          0.93       0.93  \n",
       "TFBS                 3.00      11.00  \n",
       "TFBSPeaks            5.00      11.00  \n",
       "TFBSPeaksMax        63.90     105.54  \n",
       "EncOCpolIISig        0.01       0.06  \n",
       "EncOCpolIIPVal       0.00       3.26  \n",
       "EncOCmycSig          0.14       0.14  \n",
       "EncOCmycPVal         2.28       3.46  \n",
       "EncOCctcfSig         0.16       1.71  \n",
       "EncOCctcfPVal        1.65      13.40  \n",
       "EncOCFairePVal       1.18       3.17  \n",
       "EncOCFaireSig        0.02       0.03  \n",
       "EncOCDNasePVal       1.02      16.00  \n",
       "EncOCCombPVal        0.96      15.26  \n",
       "EncOCC               4.00       4.00  \n",
       "EncOCDNaseSig        0.03       0.75  \n",
       "TG_ASN               0.32       1.00  \n",
       "ESP_AF               0.08       0.98  \n",
       "ESP_AFR              0.07       0.93  \n",
       "TG_EUR               0.22       1.00  \n",
       "ESP_EUR              0.06       1.00  \n",
       "TG_AFR               0.21       0.88  \n",
       "TG_AMR               0.23       0.99  \n",
       "TG_AF                0.18       0.97  \n",
       "Dst2Splice          10.00      20.00  \n",
       "Grantham           101.00     215.00  \n",
       "PolyPhenVal          0.94       1.00  \n",
       "SIFTval              0.32       1.00  \n",
       "relProtPos           0.49       0.98  \n",
       "relCDSpos            0.48       0.98  \n",
       "protPos            895.50    1811.00  \n",
       "CDSpos            2684.50    5433.00  \n",
       "relcDNApos           0.34       0.96  \n",
       "cDNApos           3219.75    9120.00  \n",
       "GerpRSpval           0.00       0.00  \n",
       "GerpRS             681.10    1012.00  \n",
       "EncExp              41.35     142.50  \n",
       "PHRED               32.00      52.00  \n",
       "mamPhCons            1.00       1.00  \n",
       "mamPhyloP            2.53       2.94  \n",
       "mapAbility20bp       1.00       1.00  \n",
       "mapAbility35bp       1.00       1.00  \n",
       "minDistTSE      151562.75  372078.00  \n",
       "minDistTSS      240729.50  288672.00  \n",
       "GerpN                5.58       6.17  \n",
       "GC                   0.58       0.83  \n",
       "mutIndex            36.25     121.00  \n",
       "EncH3K4Me3           3.12      29.16  \n",
       "fitCons              0.63       0.68  \n",
       "priPhCons            0.97       1.00  \n",
       "priPhyloP            0.64       0.65  \n",
       "EncH3K4Me1           4.76      36.72  \n",
       "EncH3K27Ac           4.73      86.80  \n",
       "CpG                  0.07       0.29  \n",
       "ConsScore            7.00       8.00  \n",
       "EncNucleo            2.10       4.10  \n",
       "dnaRoll              1.48      11.24  \n",
       "dnaMGW               0.35       1.83  \n",
       "RawScore             6.72      15.70  \n",
       "Length               0.00      26.00  \n",
       "GerpS                5.50       6.17  \n",
       "verPhCons            1.00       1.00  \n",
       "bStatistic         844.00     866.00  \n",
       "cHmmBivFlnk          0.00       0.20  \n",
       "cHmmEnh              0.04       0.21  \n",
       "cHmmEnhBiv           0.00       0.05  \n",
       "cHmmEnhG             0.00       0.03  \n",
       "cHmmHet              0.01       0.20  \n",
       "dnaProT              2.53       8.96  \n",
       "cHmmQuies            0.50       0.63  \n",
       "cHmmReprPCWk         0.34       0.39  \n",
       "cHmmTssA             0.00       0.19  \n",
       "cHmmTssAFlnk         0.00       0.11  \n",
       "cHmmTssBiv           0.00       0.45  \n",
       "cHmmTx               0.08       0.15  \n",
       "cHmmTxFlnk           0.00       0.01  \n",
       "cHmmTxWk             0.16       0.36  \n",
       "cHmmZnfRpts          0.00       0.02  \n",
       "dnaHelT              1.19       4.24  \n",
       "cHmmReprPC           0.03       0.14  \n",
       "verPhyloP            4.98       6.33  "
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# display the description for the numerical columns\n",
    "datatable_num = datatable_pd.select_dtypes(exclude=[np.object])\n",
    "# exclude the CHROM, POS\n",
    "des_num = datatable_num[datatable_num.columns.difference(['CHROM', 'POS'])].describe().T\n",
    "# drop features without any available samples\n",
    "des_num = des_num.drop(des_num[des_num['count'] < 1].index)\n",
    "des_num.sort_values(by='count').round(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>count</th>\n",
       "      <th>unique</th>\n",
       "      <th>top</th>\n",
       "      <th>freq</th>\n",
       "      <th>nr_unique</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ID</th>\n",
       "      <td>248</td>\n",
       "      <td>[MYO5B:c.5616-2A&gt;G, MYO5B:c.5392C&gt;T, MYO5B:c.5...</td>\n",
       "      <td>PopulationGAVINr0.4</td>\n",
       "      <td>146</td>\n",
       "      <td>103</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>REF</th>\n",
       "      <td>248</td>\n",
       "      <td>[T, G, C, AA, A, CGGG, TC, AG, TCTGA, CA, TA, ...</td>\n",
       "      <td>C</td>\n",
       "      <td>75</td>\n",
       "      <td>20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ALT</th>\n",
       "      <td>248</td>\n",
       "      <td>[C, A, CCA, GC, G, AGAG, CC, T, TC, TGAG, TTCT...</td>\n",
       "      <td>A</td>\n",
       "      <td>77</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>INFO</th>\n",
       "      <td>248</td>\n",
       "      <td>[Pathogenic, Benign]</td>\n",
       "      <td>Benign</td>\n",
       "      <td>195</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Type</th>\n",
       "      <td>248</td>\n",
       "      <td>[SNV, INS, DEL]</td>\n",
       "      <td>SNV</td>\n",
       "      <td>219</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>isTv</th>\n",
       "      <td>218</td>\n",
       "      <td>[False, nan, True]</td>\n",
       "      <td>False</td>\n",
       "      <td>134</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AnnoType</th>\n",
       "      <td>248</td>\n",
       "      <td>[Transcript, CodingTranscript, RegulatoryFeature]</td>\n",
       "      <td>CodingTranscript</td>\n",
       "      <td>189</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Consequence</th>\n",
       "      <td>248</td>\n",
       "      <td>[3PRIME_UTR, STOP_GAINED, FRAME_SHIFT, NON_SYN...</td>\n",
       "      <td>NON_SYNONYMOUS</td>\n",
       "      <td>107</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ConsDetail</th>\n",
       "      <td>248</td>\n",
       "      <td>[3_prime_UTR, stop_gained,splice, stop_gained,...</td>\n",
       "      <td>missense</td>\n",
       "      <td>102</td>\n",
       "      <td>21</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Segway</th>\n",
       "      <td>248</td>\n",
       "      <td>[R2, F0, R1, TF2, TF0, L0, R4, L1, R3, R0, GE0...</td>\n",
       "      <td>TF0</td>\n",
       "      <td>76</td>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GeneID</th>\n",
       "      <td>246</td>\n",
       "      <td>[ENSG00000167306, nan]</td>\n",
       "      <td>ENSG00000167306</td>\n",
       "      <td>246</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>FeatureID</th>\n",
       "      <td>248</td>\n",
       "      <td>[ENST00000285039, ENSR00000525850, ENSR0000011...</td>\n",
       "      <td>ENST00000285039</td>\n",
       "      <td>246</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CCDS</th>\n",
       "      <td>246</td>\n",
       "      <td>[CCDS42436.1, nan]</td>\n",
       "      <td>CCDS42436.1</td>\n",
       "      <td>246</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GeneName</th>\n",
       "      <td>246</td>\n",
       "      <td>[MYO5B, nan]</td>\n",
       "      <td>MYO5B</td>\n",
       "      <td>246</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Domain</th>\n",
       "      <td>175</td>\n",
       "      <td>[nan, ndomain, ncoils]</td>\n",
       "      <td>ndomain</td>\n",
       "      <td>144</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Dst2SplType</th>\n",
       "      <td>88</td>\n",
       "      <td>[nan, DONOR, ACCEPTOR]</td>\n",
       "      <td>ACCEPTOR</td>\n",
       "      <td>49</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Exon</th>\n",
       "      <td>208</td>\n",
       "      <td>[40/40, 39/40, 36/40, 35/40, nan, 33/40, 31/40...</td>\n",
       "      <td>10/40</td>\n",
       "      <td>20</td>\n",
       "      <td>36</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Intron</th>\n",
       "      <td>39</td>\n",
       "      <td>[nan, 33/39, 22/39, 18/39, 15/39, 10/39, 8/39,...</td>\n",
       "      <td>16/39</td>\n",
       "      <td>8</td>\n",
       "      <td>25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>oAA</th>\n",
       "      <td>170</td>\n",
       "      <td>[nan, Q, R, L, -, AR, P, I, F, C, N, G, W, S, ...</td>\n",
       "      <td>R</td>\n",
       "      <td>28</td>\n",
       "      <td>23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>nAA</th>\n",
       "      <td>170</td>\n",
       "      <td>[nan, *, R, P, G, L, C, F, S, W, H, V, E, M, Y...</td>\n",
       "      <td>*</td>\n",
       "      <td>34</td>\n",
       "      <td>22</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PolyPhenCat</th>\n",
       "      <td>107</td>\n",
       "      <td>[nan, probably_damaging, benign, possibly_dama...</td>\n",
       "      <td>benign</td>\n",
       "      <td>52</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SIFTcat</th>\n",
       "      <td>107</td>\n",
       "      <td>[nan, deleterious, tolerated]</td>\n",
       "      <td>tolerated</td>\n",
       "      <td>56</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "            count                                             unique  \\\n",
       "ID            248  [MYO5B:c.5616-2A>G, MYO5B:c.5392C>T, MYO5B:c.5...   \n",
       "REF           248  [T, G, C, AA, A, CGGG, TC, AG, TCTGA, CA, TA, ...   \n",
       "ALT           248  [C, A, CCA, GC, G, AGAG, CC, T, TC, TGAG, TTCT...   \n",
       "INFO          248                               [Pathogenic, Benign]   \n",
       "Type          248                                    [SNV, INS, DEL]   \n",
       "isTv          218                                 [False, nan, True]   \n",
       "AnnoType      248  [Transcript, CodingTranscript, RegulatoryFeature]   \n",
       "Consequence   248  [3PRIME_UTR, STOP_GAINED, FRAME_SHIFT, NON_SYN...   \n",
       "ConsDetail    248  [3_prime_UTR, stop_gained,splice, stop_gained,...   \n",
       "Segway        248  [R2, F0, R1, TF2, TF0, L0, R4, L1, R3, R0, GE0...   \n",
       "GeneID        246                             [ENSG00000167306, nan]   \n",
       "FeatureID     248  [ENST00000285039, ENSR00000525850, ENSR0000011...   \n",
       "CCDS          246                                 [CCDS42436.1, nan]   \n",
       "GeneName      246                                       [MYO5B, nan]   \n",
       "Domain        175                             [nan, ndomain, ncoils]   \n",
       "Dst2SplType    88                             [nan, DONOR, ACCEPTOR]   \n",
       "Exon          208  [40/40, 39/40, 36/40, 35/40, nan, 33/40, 31/40...   \n",
       "Intron         39  [nan, 33/39, 22/39, 18/39, 15/39, 10/39, 8/39,...   \n",
       "oAA           170  [nan, Q, R, L, -, AR, P, I, F, C, N, G, W, S, ...   \n",
       "nAA           170  [nan, *, R, P, G, L, C, F, S, W, H, V, E, M, Y...   \n",
       "PolyPhenCat   107  [nan, probably_damaging, benign, possibly_dama...   \n",
       "SIFTcat       107                      [nan, deleterious, tolerated]   \n",
       "\n",
       "                             top freq  nr_unique  \n",
       "ID           PopulationGAVINr0.4  146        103  \n",
       "REF                            C   75         20  \n",
       "ALT                            A   77         14  \n",
       "INFO                      Benign  195          2  \n",
       "Type                         SNV  219          3  \n",
       "isTv                       False  134          3  \n",
       "AnnoType        CodingTranscript  189          3  \n",
       "Consequence       NON_SYNONYMOUS  107         10  \n",
       "ConsDetail              missense  102         21  \n",
       "Segway                       TF0   76         17  \n",
       "GeneID           ENSG00000167306  246          2  \n",
       "FeatureID        ENST00000285039  246          3  \n",
       "CCDS                 CCDS42436.1  246          2  \n",
       "GeneName                   MYO5B  246          2  \n",
       "Domain                   ndomain  144          3  \n",
       "Dst2SplType             ACCEPTOR   49          3  \n",
       "Exon                       10/40   20         36  \n",
       "Intron                     16/39    8         25  \n",
       "oAA                            R   28         23  \n",
       "nAA                            *   34         22  \n",
       "PolyPhenCat               benign   52          4  \n",
       "SIFTcat                tolerated   56          3  "
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# display the description for columns with object type of values in the table\n",
    "datatable_obj = datatable_pd.select_dtypes(include=[np.object])\n",
    "des_obj = datatable_obj.describe()\n",
    "nr_unique = []\n",
    "unique =[]\n",
    "for col_name in datatable_obj.columns:\n",
    "    nr_unique.append(len(datatable_obj[col_name].unique()))\n",
    "    unique.append(datatable_obj[col_name].unique())\n",
    "assert len(nr_unique) == len(des_obj.columns) == len(unique)\n",
    "des_obj = des_obj.T\n",
    "des_obj['nr_unique'] = nr_unique\n",
    "des_obj['unique'] = unique\n",
    "des_obj"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# check if there is missing values for RawScore column\n",
    "datatable_pd['RawScore'].isnull().values.ravel().sum()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "targetScan 248\n",
      "motifECount 248\n",
      "motifEName 248\n",
      "motifEHIPos 248\n",
      "motifEScoreChng 248\n"
     ]
    }
   ],
   "source": [
    "# columns without any values\n",
    "null_amount = []\n",
    "for column in datatable_pd.columns:\n",
    "    null = datatable_pd[column].isnull().values.ravel().sum()\n",
    "    if null > 247: # columns without any values\n",
    "        print(column, null)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "# convert all objects in datatable_pd to dummies\n",
    "categories={} # contains all the levels in those feature columns\n",
    "categorical_feature_names = [col for col in datatable_pd.select_dtypes(include=[np.object]).columns \n",
    "                             if col not in ['INFO','ID','GeneID','FeatureID','GeneName','REF','ALT','CCDS','AnnoType']]\n",
    "for f in datatable_pd.columns:\n",
    "    datatable_pd[f] = datatable_pd[f].astype('category')\n",
    "    categories[f] = datatable_pd[f].cat.categories\n",
    "\n",
    "new_datatable = pd.get_dummies(datatable_pd,columns=categorical_feature_names)\n",
    "new_datatable['INFO'] = new_datatable['INFO'].cat.codes\n",
    "\n",
    "# drop CHROM, POS and etc\n",
    "new_datatable = new_datatable.drop(columns=['CHROM','POS','ID','GeneID',\n",
    "                                            'FeatureID','GeneName','REF',\n",
    "                                            'ALT','CCDS','AnnoType'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>INFO</th>\n",
       "      <th>Length</th>\n",
       "      <th>isDerived</th>\n",
       "      <th>ConsScore</th>\n",
       "      <th>GC</th>\n",
       "      <th>CpG</th>\n",
       "      <th>mapAbility20bp</th>\n",
       "      <th>mapAbility35bp</th>\n",
       "      <th>scoreSegDup</th>\n",
       "      <th>priPhCons</th>\n",
       "      <th>...</th>\n",
       "      <th>nAA_S</th>\n",
       "      <th>nAA_T</th>\n",
       "      <th>nAA_V</th>\n",
       "      <th>nAA_W</th>\n",
       "      <th>nAA_Y</th>\n",
       "      <th>PolyPhenCat_benign</th>\n",
       "      <th>PolyPhenCat_possibly_damaging</th>\n",
       "      <th>PolyPhenCat_probably_damaging</th>\n",
       "      <th>SIFTcat_deleterious</th>\n",
       "      <th>SIFTcat_tolerated</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>True</td>\n",
       "      <td>2</td>\n",
       "      <td>0.38</td>\n",
       "      <td>0.00</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.93</td>\n",
       "      <td>0.159</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>True</td>\n",
       "      <td>8</td>\n",
       "      <td>0.32</td>\n",
       "      <td>0.04</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.908</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>True</td>\n",
       "      <td>8</td>\n",
       "      <td>0.32</td>\n",
       "      <td>0.04</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.984</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>True</td>\n",
       "      <td>8</td>\n",
       "      <td>0.51</td>\n",
       "      <td>0.07</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.584</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>True</td>\n",
       "      <td>7</td>\n",
       "      <td>0.49</td>\n",
       "      <td>0.05</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.993</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows Ã— 258 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   INFO Length isDerived ConsScore    GC   CpG mapAbility20bp mapAbility35bp  \\\n",
       "0     1      0      True         2  0.38  0.00            1.0            1.0   \n",
       "1     1      0      True         8  0.32  0.04            1.0            1.0   \n",
       "2     1      0      True         8  0.32  0.04            1.0            1.0   \n",
       "3     1      0      True         8  0.51  0.07            1.0            1.0   \n",
       "4     1      2      True         7  0.49  0.05            1.0            1.0   \n",
       "\n",
       "  scoreSegDup priPhCons        ...        nAA_S nAA_T nAA_V nAA_W nAA_Y  \\\n",
       "0        0.93     0.159        ...            0     0     0     0     0   \n",
       "1         NaN     0.908        ...            0     0     0     0     0   \n",
       "2         NaN     0.984        ...            0     0     0     0     0   \n",
       "3         NaN     0.584        ...            0     0     0     0     0   \n",
       "4         NaN     0.993        ...            0     0     0     0     0   \n",
       "\n",
       "  PolyPhenCat_benign PolyPhenCat_possibly_damaging  \\\n",
       "0                  0                             0   \n",
       "1                  0                             0   \n",
       "2                  0                             0   \n",
       "3                  0                             0   \n",
       "4                  0                             0   \n",
       "\n",
       "  PolyPhenCat_probably_damaging SIFTcat_deleterious SIFTcat_tolerated  \n",
       "0                             0                   0                 0  \n",
       "1                             0                   0                 0  \n",
       "2                             0                   0                 0  \n",
       "3                             0                   0                 0  \n",
       "4                             0                   0                 0  \n",
       "\n",
       "[5 rows x 258 columns]"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "new_datatable.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "scoreSegDup 229\n",
      "GerpRS 32\n",
      "GerpRSpval 32\n",
      "mirSVR-Score 230\n",
      "mirSVR-E 230\n",
      "mirSVR-Aln 230\n",
      "targetScan 248\n",
      "EncExp 8\n",
      "EncOCC 220\n",
      "EncOCCombPVal 220\n",
      "EncOCDNasePVal 220\n",
      "EncOCFairePVal 220\n",
      "EncOCpolIIPVal 220\n",
      "EncOCctcfPVal 220\n",
      "EncOCmycPVal 220\n",
      "EncOCDNaseSig 220\n",
      "EncOCFaireSig 220\n",
      "EncOCpolIISig 220\n",
      "EncOCctcfSig 220\n",
      "EncOCmycSig 220\n",
      "tOverlapMotifs 241\n",
      "motifDist 241\n",
      "motifECount 248\n",
      "motifEName 248\n",
      "motifEHIPos 248\n",
      "motifEScoreChng 248\n",
      "TFBS 227\n",
      "TFBSPeaks 227\n",
      "TFBSPeaksMax 227\n",
      "ESP_AF 214\n",
      "ESP_AFR 214\n",
      "ESP_EUR 214\n",
      "TG_AF 205\n",
      "TG_ASN 219\n",
      "TG_AMR 207\n",
      "TG_AFR 210\n",
      "TG_EUR 214\n",
      "cDNApos 40\n",
      "relcDNApos 40\n",
      "CDSpos 58\n",
      "relCDSpos 58\n",
      "protPos 58\n",
      "relProtPos 58\n",
      "Dst2Splice 160\n",
      "Grantham 141\n",
      "PolyPhenVal 141\n",
      "SIFTval 141\n"
     ]
    }
   ],
   "source": [
    "# print columns without missing values\n",
    "null_amount = []\n",
    "for column in new_datatable.columns:\n",
    "    null = new_datatable[column].isnull().values.ravel().sum()\n",
    "    if null > 0:\n",
    "        print(column, null)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(186, 258) (62, 258)\n"
     ]
    }
   ],
   "source": [
    "# split train and test datasets\n",
    "train, test = train_test_split(new_datatable,test_size=0.25)\n",
    "print(train.shape, test.shape)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "tp fn fp tn\n",
      "0  13 0  49\n"
     ]
    }
   ],
   "source": [
    "# model using SVM with CADD\n",
    "model_svm = SVC()\n",
    "model_svm.fit(train['RawScore'].values.reshape(-1,1),train['INFO'])\n",
    "pred_svm = model_svm.predict(test['RawScore'].values.reshape(-1,1))\n",
    "tn, fp, fn, tp = confusion_matrix(test['INFO'],pred_svm).ravel()\n",
    "print('tp fn fp tn\\n%d  %d %d  %d'%(tp,fn,fp,tn))\n",
    "# this model assigns negative to all samples"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "coefficients = pd.concat([pd.DataFrame(train.columns),pd.DataFrame(np.transpose(model.coef_))], axis = 1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0    verPhyloP_6.334\n",
       "0           0.553367\n",
       "dtype: object"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "coefficients.max()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "py3",
   "language": "python",
   "name": "py3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
