{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import os\n",
    "import numpy as np\n",
    "\n",
    "from lib.read_data import dataset,Datasets"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "def display_column_names(column_names,size = 4):\n",
    "    '''\n",
    "    when there are too many columns\n",
    "    display every four column names in a line\n",
    "    '''\n",
    "    for i in range(1,len(column_names),size):\n",
    "        print(column_names[i:i+4].values)\n",
    "\n",
    "def add_new_column_to_cadd_gavin_tables(cadd_anno,gavin_anno):\n",
    "    '''\n",
    "    very specific for the case here only\n",
    "    probably needs better structuring\n",
    "    '''\n",
    "    cadd_anno['chr_pos'] = cadd_anno.apply(lambda row: '%s_%s'%\n",
    "                                         (row['#Chrom'],row['Pos']),axis=1)\n",
    "    gavin_anno['chr_pos'] = gavin_anno.apply(lambda row: '%s_%s'%\n",
    "                                         (row['chr'],row['pos']),axis=1)\n",
    "    print('Added new columns to both tables')    \n",
    "\n",
    "def compare_two_anno_tables(cadd_anno, gavin_anno):\n",
    "    '''\n",
    "    examine whether all enriched varaints are del type\n",
    "    input: Pandas.DataFrame\n",
    "    '''\n",
    "    different_variants = {}\n",
    "    diff_set = set(cadd_anno['chr_pos']) - set(gavin_anno['chr_pos'])\n",
    "#     return diff_set\n",
    "    for variant in diff_set:\n",
    "        chr_id,pos_id = variant.split('_')\n",
    "        pos_id = np.int64(pos_id)\n",
    "        variant_info = cadd_anno.loc[(cadd_anno['#Chrom']==chr_id)&(cadd_anno['Pos']==pos_id),:]\n",
    "        different_variants[variant] = variant_info\n",
    "    return different_variants        \n",
    "        \n",
    "      \n",
    "def read_data_set(data_table,test_size=0.25,BENCHMARK=False):\n",
    "    '''\n",
    "    convert a pandas dataframe data table into Datasets(dataset,dataset)\n",
    "    '''\n",
    "    train, test = train_test_split(data_table,test_size=0.25)\n",
    "    train_x = train[[col for col in train.columns\n",
    "                     if col not in ['group_PATHOGENIC','group_BENIGN']]]\n",
    "    features = train_x.columns\n",
    "    train_x = np.array(train_x)\n",
    "    test_x = np.array(test[[col for col in train.columns\n",
    "                            if col not in ['group']]])\n",
    "    train_y = np.array(train['group'],dtype=np.int8)\n",
    "    test_y = np.array(test['group'],dtype=np.int8)\n",
    "\n",
    "\n",
    "    if BENCHMARK:\n",
    "        return Datasets(train=dataset(train_x,train_y,features),\n",
    "                        test=dataset(test_x,test_y,features)),\\\n",
    "                        train['gavin_res'],\\\n",
    "                        test['gavin_res']\n",
    "    return Datasets(train=dataset(train_x,train_y,features),\n",
    "                    test=dataset(test_x,test_y,features))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "# load original cadd annotation\n",
    "cadd_file = os.path.join('data','cadd_annotation.tsv')\n",
    "dtype = {'#Chrom':np.object,\n",
    "         'isDerived':np.object,\n",
    "         'motifEName':np.object,\n",
    "         'motifEHIPos':np.object,\n",
    "         'PolyPhenCat':np.object,\n",
    "         'SIFTcat':np.object}\n",
    "cadd_anno = pd.read_csv(cadd_file,sep='\\t',dtype=dtype)\n",
    "\n",
    "# load gavin annotation\n",
    "gavin_file = os.path.join('data','calibrationvariants_r0.4.tsv')\n",
    "gavin_anno = pd.read_csv(gavin_file,sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#Chrom</th>\n",
       "      <th>Pos</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Anc</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Type</th>\n",
       "      <th>Length</th>\n",
       "      <th>isTv</th>\n",
       "      <th>isDerived</th>\n",
       "      <th>AnnoType</th>\n",
       "      <th>...</th>\n",
       "      <th>Intron</th>\n",
       "      <th>oAA</th>\n",
       "      <th>nAA</th>\n",
       "      <th>Grantham</th>\n",
       "      <th>PolyPhenCat</th>\n",
       "      <th>PolyPhenVal</th>\n",
       "      <th>SIFTcat</th>\n",
       "      <th>SIFTval</th>\n",
       "      <th>RawScore</th>\n",
       "      <th>PHRED</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>1167659</td>\n",
       "      <td>A</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>False</td>\n",
       "      <td>TRUE</td>\n",
       "      <td>CodingTranscript</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>M</td>\n",
       "      <td>V</td>\n",
       "      <td>21.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>deleterious</td>\n",
       "      <td>0.00</td>\n",
       "      <td>1.106787</td>\n",
       "      <td>11.25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>1167674</td>\n",
       "      <td>C</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>False</td>\n",
       "      <td>TRUE</td>\n",
       "      <td>CodingTranscript</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R</td>\n",
       "      <td>W</td>\n",
       "      <td>101.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>deleterious</td>\n",
       "      <td>0.03</td>\n",
       "      <td>4.482313</td>\n",
       "      <td>24.20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>1167680</td>\n",
       "      <td>TGGC</td>\n",
       "      <td>TGGC</td>\n",
       "      <td>T</td>\n",
       "      <td>DEL</td>\n",
       "      <td>3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>TRUE</td>\n",
       "      <td>CodingTranscript</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.924056</td>\n",
       "      <td>15.74</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>1167680</td>\n",
       "      <td>TGGC</td>\n",
       "      <td>TGGC</td>\n",
       "      <td>T</td>\n",
       "      <td>DEL</td>\n",
       "      <td>3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>TRUE</td>\n",
       "      <td>Intergenic</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.924056</td>\n",
       "      <td>15.74</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>1167680</td>\n",
       "      <td>TGGC</td>\n",
       "      <td>TGGC</td>\n",
       "      <td>T</td>\n",
       "      <td>DEL</td>\n",
       "      <td>3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>TRUE</td>\n",
       "      <td>RegulatoryFeature</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.924056</td>\n",
       "      <td>15.74</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows Ã— 116 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  #Chrom      Pos   Ref   Anc Alt Type  Length   isTv isDerived  \\\n",
       "0      1  1167659     A     A   G  SNV       0  False      TRUE   \n",
       "1      1  1167674     C     C   T  SNV       0  False      TRUE   \n",
       "2      1  1167680  TGGC  TGGC   T  DEL       3    NaN      TRUE   \n",
       "3      1  1167680  TGGC  TGGC   T  DEL       3    NaN      TRUE   \n",
       "4      1  1167680  TGGC  TGGC   T  DEL       3    NaN      TRUE   \n",
       "\n",
       "            AnnoType  ...   Intron  oAA  nAA  Grantham  PolyPhenCat  \\\n",
       "0   CodingTranscript  ...      NaN    M    V      21.0          NaN   \n",
       "1   CodingTranscript  ...      NaN    R    W     101.0          NaN   \n",
       "2   CodingTranscript  ...      NaN    R    -       NaN          NaN   \n",
       "3         Intergenic  ...      NaN  NaN  NaN       NaN          NaN   \n",
       "4  RegulatoryFeature  ...      NaN  NaN  NaN       NaN          NaN   \n",
       "\n",
       "   PolyPhenVal      SIFTcat  SIFTval  RawScore  PHRED  \n",
       "0          NaN  deleterious     0.00  1.106787  11.25  \n",
       "1          NaN  deleterious     0.03  4.482313  24.20  \n",
       "2          NaN          NaN      NaN  1.924056  15.74  \n",
       "3          NaN          NaN      NaN  1.924056  15.74  \n",
       "4          NaN          NaN      NaN  1.924056  15.74  \n",
       "\n",
       "[5 rows x 116 columns]"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cadd_anno.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['Pos' 'Ref' 'Anc' 'Alt']\n",
      "['Type' 'Length' 'isTv' 'isDerived']\n",
      "['AnnoType' 'Consequence' 'ConsScore' 'ConsDetail']\n",
      "['GC' 'CpG' 'mapAbility20bp' 'mapAbility35bp']\n",
      "['scoreSegDup' 'priPhCons' 'mamPhCons' 'verPhCons']\n",
      "['priPhyloP' 'mamPhyloP' 'verPhyloP' 'GerpN']\n",
      "['GerpS' 'GerpRS' 'GerpRSpval' 'bStatistic']\n",
      "['mutIndex' 'dnaHelT' 'dnaMGW' 'dnaProT']\n",
      "['dnaRoll' 'mirSVR-Score' 'mirSVR-E' 'mirSVR-Aln']\n",
      "['targetScan' 'fitCons' 'cHmmTssA' 'cHmmTssAFlnk']\n",
      "['cHmmTxFlnk' 'cHmmTx' 'cHmmTxWk' 'cHmmEnhG']\n",
      "['cHmmEnh' 'cHmmZnfRpts' 'cHmmHet' 'cHmmTssBiv']\n",
      "['cHmmBivFlnk' 'cHmmEnhBiv' 'cHmmReprPC' 'cHmmReprPCWk']\n",
      "['cHmmQuies' 'EncExp' 'EncH3K27Ac' 'EncH3K4Me1']\n",
      "['EncH3K4Me3' 'EncNucleo' 'EncOCC' 'EncOCCombPVal']\n",
      "['EncOCDNasePVal' 'EncOCFairePVal' 'EncOCpolIIPVal' 'EncOCctcfPVal']\n",
      "['EncOCmycPVal' 'EncOCDNaseSig' 'EncOCFaireSig' 'EncOCpolIISig']\n",
      "['EncOCctcfSig' 'EncOCmycSig' 'Segway' 'tOverlapMotifs']\n",
      "['motifDist' 'motifECount' 'motifEName' 'motifEHIPos']\n",
      "['motifEScoreChng' 'TFBS' 'TFBSPeaks' 'TFBSPeaksMax']\n",
      "['isKnownVariant' 'ESP_AF' 'ESP_AFR' 'ESP_EUR']\n",
      "['TG_AF' 'TG_ASN' 'TG_AMR' 'TG_AFR']\n",
      "['TG_EUR' 'minDistTSS' 'minDistTSE' 'GeneID']\n",
      "['FeatureID' 'CCDS' 'GeneName' 'cDNApos']\n",
      "['relcDNApos' 'CDSpos' 'relCDSpos' 'protPos']\n",
      "['relProtPos' 'Domain' 'Dst2Splice' 'Dst2SplType']\n",
      "['Exon' 'Intron' 'oAA' 'nAA']\n",
      "['Grantham' 'PolyPhenCat' 'PolyPhenVal' 'SIFTcat']\n",
      "['SIFTval' 'RawScore' 'PHRED']\n"
     ]
    }
   ],
   "source": [
    "display_column_names(cadd_anno.columns)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene</th>\n",
       "      <th>chr</th>\n",
       "      <th>pos</th>\n",
       "      <th>ref</th>\n",
       "      <th>alt</th>\n",
       "      <th>group</th>\n",
       "      <th>effect</th>\n",
       "      <th>impact</th>\n",
       "      <th>cadd</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NUP107</td>\n",
       "      <td>12</td>\n",
       "      <td>69107589</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>PATHOGENIC</td>\n",
       "      <td>splice_donor_variant</td>\n",
       "      <td>HIGH</td>\n",
       "      <td>24.70</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NUP107</td>\n",
       "      <td>12</td>\n",
       "      <td>69090619</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>PATHOGENIC</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>28.10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NUP107</td>\n",
       "      <td>12</td>\n",
       "      <td>69084526</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>PATHOGENIC</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>19.16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NUP107</td>\n",
       "      <td>12</td>\n",
       "      <td>69129114</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>PATHOGENIC</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>31.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NUP107</td>\n",
       "      <td>12</td>\n",
       "      <td>69109513</td>\n",
       "      <td>CAGAAG</td>\n",
       "      <td>C</td>\n",
       "      <td>PATHOGENIC</td>\n",
       "      <td>frameshift_variant</td>\n",
       "      <td>HIGH</td>\n",
       "      <td>35.00</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     gene chr       pos     ref alt       group                effect  \\\n",
       "0  NUP107  12  69107589       G   A  PATHOGENIC  splice_donor_variant   \n",
       "1  NUP107  12  69090619       G   T  PATHOGENIC      missense_variant   \n",
       "2  NUP107  12  69084526       G   A  PATHOGENIC      missense_variant   \n",
       "3  NUP107  12  69129114       A   C  PATHOGENIC      missense_variant   \n",
       "4  NUP107  12  69109513  CAGAAG   C  PATHOGENIC    frameshift_variant   \n",
       "\n",
       "     impact   cadd  \n",
       "0      HIGH  24.70  \n",
       "1  MODERATE  28.10  \n",
       "2  MODERATE  19.16  \n",
       "3  MODERATE  31.00  \n",
       "4      HIGH  35.00  "
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gavin_anno.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(930, 116) (846, 9)\n"
     ]
    }
   ],
   "source": [
    "myh7_cadd = cadd_anno.loc[cadd_anno['GeneName']=='MYH7',:].copy()\n",
    "myh7_gavin = gavin_anno.loc[gavin_anno['gene']=='MYH7',:].copy()\n",
    "print(myh7_cadd.shape,myh7_gavin.shape)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Added new columns to both tables\n"
     ]
    }
   ],
   "source": [
    "# annotate cadd_anno with group column in gavin_anno\n",
    "add_new_column_to_cadd_gavin_tables(myh7_cadd,myh7_gavin)\n",
    "# a dictionary with gavin group info\n",
    "chrpos_info_dic = myh7_gavin.set_index('chr_pos')['group'].to_dict()\n",
    "# map the myh7_cadd with the chrpos_info_dic\n",
    "def map_myh7_chrposDic(row):\n",
    "    if row in chrpos_info_dic:\n",
    "        return chrpos_info_dic[row]\n",
    "    else:\n",
    "        return 'NA'\n",
    "myh7_cadd['INFO'] = myh7_cadd['chr_pos'].apply(map_myh7_chrposDic)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(930, 118)"
      ]
     },
     "execution_count": 39,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "myh7_cadd.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {
    "scrolled": false
   },
   "outputs": [],
   "source": [
    "all_variants = myh7_cadd.drop(myh7_cadd.index[myh7_cadd.INFO=='NA'],axis=0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [],
   "source": [
    "all_variants.to_csv('data/mhy7.tsv',sep='\\t',index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# just for fun, try use the gavin annotation file for pathogenicity prediction"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "categorical_feature_names = \\\n",
    "gavin_anno.select_dtypes(include=[np.object]).columns\n",
    "for col in categorical_feature_names:\n",
    "    gavin_anno[col] = gavin_anno[col].astype('category')\n",
    "dummy_features = pd.get_dummies(gavin_anno,\n",
    "                                columns=categorical_feature_names)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(346278, 14199)"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dummy_features.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "cadd_scale = dummy_features['cadd'].max() - dummy_features['cadd'].min()\n",
    "dummy_features['cadd_scaled'] = (dummy_features['cadd']-\n",
    "                                 dummy_features['cadd'].min())/cadd_scale"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# need a better way for scoring the data otherwise too slow, probably not in txt format"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# save the processed table\n",
    "dummy_features.to_csv(os.path.join('data','processed_gavin_annotation.tsv'),sep='\\t',index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# if I don't use ['chr','gene','alt','ref']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "categorical_feature_names = ['group','impact','effect']\n",
    "for col in categorical_feature_names:\n",
    "    gavin_anno[col] = gavin_anno[col].astype('category')\n",
    "dummy_light = pd.get_dummies(gavin_anno,\n",
    "                                columns=categorical_feature_names)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene</th>\n",
       "      <th>chr</th>\n",
       "      <th>pos</th>\n",
       "      <th>ref</th>\n",
       "      <th>alt</th>\n",
       "      <th>cadd</th>\n",
       "      <th>group_PATHOGENIC</th>\n",
       "      <th>group_POPULATION</th>\n",
       "      <th>impact_HIGH</th>\n",
       "      <th>impact_LOW</th>\n",
       "      <th>...</th>\n",
       "      <th>effect_protein_protein_contact</th>\n",
       "      <th>effect_rare_amino_acid_variant</th>\n",
       "      <th>effect_splice_acceptor_variant</th>\n",
       "      <th>effect_splice_donor_variant</th>\n",
       "      <th>effect_splice_region_variant</th>\n",
       "      <th>effect_start_lost</th>\n",
       "      <th>effect_stop_gained</th>\n",
       "      <th>effect_stop_lost</th>\n",
       "      <th>effect_stop_retained_variant</th>\n",
       "      <th>effect_synonymous_variant</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NUP107</td>\n",
       "      <td>12</td>\n",
       "      <td>69107589</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>24.70</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NUP107</td>\n",
       "      <td>12</td>\n",
       "      <td>69090619</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>28.10</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NUP107</td>\n",
       "      <td>12</td>\n",
       "      <td>69084526</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>19.16</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NUP107</td>\n",
       "      <td>12</td>\n",
       "      <td>69129114</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>31.00</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NUP107</td>\n",
       "      <td>12</td>\n",
       "      <td>69109513</td>\n",
       "      <td>CAGAAG</td>\n",
       "      <td>C</td>\n",
       "      <td>35.00</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows Ã— 36 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     gene chr       pos     ref alt   cadd  group_PATHOGENIC  \\\n",
       "0  NUP107  12  69107589       G   A  24.70                 1   \n",
       "1  NUP107  12  69090619       G   T  28.10                 1   \n",
       "2  NUP107  12  69084526       G   A  19.16                 1   \n",
       "3  NUP107  12  69129114       A   C  31.00                 1   \n",
       "4  NUP107  12  69109513  CAGAAG   C  35.00                 1   \n",
       "\n",
       "   group_POPULATION  impact_HIGH  impact_LOW            ...              \\\n",
       "0                 0            1           0            ...               \n",
       "1                 0            0           0            ...               \n",
       "2                 0            0           0            ...               \n",
       "3                 0            0           0            ...               \n",
       "4                 0            1           0            ...               \n",
       "\n",
       "   effect_protein_protein_contact  effect_rare_amino_acid_variant  \\\n",
       "0                               0                               0   \n",
       "1                               0                               0   \n",
       "2                               0                               0   \n",
       "3                               0                               0   \n",
       "4                               0                               0   \n",
       "\n",
       "   effect_splice_acceptor_variant  effect_splice_donor_variant  \\\n",
       "0                               0                            1   \n",
       "1                               0                            0   \n",
       "2                               0                            0   \n",
       "3                               0                            0   \n",
       "4                               0                            0   \n",
       "\n",
       "   effect_splice_region_variant  effect_start_lost  effect_stop_gained  \\\n",
       "0                             0                  0                   0   \n",
       "1                             0                  0                   0   \n",
       "2                             0                  0                   0   \n",
       "3                             0                  0                   0   \n",
       "4                             0                  0                   0   \n",
       "\n",
       "   effect_stop_lost  effect_stop_retained_variant  effect_synonymous_variant  \n",
       "0                 0                             0                          0  \n",
       "1                 0                             0                          0  \n",
       "2                 0                             0                          0  \n",
       "3                 0                             0                          0  \n",
       "4                 0                             0                          0  \n",
       "\n",
       "[5 rows x 36 columns]"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dummy_light.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "py3",
   "language": "python",
   "name": "py3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.4"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
